---
title: "231115_MT2_manuscript_analyses"
output: 
  html_notebook: 
    highlight: haddock
    theme: readable
    toc: yes
---

# 1. MT2 sgRNA design and evaluation

## 1.1 Design sgRNA
```{bash}
#retrieve MT2_Mm consensus from Dfam database
wget https://www.dfam.org/releases/Dfam_3.6/families/Dfam_curatedonly.embl.gz .
grep MT2_Mm -A 30  Dfam_curatedonly.embl
#output as follow:
<!-- NM   MT2_Mm -->
<!-- XX -->
<!-- AC   DF0004155; -->
<!-- XX -->
<!-- DE   Long terminal repeat of a class 3 endogenous retrovirus from mouse. -->
<!-- XX -->
<!-- DR   Repbase; MERVL_LTR. -->
<!-- XX -->
<!-- KW   Long terminal repeat of retrovirus-like element; MT2_Mm. -->
<!-- XX -->
<!-- OS   Mus <genus> -->
<!-- OC   cellular organisms; Eukaryota; Opisthokonta; Metazoa; Eumetazoa; -->
<!-- OC   Bilateria; Deuterostomia; Chordata; Craniata <chordates>; Vertebrata -->
<!-- OC   <vertebrates>; Gnathostomata <vertebrates>; Teleostomi; Euteleostomi; -->
<!-- OC   Sarcopterygii; Dipnotetrapodomorpha; Tetrapoda; Amniota; Mammalia; -->
<!-- OC   Theria <mammals>; Eutheria; Boreoeutheria; Euarchontoglires; Glires; -->
<!-- OC   Rodentia; Myomorpha; Muroidea; Muridae; Murinae. -->
<!-- XX -->
<!-- CC -->
<!-- CC   RepeatMasker Annotations: -->
<!-- CC        Type: LTR -->
<!-- CC        SubType: ERVL -->
<!-- CC        Species: Mus_genus -->
<!-- CC        SearchStages: 40,50 -->
<!-- CC        BufferStages:  -->
<!-- XX -->
<!-- SQ   Sequence 493 BP; 126 A; 105 C; 115 G; 147 T; 0 other; -->
<!--      tgtagtggct attcctggtt gtcaacttga caatatttgg aatgaactac aatccggaat  60 -->
<!--      tggaaggctc accagtgacc cttatctgga ggcttggaga tccttatctg gatcttggtt  120 -->
<!--      tgaagatctt gagccatagt ggctatggat tccagaagat tgaatctccg agtttaagga  180 -->
<!--      acacaccttt aatctgggct acgcctttca tctgggatta aaggtgtggt ggaacacacc  240 -->
<!--      tttaatctgg gctacacctt ctgctggaga caatataagg acattggaag aagggagtct  300 -->
<!--      agctcttgct cttgctcctt cgcctgcttg ctgcgtgaga ctgagtaact gctagatcct  360 -->
<!--      tggacttcca ttcacagctg cgactgaaca attgttggga attgggctgc cgactgtaag  420 -->
<!--      tcatcaataa attcctttac tatctagaga ctatccataa gttctgtgac tctagagaac  480 -->
<!--      cctgactaat aca                                                     493 -->

#copy the 493 consensus as input and use CRISPOR (http://crispor.tefor.net) to design sgRNA 
# save output as csv for downstream analyses
```

## 1.2 sgRNA on- and off-target analyses

Process CRISPOR output using custom perl scripts (in folder: perl_scripts/)
```{bash}
### process CRISPOR output for cas-offinder analyses
perl process_CRISPOR_output4Cas_OFFinder.pl guides_MT2_Mm_consensus_GCA_001632555.1-unknownLoc.csv MT2_Mm

### run cas-offinder analyses
perl cas_offinder.pl MT2_Mm

### process cas-offinder outputs
perl process_cas_offinder_output.pl  guides_MT2_Mm_consensus_GCA_001632555.1-unknownLoc.csv MT2_Mm

### generate outputs for R graphing
perl related_TE_families_count.pl guides_MT2_Mm_consensus_GCA_001632555.1-unknownLoc.csv MT2_Mm
```

### 1.2.1 on- and off-target heatmap
```{r}
suppressMessages(library(RColorBrewer))
suppressMessages(library(circlize))
suppressMessages(library(ComplexHeatmap))
```

```{r fig.height=4, fig.width=5}
source("./utils.R")
#col = sgRNAs; #row = all MT2_Mm copies in mm10
#0: not-targeted
#1: targeted
#LTR_id Length X102rev.57.29.1.TTCAAACCAAGATCCAGATA X109forw.38.45.1.GAGGCTTGGAGATCCTTATC X116forw.37.45.1.GGAGATCCTTATCTGGATCT
#chr10:10161076-10161569    494   0                                     1
on_mis0 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis0_count.txt", header = T)
on_mis1 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis1_count.txt", header = T)
on_mis2 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis2_count.txt", header = T)
on_mis3 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis3_count.txt", header = T)

mat_on_mis0 <- t(as.matrix(on_mis0[, c(3:ncol(on_mis0))]))
mat_on_mis1 <- t(as.matrix(on_mis1[, c(3:ncol(on_mis1))]))
mat_on_mis2 <- t(as.matrix(on_mis2[, c(3:ncol(on_mis2))]))
mat_on_mis3 <- t(as.matrix(on_mis3[, c(3:ncol(on_mis3))]))

row.names(mat_on_mis0) <- unname(sapply(colnames(on_mis0)[3:ncol(on_mis0)], simplify_sg_id))
row.names(mat_on_mis1) <- unname(sapply(colnames(on_mis1)[3:ncol(on_mis1)], simplify_sg_id))
row.names(mat_on_mis2) <- unname(sapply(colnames(on_mis2)[3:ncol(on_mis2)], simplify_sg_id))
row.names(mat_on_mis3) <- unname(sapply(colnames(on_mis3)[3:ncol(on_mis3)], simplify_sg_id))

ha <- HeatmapAnnotation(length = on_mis0$Length, border = T,
                       col = list(length = colorRamp2(c(0, 500), c("white", "firebrick3"))),
                       annotation_name_side = "left")
onTarget_heatmap <- Heatmap(mat_on_mis0, col = c("grey90", "cornflowerblue"),
        cluster_rows = F, show_column_dend = F,
        cluster_columns = T, show_row_names = T, row_names_side = "left", 
        name = "On_Target", border = T, bottom_annotation = ha, width = unit(12, "cm"))
#onTarget_heatmap
```

```{r}
off_mis0 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_offTarget_mis0_count.txt", header = T)
off_mis1 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_offTarget_mis1_count.txt", header = T)
off_mis2 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_offTarget_mis2_count.txt", header = T)
off_mis3 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_offTarget_mis3_count.txt", header = T)

off_mis0$sg_id <- sapply(off_mis0$sg_id, simplify_sg_id)
off_mis1$sg_id <- sapply(off_mis1$sg_id, simplify_sg_id)
off_mis2$sg_id <- sapply(off_mis2$sg_id, simplify_sg_id)
off_mis3$sg_id <- sapply(off_mis3$sg_id, simplify_sg_id)

keep1 <- c("MT2C_Mm", "MT2B2", "MT2B", "MT2B1")
keep2 <- c("MT2C_Mm", "MT2B2", "MT2B", "MT2B1", "MT2A")

mat_off_mis0 <- as.matrix(off_mis0[, c(2:ncol(off_mis0))])[, keep1]
mat_off_mis1 <- as.matrix(off_mis1[, c(2:ncol(off_mis1))])[, keep2]
mat_off_mis2 <- as.matrix(off_mis2[, c(2:ncol(off_mis2))])[, keep2]
mat_off_mis3 <- as.matrix(off_mis3[, c(2:ncol(off_mis3))])[, keep2]

row.names(mat_off_mis0) <- off_mis0$sg_id
row.names(mat_off_mis1) <- off_mis1$sg_id
row.names(mat_off_mis2) <- off_mis2$sg_id
row.names(mat_off_mis3) <- off_mis3$sg_id

off_mis0_htp <- Heatmap(mat_off_mis0, colorRamp2(c(0, 400), 
                       c("white", "firebrick3")), show_row_names = F, 
                       column_names_side = "top", cluster_columns = T, 
                       show_column_dend = F, name = "Mismatch=0", border = T, width = unit(2, "cm"),
                       rect_gp  = gpar(col = "black", lwd = 0.5))

off_mis1_htp <- Heatmap(mat_off_mis1, colorRamp2(c(0, 400), 
                       c("white", "firebrick3")), show_row_names = F, 
                       column_names_side = "top", cluster_columns = T, 
                       show_column_dend = F, name = "Mismatch=1", border = T, width = unit(2, "cm"),
                       rect_gp  = gpar(col = "black", lwd = 0.5))

off_mis2_htp <- Heatmap(mat_off_mis2, colorRamp2(c(0, 400), 
                       c("white", "firebrick3")), show_row_names = F, 
                       column_names_side = "top", cluster_columns = T, 
                       show_column_dend = F, name = "Mismatch=2", border = T, width = unit(2, "cm"),
                       rect_gp  = gpar(col = "black", lwd = 0.5))

off_mis3_htp <- Heatmap(mat_off_mis3, colorRamp2(c(0, 400), 
                       c("white", "firebrick3")), show_row_names = F, 
                       column_names_side = "top", cluster_columns = T, 
                       show_column_dend = F, name = "Mismatch=3", border = T, width = unit(2, "cm"),
                       rect_gp  = gpar(col = "black", lwd = 0.5))
#off_mis0_htp + off_mis1_htp + off_mis2_htp + off_mis3_htp
```

```{r, fig.height=4, fig.width=6}
onTarget_heatmap + off_mis0_htp + off_mis1_htp + off_mis2_htp + off_mis3_htp
pdf("./figures/231115_MT2_Mm_sgRNA_on_off_target.pdf", width = 14, height = 8)
onTarget_heatmap + off_mis0_htp + off_mis1_htp + off_mis2_htp + off_mis3_htp
dev.off()
```

Based on this heatmap, pick following 6x sgRNAs
"196forw_50_36", 
"285forw_59_63", 
"109forw_38_45",
"410rev_63_34", 
"90forw_63_37", 
"358rev_64_56"

### 1.2.2 sgRNA stack plot
```{r fig.height=2, fig.width=3}
### for MT2_Mm
on_mis0 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis0_count.txt", header = T)
on_mis1 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis1_count.txt", header = T)
on_mis2 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis2_count.txt", header = T)
on_mis3 <- read.table("./R_input/sgRNA_design/MT2_Mm_20bp_sgRNA_onTarget_mis3_count.txt", header = T)

simple_sg_id <- unname(sapply(colnames(on_mis0)[3:ncol(on_mis0)], simplify_sg_id))
colnames(on_mis0) <- c("LTR_id", "Length", simple_sg_id)          
colnames(on_mis1) <- c("LTR_id", "Length", simple_sg_id)
colnames(on_mis2) <- c("LTR_id", "Length", simple_sg_id)
colnames(on_mis3) <- c("LTR_id", "Length", simple_sg_id)

sg_set <- c("196forw_50_36", "285forw_59_63", "109forw_38_45",
             "410rev_63_34", "90forw_63_37", "358rev_64_56")

#calculate # of MT2_Mm targeted at 0 mismatch
mis0 <- rowSums(on_mis0[, sg_set])
tmp <- as.data.frame(table(mis0))
#tmp
#0	146			
#1	80			
#2	142			
#3	176			
#4	324			
#5	607			
#6	1192	

pdf("./figures/231115_MT2_Mm_targeting_prediction.pdf", width = 6, height = 5)
Stack_plot(on_mis0, on_mis1, on_mis2, on_mis3, sg_set, 
                   "MT2_Mm", max = 6)
dev.off()
Stack_plot(on_mis0, on_mis1, on_mis2, on_mis3, sg_set, 
                   "MT2_Mm", max = 6)
### For other closely related subfamilies
count_files <- list()
count_files[["MT2C_Mm_mis0"]] <- "./R_input/sgRNA_design/MT2C_Mm_20bp_sgRNA_onTarget_mis0_count.txt"
count_files[["MT2C_Mm_mis1"]] <- "./R_input/sgRNA_design/MT2C_Mm_20bp_sgRNA_onTarget_mis1_count.txt"
count_files[["MT2C_Mm_mis2"]] <- "./R_input/sgRNA_design/MT2C_Mm_20bp_sgRNA_onTarget_mis2_count.txt"
count_files[["MT2C_Mm_mis3"]] <- "./R_input/sgRNA_design/MT2C_Mm_20bp_sgRNA_onTarget_mis3_count.txt"

count_files[["MT2B_mis0"]] <- "./R_input/sgRNA_design/MT2B_20bp_sgRNA_onTarget_mis0_count.txt"
count_files[["MT2B_mis1"]] <- "./R_input/sgRNA_design/MT2B_20bp_sgRNA_onTarget_mis1_count.txt"
count_files[["MT2B_mis2"]] <- "./R_input/sgRNA_design/MT2B_20bp_sgRNA_onTarget_mis2_count.txt"
count_files[["MT2B_mis3"]] <- "./R_input/sgRNA_design/MT2B_20bp_sgRNA_onTarget_mis3_count.txt"
  
count_files[["MT2B1_mis0"]] <- "./R_input/sgRNA_design/MT2B1_20bp_sgRNA_onTarget_mis0_count.txt"
count_files[["MT2B1_mis1"]] <- "./R_input/sgRNA_design/MT2B1_20bp_sgRNA_onTarget_mis1_count.txt"
count_files[["MT2B1_mis2"]] <- "./R_input/sgRNA_design/MT2B1_20bp_sgRNA_onTarget_mis2_count.txt"
count_files[["MT2B1_mis3"]] <- "./R_input/sgRNA_design/MT2B1_20bp_sgRNA_onTarget_mis3_count.txt"

count_files[["MT2B2_mis0"]] <- "./R_input/sgRNA_design/MT2B2_20bp_sgRNA_onTarget_mis0_count.txt"
count_files[["MT2B2_mis1"]] <- "./R_input/sgRNA_design/MT2B2_20bp_sgRNA_onTarget_mis1_count.txt"
count_files[["MT2B2_mis2"]] <- "./R_input/sgRNA_design/MT2B2_20bp_sgRNA_onTarget_mis2_count.txt"
count_files[["MT2B2_mis3"]] <- "./R_input/sgRNA_design/MT2B2_20bp_sgRNA_onTarget_mis3_count.txt"

df.list <- list()
for(i in 1:length(count_files)){
  df.list[[i]] <- read.table(count_files[[i]], header = T)
  simple_sg_id <- unname(sapply(colnames(df.list[[i]])[3:ncol(df.list[[i]])], 
                                simplify_sg_id))
  colnames(df.list[[i]]) <- c("LTR_id", "Length", simple_sg_id)
}
names(df.list) <- names(count_files)

#calculate # of MT2C_Mm targeted at 0 mismatch
mis0 <- rowSums(df.list$MT2C_Mm_mis0[, sg_set])
tmp <- as.data.frame(table(mis0))
tmp
#0	1425			
#1	452			
#2	76			
#3	18			
#4	9			
#5	2		
Stack_plot(df.list$MT2C_Mm_mis0, 
                   df.list$MT2C_Mm_mis1,
                   df.list$MT2C_Mm_mis2,
                   df.list$MT2C_Mm_mis3,
                   sg_set, 
                   "MT2C_Mm", max = 6)
Stack_plot(df.list$MT2B_mis0, 
                   df.list$MT2B_mis1,
                   df.list$MT2B_mis2,
                   df.list$MT2B_mis3,
                   sg_set, 
                   "MT2B", max = 6)
Stack_plot(df.list$MT2B1_mis0, 
                   df.list$MT2B1_mis1,
                   df.list$MT2B1_mis2,
                   df.list$MT2B1_mis3,
                   sg_set, 
                   "MT2B1", max = 6)
Stack_plot(df.list$MT2B2_mis0, 
                   df.list$MT2B2_mis1,
                   df.list$MT2B2_mis2,
                   df.list$MT2B2_mis3,
                   sg_set, 
                   "MT2B2", max = 6)

pdf("./figures/231115_Others_targeting_prediction.pdf", width = 6, height = 5)
Stack_plot(df.list$MT2C_Mm_mis0, 
                   df.list$MT2C_Mm_mis1,
                   df.list$MT2C_Mm_mis2,
                   df.list$MT2C_Mm_mis3,
                   sg_set, 
                   "MT2C_Mm", max = 6)
Stack_plot(df.list$MT2B_mis0, 
                   df.list$MT2B_mis1,
                   df.list$MT2B_mis2,
                   df.list$MT2B_mis3,
                   sg_set, 
                   "MT2B", max = 6)
Stack_plot(df.list$MT2B1_mis0, 
                   df.list$MT2B1_mis1,
                   df.list$MT2B1_mis2,
                   df.list$MT2B1_mis3,
                   sg_set, 
                   "MT2B1", max = 6)
Stack_plot(df.list$MT2B2_mis0, 
                   df.list$MT2B2_mis1,
                   df.list$MT2B2_mis2,
                   df.list$MT2B2_mis3,
                   sg_set, 
                   "MT2B2", max = 6)
dev.off()
```

# 2. dCas9 binding analyses

## 2.1 input annotations and peaks
```{r input mm10 repeats}
suppressMessages(library("rtracklayer"))

# import mm10 repeat masker bed file
mm10_rmsk <- import.bed("/Volumes/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed")
length(mm10_rmsk) #3725827

# get MT2_Mm & MT2C_Mm
MT2Mm <- mm10_rmsk[grep("MT2_Mm", mm10_rmsk$name),]
MT2CMm <- mm10_rmsk[grep("MT2C_Mm", mm10_rmsk$name),]

length(MT2Mm) #2667
length(MT2CMm) #1982
```

```{r input narrowPeak}
# declare names and types for the extra BED columns
extraCols_narrowPeak <- c(FoldChange="numeric", pVal="numeric",
qVal="numeric", summit="integer")

#following peak files were input
CTR_dCas9_peaks <- import.bed("./R_input/dCas9_ChIPseq/CTRi_ESCs_Cas9_ChIP_rep1_multi_peaks.narrowPeak",
                              extraCols=extraCols_narrowPeak)
CTR_HA_peaks <- import.bed("./R_input/dCas9_ChIPseq/CTRi_ESCs_HA_ChIP_rep1_multi_peaks.narrowPeak",
                           extraCols=extraCols_narrowPeak)
dCas9_peaks <- import.bed("./R_input/dCas9_ChIPseq/Cas9_ChIP_multi_peaks.narrowPeak",
                          extraCols=extraCols_narrowPeak)
HA_peaks <- import.bed("./R_input/dCas9_ChIPseq/HA_ChIP_multi_peaks.narrowPeak",
                       extraCols=extraCols_narrowPeak)

length(CTR_dCas9_peaks) #5195
length(CTR_HA_peaks) #1546
length(dCas9_peaks) #7529
length(HA_peaks) #4727
```
## 2.2 filtering peaks
### 2.2.1 filter peaks in non-dCas9 group

```{bash}
bedtools intersect -a Cas9_ChIP_multi_peaks.narrowPeak -b CTRi_ESCs_Cas9_ChIP_rep1_multi_peaks.narrowPeak -v > Cas9_ChIP_multi_peaks_filtered.narrowPeak
bedtools intersect -a HA_ChIP_multi_peaks.narrowPeak -b CTRi_ESCs_HA_ChIP_rep1_multi_peaks.narrowPeak -v > HA_ChIP_multi_peaks_filtered.narrowPeak
bedtools intersect -a Cas9_ChIP_multi_peaks_filtered.narrowPeak -b HA_ChIP_multi_peaks_filtered.narrowPeak > Cas9_HA_overlap.narrowPeak
```

### 2.2.2 overlap between dCas9 and HA ChIP
```{r overlap between dCas9 & HA peaks, fig.height = 1, figh.width = 2 }
dCas9_filtered_peaks <- import.bed("./R_input/dCas9_ChIPseq/Cas9_ChIP_multi_peaks_filtered.narrowPeak", 
                                   extraCols=extraCols_narrowPeak)
HA_filtered_peaks <- import.bed("./R_input/dCas9_ChIPseq/HA_ChIP_multi_peaks_filtered.narrowPeak",
                                extraCols=extraCols_narrowPeak)
length(dCas9_filtered_peaks) #5784
length(HA_filtered_peaks) #4144

dCas9_HA_overlap <- import.bed("./R_input/dCas9_ChIPseq/Cas9_HA_overlap.narrowPeak",
                               extraCols=extraCols_narrowPeak)
length(dCas9_HA_overlap) #4072

#make a venndiagram.
suppressMessages(library("VennDiagram"))
# draw.pairwise.venn(area1 = 5784,    # Draw pairwise venn diagram
#                    area2 = 4144,
#                    cross.area = 4072,
#                    fill = c("blue", "red"))
pdf("./figures/231115_vennDiagram_Cas9_HA_peaks.pdf", width = 5, height = 5)
#grid.newpage()                    # Create new plotting page
draw.pairwise.venn(area1 = 5784,    # Draw pairwise venn diagram
                   area2 = 4144,
                   cross.area = 4072,
                   fill = c("blue", "red"))
dev.off()
```

## 2.3 annotate peaks (MT2 or others)

### 2.3.1 dCas9 peak annotations
```{r load annotation, fig.height = 2, fig.width = 3}
suppressMessages(library(ChIPpeakAnno))
suppressMessages(library(GenomicFeatures))
suppressMessages(library(ChIPseeker))
suppressMessages(library(dplyr))
suppressMessages(library(ggplot2))
suppressMessages(library(cowplot))
suppressMessages(library(ggsci))
suppressMessages(library(reshape2))
suppressMessages(library(tidyr))
source("./utils.R")

mm10_gene_annotation <- makeTxDbFromGFF("/Volumes/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/gencode.vM25.annotation.gtf")

#Annotate peak to gene features
annotate_peak <- annotatePeak(dCas9_HA_overlap, tssRegion = c(-1000, 500),
                                    TxDb = mm10_gene_annotation)
annotate_peak <- as.GRanges(annotate_peak)
annotate_peak$simpleAnnotation <- simplifyAnnotation(annotate_peak$annotation)

table(annotate_peak$simpleAnnotation)
#Intergenic Intragenic   Promoter 
#      2297       1272        471 

MT2Mm_overlap <- findOverlaps(annotate_peak, MT2Mm)
MT2CMm_overlap <- findOverlaps(annotate_peak, MT2CMm)
TE_overlap <- findOverlaps(annotate_peak, mm10_rmsk)

annotate_peak$Peaks2 <- "Non_TE"
annotate_peak$Peaks2[queryHits(TE_overlap)] <- "other_TE"
annotate_peak$Peaks2[queryHits(MT2CMm_overlap)] <- "MT2C_Mm"
annotate_peak$Peaks2[queryHits(MT2Mm_overlap)] <- "MT2_Mm"

table(annotate_peak$Peaks2)
#MT2_Mm  MT2C_Mm   Non_TE other_TE 
#    2485     1090      207      258 

#replace others with gene annotation
annotate_peak$Peaks2[which(annotate_peak$Peaks2 == "Non_TE")] <- 
  annotate_peak$simpleAnnotation[grep("Non_TE", annotate_peak$Peaks2)]
annotate_peak$Peaks2[which(annotate_peak$Peaks2 == "other_TE")] <-
  annotate_peak$simpleAnnotation[grep("other_TE", annotate_peak$Peaks2)]
table(annotate_peak$Peaks2)
#Intergenic Intragenic     MT2_Mm    MT2C_Mm   Promoter 
#       166        198       2485       1090        101

df4Pie <- as.data.frame(table(annotate_peak$Peaks2)) %>%
              mutate(perc = `Freq` / sum(`Freq`)) %>% 
              mutate(labels = scales::percent(perc))
ggplot(df4Pie, aes(x = "", y = perc, fill = Var1)) +
        geom_col(color = "black") +
        coord_polar(theta = "y") + 
        geom_text(aes(label = labels),
        position = position_stack(vjust = 0.5)) +
        guides(fill = guide_legend(title = NULL)) +
        scale_fill_npg() +
        #geom_label(aes(label = labels), 
        #           color = rep("black", nrow(df4cas9Pie)),
        #           position = position_stack(vjust = 0.1),
        #show.legend = FALSE) + 
        theme_void() +
        ggtitle("dCas9 ChIP")
pdf("./figures/231115_ChIPpeaks_distribution.pdf", height = 4, width = 4)
ggplot(df4Pie, aes(x = "", y = perc, fill = Var1)) +
        geom_col(color = "black") +
        coord_polar(theta = "y") + 
        geom_text(aes(label = labels),
        position = position_stack(vjust = 0.5)) +
        guides(fill = guide_legend(title = NULL)) +
        scale_fill_npg() +
        #geom_label(aes(label = labels), 
        #           color = rep("black", nrow(df4cas9Pie)),
        #           position = position_stack(vjust = 0.1),
        #show.legend = FALSE) + 
        theme_void() +
        ggtitle("dCas9 ChIP")
dev.off()
```
### 2.3.2 annotations of dCas9-MT2 peaks 
```{r MT2Mm/MT2CMm_peak distribution}
df4MT2_Pie <- as.data.frame(
              table(annotate_peak[which(annotate_peak$Peaks2 == "MT2C_Mm" |
                                        annotate_peak$Peaks2 == "MT2_Mm")]$simpleAnnotation)) %>%
              mutate(perc = `Freq` / sum(`Freq`)) %>% 
              mutate(labels = scales::percent(perc))

pdf("./figures/231115_MT2Mm_MT2C_Mm_peaks_distribution.pdf", height = 4, width = 4)
  ggplot(df4MT2_Pie, aes(x = "", y = perc, fill = Var1)) +
          geom_col(color = "black") +
          coord_polar(theta = "y") + 
          geom_text(aes(label = labels),
          position = position_stack(vjust = 0.5)) +
          guides(fill = guide_legend(title = NULL)) +
          scale_fill_npg() +
          theme_void() +
          ggtitle("MT2C_Mm/MT2_Mm ChIP peaks")
dev.off()
ggplot(df4MT2_Pie, aes(x = "", y = perc, fill = Var1)) +
          geom_col(color = "black") +
          coord_polar(theta = "y") + 
          geom_text(aes(label = labels),
          position = position_stack(vjust = 0.5)) +
          guides(fill = guide_legend(title = NULL)) +
          scale_fill_npg() +
          theme_void() +
          ggtitle("MT2C_Mm/MT2_Mm ChIP peaks")
```

### 2.3.3  off-target enrichment score
```{r check ChIPsignal detailed groups, fig.height = 2, fig.width = 3}
tmp1 <- data.frame(group = annotate_peak$Peaks2,
                  ChIP_signal = annotate_peak$FoldChange)

#perform two-sided wilcox test 
wilcox.test(tmp1[which(tmp1$group == "MT2C_Mm"), 2], 
            tmp1[which(tmp1$group != "MT2_Mm" & tmp1$group != "MT2C_Mm"), 2],
            alternative = "two.sided")$p.value

pdf("./figures/231115_ChIPpeaks_enrichment.pdf", height = 4, width = 4)
ggplot(tmp1, aes(x = group, y = ChIP_signal)) + 
  #geom_violin(width = 1.2, fill = "grey") +
  geom_boxplot(color = "black", alpha = 0.2, outlier.color = NA) + 
  ylim(0, 900) + theme_cowplot(16) +
  #geom_hline(yintercept= 50, linetype="dashed", color = "red") +
  scale_x_discrete(limits = c("MT2_Mm","MT2C_Mm", "Promoter", "Intragenic", "Intergenic")) +
  #                  labels = c("MT2_Mm\nMT2C_Mm peaks\n(3632)", "Others (47121)")) +
  theme(axis.title.x = element_blank()) +
  ylab("ChIP enrichment") +
  ggtitle("dCas9_ChIP")
dev.off()
ggplot(tmp1, aes(x = group, y = ChIP_signal)) + 
  #geom_violin(width = 1.2, fill = "grey") +
  geom_boxplot(color = "black", alpha = 0.2, outlier.color = NA) + 
  ylim(0, 900) + theme_cowplot(16) +
  #geom_hline(yintercept= 50, linetype="dashed", color = "red") +
  scale_x_discrete(limits = c("MT2_Mm","MT2C_Mm", "Promoter", "Intragenic", "Intergenic")) +
  #                  labels = c("MT2_Mm\nMT2C_Mm peaks\n(3632)", "Others (47121)")) +
  theme(axis.title.x = element_blank()) +
  ylab("ChIP enrichment") +
  ggtitle("dCas9_ChIP")
```
### 2.3.4 bound vs unbound TE subfamlies
Priority list
MT2_Mm -> MT2C_Mm -> MERVL-int -> MT2B2 -> MT2B -> MT2B1 -> MTA_Mm

```{r stack plot}
MT2Mm <- mm10_rmsk[grep("MT2_Mm:ERVL", mm10_rmsk$name),]
MT2CMm <- mm10_rmsk[grep("MT2C_Mm:ERVL", mm10_rmsk$name),]
#MERVL_int <- mm10_rmsk[grep("MERVL-int:ERVL", mm10_rmsk$name),]
MT2B2 <- mm10_rmsk[grep("MT2B2:ERVL", mm10_rmsk$name),]
MT2B1 <- mm10_rmsk[grep("MT2B1:ERVL", mm10_rmsk$name),]
MT2B <- mm10_rmsk[grep("MT2B:ERVL", mm10_rmsk$name),]
MTA_Mm <- mm10_rmsk[grep("MTA_Mm:ERVL", mm10_rmsk$name),]

length(MT2Mm) #2667
length(MT2CMm) #1982
length(MT2B2) #3655
length(MT2B1) #7248
length(MT2B) #13977
length(MTA_Mm) #15623

df4boundCas9 <- data.frame(
  repeats = c("MT2_Mm", "MT2C_Mm",
              "MT2B2", "MT2B1", "MT2B", "MTA_Mm"),
  bound = c(
      length(findOverlaps(annotate_peak, MT2Mm)),
      length(findOverlaps(annotate_peak, MT2CMm)),
      length(findOverlaps(annotate_peak, MT2B2)),
      length(findOverlaps(annotate_peak, MT2B1)),
      length(findOverlaps(annotate_peak, MT2B)),
      length(findOverlaps(annotate_peak, MTA_Mm))),
  total = c(
      length(MT2Mm), #2612
      length(MT2CMm), #1941
      length(MT2B2), #3649
      length(MT2B1), #6955
      length(MT2B), #13806
      length(MTA_Mm)
  ))
df4boundCas9$unbound <- df4boundCas9$total - df4boundCas9$bound

df4boundCas9 <- gather(df4boundCas9, key = "group", value = "count", -repeats, -total)
df4boundCas9$pct <- df4boundCas9$count / df4boundCas9$total *100

pdf("./figures/231115_fractions_bound_by_dCas9.pdf", width = 6, height = 6)
ggplot(df4boundCas9, aes(x = repeats, y = pct, 
                           fill = factor(group, levels = c("unbound", "bound")))) + 
  geom_bar(stat = "identity", position = "fill", color = "black") + 
  scale_fill_manual(values = c("white", "black")) +
  scale_x_discrete(limits = c("MT2_Mm","MT2C_Mm", "MT2B2", "MT2B1", "MT2B", "MTA_Mm")) +
  theme_cowplot(16) +
  ggtitle("dCas9 ChIP") + 
  theme(axis.title.x = element_blank(),
        legend.title=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 0.9, hjust=0.9)) +
  ylab("Fraction bound")
dev.off()
ggplot(df4boundCas9, aes(x = repeats, y = pct, 
                           fill = factor(group, levels = c("unbound", "bound")))) + 
  geom_bar(stat = "identity", position = "fill", color = "black") + 
  scale_fill_manual(values = c("white", "black")) +
  scale_x_discrete(limits = c("MT2_Mm","MT2C_Mm", "MT2B2", "MT2B1", "MT2B", "MTA_Mm")) +
  theme_cowplot(16) +
  ggtitle("dCas9 ChIP") + 
  theme(axis.title.x = element_blank(),
        legend.title=element_blank(),
        axis.text.x = element_text(angle = 30, vjust = 0.9, hjust=0.9)) +
  ylab("Fraction bound")
```

## 2.4 write output as table
```{r}
write.table(
  data.frame(seqnames = seqnames(annotate_peak),
             starts = start(annotate_peak) -1 ,
             ends = end(annotate_peak),
             names = annotate_peak$name,
             scores = annotate_peak$score,
             strands = strand(annotate_peak),
             signal = annotate_peak$FoldChange,
             pVal = annotate_peak$pVal,
             qVal = annotate_peak$qVal,
             annotation = annotate_peak$annotation,
             geneId = annotate_peak$geneId,
             distanceToTSS = annotate_peak$distanceToTSS,
             simpleAnnotation = annotate_peak$simpleAnnotation,
             TE_peaks = annotate_peak$Peaks2
             ),
  "./R_output/dCas9_ChIP/231115_dCas9_peaks_annotated.txt", quote = F, 
  sep = "\t", row.names = F, col.names = F
)

#write out bed file for only MT2_Mm & MT2C_Mm peaks
annotate_peak_MT2only <- annotate_peak[which(annotate_peak$Peaks2 == "MT2C_Mm" |
                                        annotate_peak$Peaks2 == "MT2_Mm")]
annotate_peak_MT2Mm <- annotate_peak[which(annotate_peak$Peaks2 == "MT2_Mm")]
annotate_peak_MT2CMm <- annotate_peak[which(annotate_peak$Peaks2 == "MT2C_Mm")]
annotate_peak_offtarget <- annotate_peak[which(annotate_peak$Peaks2 != "MT2C_Mm" &
                                               annotate_peak$Peaks2 != "MT2_Mm")]

length(annotate_peak) #4040
length(annotate_peak_MT2only) #3575
length(annotate_peak_MT2Mm) #2485
length(annotate_peak_MT2CMm) #1090
length(annotate_peak_offtarget) #465

write.table(
  data.frame(seqnames = seqnames(annotate_peak_MT2only),
             starts = start(annotate_peak_MT2only) -1 ,
             ends = end(annotate_peak_MT2only),
             names = annotate_peak_MT2only$Peaks2,
             scores = annotate_peak_MT2only$score,
             strands = strand(annotate_peak_MT2only)
             ),
  "./R_output/dCas9_ChIP/231115_dCas9_MT2Mm_MT2CMm.bed", quote = F, 
  sep = "\t", row.names = F, col.names = F
)

write.table(
  data.frame(seqnames = seqnames(annotate_peak_MT2Mm),
             starts = start(annotate_peak_MT2Mm) -1 ,
             ends = end(annotate_peak_MT2Mm),
             names = annotate_peak_MT2Mm$Peaks2,
             scores = annotate_peak_MT2Mm$score,
             strands = strand(annotate_peak_MT2Mm)
             ),
  "./R_output/dCas9_ChIP/231115_dCas9_MT2Mm.bed", quote = F, 
  sep = "\t", row.names = F, col.names = F
)
write.table(
  data.frame(seqnames = seqnames(annotate_peak_MT2CMm),
             starts = start(annotate_peak_MT2CMm) -1 ,
             ends = end(annotate_peak_MT2CMm),
             names = annotate_peak_MT2CMm$Peaks2,
             scores = annotate_peak_MT2CMm$score,
             strands = strand(annotate_peak_MT2CMm)
             ),
  "./R_output/dCas9_ChIP/231115_dCas9_MT2CMm.bed", quote = F, 
  sep = "\t", row.names = F, col.names = F
)
write.table(
  data.frame(seqnames = seqnames(annotate_peak_offtarget),
             starts = start(annotate_peak_offtarget) -1 ,
             ends = end(annotate_peak_offtarget),
             names = annotate_peak_offtarget$Peaks2,
             scores = annotate_peak_offtarget$score,
             strands = strand(annotate_peak_offtarget)
             ),
  "./R_output/dCas9_ChIP/231115_dCas9_offtarget.bed", quote = F, 
  sep = "\t", row.names = F, col.names = F
)
```

## 2.5 dCas9/HA DeepTool heatmap
```{bash add strand to bed}
#The above bed file has no strand info for MT2_Mm & MT2C_Mm
#add strand using following command lines
module load bedtools
grep MT2_Mm /data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed > mm10_MT2Mm_MT2CMm.bed
grep MT2C_Mm /data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed >> mm10_MT2Mm_MT2CMm.bed
bedtools intersect -a mm10_MT2Mm_MT2CMm.bed -b 231115_dCas9_MT2Mm_MT2CMm.bed -wa > 231115_dCas9_MT2Mm_MT2CMm.bed
```

```{bash deeptool enrichment}
#!/bin/bash
#BSUB -W 1:00
#BSUB -n 2
#BSUB -M 10000
#BSUB -e %J.err
#BSUB -o %J.out

module load bedtools/2.30.0
source ~/.bash_profile
conda activate deeptools

te1="MT2_Mm"
te2="MT2C_Mm"

te_gtf="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.gtf"

grep ${te1} $te_gtf > ${te1}.gtf
grep ${te2} $te_gtf > ${te2}.gtf

dcas9_uniq_bw="./bigwigs/MT2Mmi_ESCs_Cas9_ChIP_rep1_mm10.sorted.Q30.dedup.bw"
dcas9_multi_bw="./bigwigs/MT2Mmi_ESCs_Cas9_ChIP_rep1_mm10.sorted.multi.dedup.bw"
HA_uniq_bw="./bigwigs/MT2Mmi_ESCs_HA_ChIP_rep1_mm10.sorted.Q30.dedup.bw"
HA_multi_bw="./bigwigs/MT2Mmi_ESCs_HA_ChIP_rep1_mm10.sorted.multi.dedup.bw"

#computeMatrix--------------------
computeMatrix scale-regions \
		-S $dcas9_uniq_bw $dcas9_multi_bw $HA_uniq_bw $HA_multi_bw \
		-R ${te1}.gtf ${te2}.gtf \
		-a 2000 -b 2000 \
		--outFileName ${te1}.${te2}.2000flank_mat.gz \
		--sortRegions 'descend' \
		--samplesLabel 'dCas9 (uniq)' 'dcas9 (multi)' 'HA (uniq)' 'HA (multi)' \
		--transcriptID 'exon' \
		--missingDataAsZero

plotHeatmap --matrixFile ${te1}.${te2}.2000flank_mat.gz \
		    --outFileName 230716_${te1}.${te2}_2000flank.pdf \
		    --dpi 300 \
	    	    --sortRegions "descend" \
	  	    --colorMap "Blues" \
		    --outFileSortedRegions ${te1}.${te2}_1000flank.bed \
		    --whatToShow "heatmap and colorbar" \
		    --boxAroundHeatmaps no \
		    --samplesLabel 'dCas9 (uniq)' 'dcas9 (multi)' 'HA (uniq)' 'HA (multi)' \
	            --legendLocation "lower-center"
```

# 3. MT2 dynamic expression

## 3.1 classify FL and solo ERV

### 3.1.1 identify FL MERVL

Strategy
step1 : retrieve MERVL or MERVL_2A-int 5 kb\ 
step2 : find nearest MT2 of MERVL or MERVL_2A-int\ 
step3 : identify the nearest upstream and downstream of sense MT2\  
step4 : require the distance of these two MT2_Mm insertions < 10kb\ 

```{perl}
#!/usr/bin/perl

use strict;

my @LTR = ("MT2_Mm", "MT2C_Mm");
my @INSERT = ("MERVL-int", "MERVL_2A-int");
my $insert_length = 5000; #5kb
my $full_length = 10000; #10kb

my $TE_BED = "/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed";
my $chr_Length = "/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/chrNameLength.txt";
system("sort -k1,1 -k2,2n $chr_Length > ${chr_Length}_sorted.txt");

#step1 : retrieve ERVL or MERVL_2A-int > 5kb
foreach(@INSERT){
        my $tmp_insert = $_;
        system("grep $tmp_insert $TE_BED > mm10_rmsk_${tmp_insert}.bed");
        open (BED, "mm10_rmsk_${tmp_insert}.bed") or die$!;
        open (OUT, ">mm10_rmsk_${tmp_insert}_${insert_length}bp.bed") or die$!;
        while (my $line = <BED>){
                chomp $line;
                my @split = split(/\s+/, $line);
                if($split[2] - $split[1] >= $insert_length){
                        print OUT "$line\n";
                }
        }
        close BED;
        close OUT;
}

#combine MERVL and MERVL_2A-int
foreach(@INSERT){
        my $tmp_insert = $_;
        if (-e "mm10_rmsk_MERVL.bed"){
                system("cat mm10_rmsk_${tmp_insert}_${insert_length}bp.bed >> mm10_rmsk_MERVL.bed");    
        } else{
                system("cat mm10_rmsk_${tmp_insert}_${insert_length}bp.bed > mm10_rmsk_MERVL.bed")
        }
}
system("sort -k1,1 -k2,2n mm10_rmsk_MERVL.bed > mm10_rmsk_MERVL_sorted.bed");

#step2 : find nearest MT2_Mm of MERVL or MERVL_2A-int 
foreach(@LTR){
        my $tmp_insert = $_;
        if(-e "mm10_rmsk_MT2.bed"){
                system("grep $tmp_insert $TE_BED >> mm10_rmsk_MT2.bed");
        } else{
                system("grep $tmp_insert $TE_BED > mm10_rmsk_MT2.bed");
        }               
}
system("sort -k1,1 -k2,2n mm10_rmsk_MT2.bed > mm10_rmsk_MT2_sorted.bed");

#only downstream LTR with the same strand
system("bedtools closest -s -iu -D a -g ${chr_Length}_sorted.txt -a mm10_rmsk_MERVL_sorted.bed -b mm10_rmsk_MT2_sorted.bed > MERVL_MT2_bedclosest.txt");

#only upstream LTR with the same strand
system("bedtools closest -s -id -D a -g ${chr_Length}_sorted.txt -a mm10_rmsk_MERVL_sorted.bed -b mm10_rmsk_MT2_sorted.bed > MT2_MERVL_bedclosest.txt");        

#step3: require the distance of these two MT2s <10kb 
system("Rscript filter_full_length_MERVL.R");

#clean out intermediate files
system("mv full_length_MERVL* output/");
#system("rm *.bed");
#system("rm *.txt");
```


```{r filter_full_length_MERVL.R}
#!/usr/bin/Rscript

#setwd("/Volumes/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/")

#full length MERVL with MT2 downstream
MERVL_MT2 <- read.table("MERVL_MT2_bedclosest.txt", sep = "\t")
colnames(MERVL_MT2) <- c("MERVL_chr", "MERVL_start", "MERVL_end",
                         "MERVL", "MERVL_score", "MERVL_strand",
                         "downstream_MT2_chr", "downstream_MT2_start", 
                         "downstream_MT2_end",
                         "downstream_MT2", "downstream_MT2_score", 
                         "downstream_MT2_strand", 
                         "MERVL_MT2_distance")

#full length MERVL with MT2 upstream
MT2_MERVL <- read.table("MT2_MERVL_bedclosest.txt", sep = "\t")
colnames(MT2_MERVL) <- c("MERVL_chr", "MERVL_start", "MERVL_end",
                         "MERVL", "MERVL_score", "MERVL_strand",
                         "upstream_MT2_chr", "upstream_MT2_start", 
                         "upstream_MT2_end",
                         "upstream_MT2", "upstream_MT2_score", 
                         "upstream_MT2_strand", 
                         "MT2_MERVL_distance")

MERVL <- merge(MT2_MERVL, MERVL_MT2[, c("MERVL",
                                        "downstream_MT2_chr", "downstream_MT2_start", 
                                        "downstream_MT2_end",
                                        "downstream_MT2", "downstream_MT2_score", 
                                        "downstream_MT2_strand", 
                                        "MERVL_MT2_distance")],
               by = "MERVL")


MERVL$length <- c()
for(i in 1:nrow(MERVL)){
  if(MERVL$MERVL_strand[i] == "+"){
    MERVL$length[i] <- MERVL$downstream_MT2_end[i] - MERVL$upstream_MT2_start[i]
  }else{
    MERVL$length[i] <- MERVL$upstream_MT2_end[i] - MERVL$downstream_MT2_start[i]
  }
}

full_length_MERVL <- MERVL[which(MERVL$length < 10000 & MERVL$length >0 ),]
#hist(full_length_MERVL$length)
full_length_MERVL <- full_length_MERVL[!duplicated(full_length_MERVL$MERVL),]
hist(full_length_MERVL$MERVL_end-full_length_MERVL$MERVL_start)
write.table(full_length_MERVL[, c("MERVL_chr", "MERVL_start", "MERVL_end", "MERVL",
                                "MERVL_score", "MERVL_strand")],
            "full_length_MERVL.bed", sep = "\t", col.names = F, quote = F, row.names =F)
            #"nonFL_MERVL.bed", sep = "\t", col.names = F, quote = F, row.names =F)
write.table(full_length_MERVL[, c("upstream_MT2_chr", "upstream_MT2_start",
                                  "upstream_MT2_end", "upstream_MT2", "upstream_MT2_score",
                                  "upstream_MT2_strand")],
            "full_length_MERVL_upMT2.bed", sep = "\t", col.names = F, quote = F, row.names = F)
            #"nonFL_MERVL_upMT2.bed", sep = "\t", col.names = F, quote = F, row.names = F)
write.table(full_length_MERVL[, c("downstream_MT2_chr", "downstream_MT2_start",
                                  "downstream_MT2_end", "downstream_MT2", "downstream_MT2_score",
                                  "downstream_MT2_strand")],
            "full_length_MERVL_downMT2.bed", sep = "\t", col.names = F, quote = F, row.names = F)
            #"nonFL_MERVL_downMT2.bed", sep = "\t", col.names = F, quote = F, row.names = F)
```

### 3.1.2 identify solo MT2
```{perl}
#!/usr/bin/perl
my @LTR = ("MT2_Mm", "MT2C_Mm");
my @INSERT = ("MERVL-int", "MERVL_2A-int");
my $insert_length = 0; #no cutoff here filter later
my $full_length = 10000; #10kb

my $TE_BED = "/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed";
my $chr_Length = "/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/chrNameLength.txt";
system("sort -k1,1 -k2,2n $chr_Length > ${chr_Length}_sorted.txt");

#step 1: retrieve MT2
foreach(@LTR){
        my $tmp_insert = $_;
        if(-e "mm10_rmsk_MT2.bed"){
                system("grep $tmp_insert $TE_BED >> mm10_rmsk_MT2.bed");
        } else{
                system("grep $tmp_insert $TE_BED > mm10_rmsk_MT2.bed");
        }
}
system("sort -k1,1 -k2,2n mm10_rmsk_MT2.bed > mm10_rmsk_MT2_sorted.bed");

#step 2: remove those that belong to full length MERVL
system("bedtools intersect -a mm10_rmsk_MT2_sorted.bed -b ./output/full_length_MERVL_downMT2.bed -v > mm10_rmsk_tmp_MT2.bed");
system("bedtools intersect -a mm10_rmsk_tmp_MT2.bed -b ./output/full_length_MERVL_upMT2.bed -v > mm10_rmsk_nonFL_MT2.bed");
system("rm mm10_rmsk_tmp_MT2.bed");
system("sort -k1,1 -k2,2n  mm10_rmsk_nonFL_MT2.bed >  mm10_rmsk_nonFL_MT2_sorted.bed");

#step3 : retrieve nearest MERVL or MERVL2A-int
foreach(@INSERT){
        my $tmp_insert = $_;
        system("grep $tmp_insert $TE_BED > mm10_rmsk_${tmp_insert}.bed");
        open (BED, "mm10_rmsk_${tmp_insert}.bed") or die$!;
        open (OUT, ">mm10_rmsk_${tmp_insert}_${insert_length}bp.bed") or die$!;
        while (my $line = <BED>){
                chomp $line;
                my @split = split(/\s+/, $line);
                if($split[2] - $split[1] >= $insert_length){
                        print OUT "$line\n";
                }
        }
        close BED;
        close OUT;
}

#combine MERVL and MERVL_2A-int
foreach(@INSERT){
        my $tmp_insert = $_;
        if (-e "mm10_rmsk_MERVL.bed"){
                system("cat mm10_rmsk_${tmp_insert}_${insert_length}bp.bed >> mm10_rmsk_MERVL.bed");    
        } else{
                system("cat mm10_rmsk_${tmp_insert}_${insert_length}bp.bed > mm10_rmsk_MERVL.bed")
        }
}
system("sort -k1,1 -k2,2n mm10_rmsk_MERVL.bed > mm10_rmsk_MERVL_sorted.bed");

#only downstream LTR with the same strand
system("bedtools closest -s -iu -D a -g ${chr_Length}_sorted.txt -a mm10_rmsk_nonFL_MT2_sorted.bed -b mm10_rmsk_MERVL_sorted.bed > MERVL_MT2_bedclosest.txt");

#only upstream LTR with the same strand
system("bedtools closest -s -id -D a -g ${chr_Length}_sorted.txt -a mm10_rmsk_nonFL_MT2_sorted.bed -b mm10_rmsk_MERVL_sorted.bed > MT2_MERVL_bedclosest.txt");  

#step4: remove MT2s with nearby MERVL/MER2A-int
system("Rscript filter_soloMT2.R");
system("mv soloMT2.bed output/");
system("mv othersMT2.bed output/");
system("rm *.bed");
system("rm *.txt");
exit;
```

```{r filter_soloMT2.R}
#!/usr/bin/Rscript

#setwd("/Volumes/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/")

#full length MERVL with MT2 downstream
MERVL_MT2 <- read.table("MERVL_MT2_bedclosest.txt", sep = "\t")
colnames(MERVL_MT2) <- c("MT2_chr", "MT2_start", "MT2_end",
                         "MT2", "MT2_score", "MT2_strand",
                         "downstream_MERVL_chr", "downstream_MERVL_start",
                         "downstream_MERVL_end",
                         "downstream_MERVL", "downstream_MERVL_score",
                         "downstream_MERVL_strand",
                         "MERVL_MT2_distance")

#full length MERVL with MT2 upstream
MT2_MERVL <- read.table("MT2_MERVL_bedclosest.txt", sep = "\t")
colnames(MT2_MERVL) <- c("MT2_chr", "MT2_start", "MT2_end",
                         "MT2", "MT2_score", "MT2_strand",
                         "upstream_MERVL_chr", "upstream_MERVL_start",
                         "upstream_MERVL_end",
                         "upstream_MERVL", "upstream_MERVL_score",
                         "upstream_MERVL_strand",
                         "MT2_MERVL_distance")

MT2 <- merge(MT2_MERVL, MERVL_MT2[, c("MT2",
                                        "downstream_MERVL_chr", "downstream_MERVL_start",
                                        "downstream_MERVL_end",
                                        "downstream_MERVL", "downstream_MERVL_score",
                                        "downstream_MERVL_strand",
                                        "MERVL_MT2_distance")],
               by = "MT2")


MT2$group <- c()
cutoff = 5000
for(i in 1:nrow(MT2)){
  if(MT2$upstream_MERVL_chr[i] == "." & MT2$downstream_MERVL_chr[i] == "."){
    MT2$group[i] <- "solo"
  } else if (MT2$upstream_MERVL_chr[i] == "." & abs(MT2$MERVL_MT2_distance[i]) > cutoff){
    MT2$group[i] <- "solo"
  } else if (MT2$downstream_MERVL_chr[i] == "." & abs(MT2$MT2_MERVL_distance[i]) > cutoff){
    MT2$group[i] <- "solo"
  } else if (abs(MT2$MERVL_MT2_distance[i]) > cutoff & abs(MT2$MT2_MERVL_distance[i]) > cutoff){
    MT2$group[i] <- "solo"
  }
  else{
    MT2$group[i] <- "others"
  }
}

write.table(MT2[which(MT2$group == "solo"), c("MT2_chr", "MT2_start", "MT2_end", "MT2",
                                  "MT2_score", "MT2_strand")],
            "soloMT2.bed", sep = "\t", col.names = F, quote = F, row.names =F)
write.table(MT2[which(MT2$group == "others"), c("MT2_chr", "MT2_start",
                                  "MT2_end", "MT2", "MT2_score",
                                  "MT2_strand")],
            "othersMT2.bed", sep = "\t", col.names = F, quote = F, row.names = F)
```

```{bash}
grep -c MT2_Mm ./R_input/FL_solo_MT2/*.bed
<!-- FL_solo_MT2/full_length_MERVL.bed:0 -->
<!-- FL_solo_MT2/full_length_MERVL_downMT2.bed:591 -->
<!-- FL_solo_MT2/full_length_MERVL_upMT2.bed:591 -->
<!-- FL_solo_MT2/othersMT2.bed:423 -->
<!-- FL_solo_MT2/soloMT2.bed:1062 -->

grep -c MT2C_Mm FL_solo_MT2/*.bed
<!-- FL_solo_MT2/full_length_MERVL.bed:0 -->
<!-- FL_solo_MT2/full_length_MERVL_downMT2.bed:8 -->
<!-- FL_solo_MT2/full_length_MERVL_upMT2.bed:8 -->
<!-- FL_solo_MT2/othersMT2.bed:251 -->
<!-- FL_solo_MT2/soloMT2.bed:1715 -->
```

### 3.1.3 GenMap mappbility
```{bash}
genmap index -F /mm10.fa -I mm10_genmap/
genmap map -I mm10_genmap/ -O mm10_k50_m3_mappability.bg -bg -T 6 -K 50 -E 3
grep  '^chr' mm10_k50_m3_mappability.bg.bedgraph > mm10_k50_m3_mappability.bg.chr.bedgraph
bedGraphToBigWig mm10_k50_m3_mappability_sorted.bg mm10_chr_size.txt mm10_k50_m3_mappability.bw
```

### 3.1.4 RNA dynamic deeptool heatmap
```{bash}
#!/bin/bash
#BSUB -W 1:00
#BSUB -n 2
#BSUB -M 10000
#BSUB -e %J.err
#BSUB -o %J.out

module load bedtools/2.30.0
source ~/.bash_profile
conda activate deeptools

#mappbility-------------------------------------
mappbility_dir="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/genmap_mappbility"
kmer50="${mappbility_dir}/mm10_kmer50_mis3_mappibility.bw"

#Region of interest------------------------------
Region_dir="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/output"
FL_MERVL="${Region_dir}/full_length_MERVL.bed"
FL_MERVL_3primeMT2="${Region_dir}/full_length_MERVL_downMT2.bed"
FL_MERVL_5primeMT2="${Region_dir}/full_length_MERVL_upMT2.bed"
soloMT2="${Region_dir}/soloMT2.bed"
otherMT2="${Region_dir}/othersMT2.bed"

grep "MT2_Mm" $FL_MERVL_3primeMT2 > FL_MERVL_3primeMT2_Mm.bed
grep "MT2_Mm" $FL_MERVL_5primeMT2 > FL_MERVL_5primeMT2_Mm.bed
grep "MT2_Mm" $soloMT2 > soloMT2_Mm.bed
grep "MT2_Mm" $otherMT2 > otherMT2_Mm.bed

#total RNAseq------------------------------------
totalRNA_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231019_remap_Chunxia_totalRNAseq/bigwigs"
total_MII="${totalRNA_dir}/MII_merged.sorted.bw"
total_1c="${totalRNA_dir}/PN5_merged.sorted.bw"
total_2c="${totalRNA_dir}/c2_merged.sorted.bw"
total_4c="${totalRNA_dir}/c4_merged.sorted.bw"
total_morula="${totalRNA_dir}/morula_merged.sorted.bw"
total_BL="${totalRNA_dir}/BL_merged.sorted.bw"

totalRNA_dir2="/data/ZYChenlab/Zhiyuan/projects/TE_project/230815_remap_totalRNA_early2C/bigwigs"
total_e2c="${totalRNA_dir2}/e2C_total_rep1Aligned.out.sorted_uniq.bw"

#polyA RNAseq------------------------------------
polyRNA_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231020_remap_Zhang_polyARNAseq/bigwigs"
polyA_MII="${polyRNA_dir}/MII_merged.sorted.bw"
polyA_PN5="${polyRNA_dir}/PN5_merged.sorted.bw"
polyA_e2c="${polyRNA_dir}/e2c_merged.sorted.bw"
polyA_c2="${polyRNA_dir}/c2_merged.sorted.bw"
polyA_c4="${polyRNA_dir}/c4_merged.sorted.bw"
polyA_c8="${polyRNA_dir}/c8_merged.sorted.bw"
polyA_ICM="${polyRNA_dir}/ICM_merged.sorted.bw"

#computeMatrix--------------------
computeMatrix scale-regions \
		-S $total_1c $total_e2c $total_2c $total_4c $total_morula $total_BL \
		-R FL_MERVL_5primeMT2_Mm.bed FL_MERVL_3primeMT2_Mm.bed soloMT2_Mm.bed otherMT2_Mm.bed \
		-a 1000 -b 1000 \
		--regionBodyLength 500 \
		--outFileName MT2_Mm.1000flank_mat.gz \
		--sortRegions 'descend' \
		--samplesLabel 'total_1c' 'total_e2c' 'total_2C' 'total_4c' 'total_morula' 'total_BL' \
		--outFileSortedRegions MT2_Mm.1000flank_mat_sorted.bed \
		--outFileNameMatrix MT2_Mm.1000flank_mat_sorted.matrix \
		--missingDataAsZero

#plotHeatmap---------------------
plotHeatmap --matrixFile MT2_Mm.1000flank_mat.gz \
		    --outFileName MT2_Mm.1000flank.pdf \
		    --dpi 300 \
	    	    --sortRegions "descend" \
	  	    --colorMap 'GnBu' 'GnBu' 'GnBu' 'GnBu' 'GnBu' 'GnBu' 
 		    --whatToShow "heatmap and colorbar" \
		    --boxAroundHeatmaps no \
		    --samplesLabel  'total_1c' 'total_e2c' 'total_2C' 'total_4c' 'total_morula' 'total_BL' ' \
	            --legendLocation "lower-center" \
		    --zMax  30 30 30 30 30 30  	
```

Replace MT2_Mm with MT2C_Mm to generate a similar heatmap

### 3.1.5 MT2C_Mm bound vs unbound RNA

The Reviewer #1 asked about what's the differences of MT2C_Mm copies that
are bound by dCas9 vs. not bound by dCas9
```{bash}
#!/bin/bash
#BSUB -W 1:00
#BSUB -n 2
#BSUB -M 10000
#BSUB -e %J.err
#BSUB -o %J.out

module load bedtools/2.30.0
source ~/.bash_profile
conda activate deeptools

#mappbility-------------------------------------
mappbility_dir="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/genmap_mappbility"
kmer50="${mappbility_dir}/mm10_kmer50_mis3_mappibility.bw"

#Region of interest------------------------------
Region_dir="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/output"
FL_MERVL="${Region_dir}/full_length_MERVL.bed"
FL_MERVL_3primeMT2="${Region_dir}/full_length_MERVL_downMT2.bed"
FL_MERVL_5primeMT2="${Region_dir}/full_length_MERVL_upMT2.bed"
soloMT2="${Region_dir}/soloMT2.bed"
otherMT2="${Region_dir}/othersMT2.bed"

grep "MT2C_Mm" $FL_MERVL_3primeMT2 > FL_MERVL_3primeMT2C_Mm.bed
grep "MT2C_Mm" $FL_MERVL_5primeMT2 > FL_MERVL_5primeMT2C_Mm.bed
grep "MT2C_Mm" $soloMT2 > soloMT2C_Mm.bed
grep "MT2C_Mm" $otherMT2 > otherMT2C_Mm.bed

#compare bound MT2C_Mm vs non-bound MT2C_Mm
cat FL_MERVL_3primeMT2C_Mm.bed FL_MERVL_5primeMT2C_Mm.bed soloMT2C_Mm.bed  otherMT2C_Mm.bed > MT2C_Mm.bed
cp /data/ZYChenlab/Zhiyuan/projects/TE_project/230829_MT2_Mm_Integrative_analyses/R_output/230817_dCas9_MT2CMm.bed .
bedtools intersect -a MT2C_Mm.bed -b 230817_dCas9_MT2CMm.bed -v > MT2C_Mm_nonbound.bed

#total RNAseq------------------------------------
totalRNA_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231019_remap_Chunxia_totalRNAseq/bigwigs"
total_2c="${totalRNA_dir}/c2_merged.sorted.bw"

#polyA RNAseq------------------------------------
polyRNA_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231020_remap_Zhang_polyARNAseq/bigwigs"
polyA_c2="${polyRNA_dir}/c2_merged.sorted.bw"

#ChIPseq-------------------------------------
ChIPseq_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/230810_MT2_Mmi_ESCs_Cas9ChIPCUTRUN/bigwigs"
dCas9_multi="${ChIPseq_dir}/MT2Mmi_ESCs_Cas9_ChIP_rep2_mm10.sorted.multi.dedup.bw"
HA_multi="${ChIPseq_dir}/MT2Mmi_ESCs_HA_ChIP_rep2_mm10.sorted.multi.dedup.bw"

#computeMatrix--------------------
computeMatrix scale-regions \
                -S $total_2c $polyA_c2 $dCas9_multi $HA_multi \
                -R 230817_dCas9_MT2CMm.bed MT2C_Mm_nonbound.bed \
                -a 1000 -b 1000 \
                --regionBodyLength 500 \
                --outFileName MT2C_Mm.1000flank_mat_RNAseqChIPseq.gz \
                --sortRegions 'descend' \
                --samplesLabel 'total_2C' 'polyA_c2' 'dCas9_multi' 'HA_multi' \
                --outFileSortedRegions MT2C_Mm.1000flank_mat_RNAseqChIPseq_sorted.bed \
                --outFileNameMatrix MT2C_Mm.1000flank_mat_RNAseqChIPseq_sorted.matrix \
                --sortUsingSamples 3 \
                --missingDataAsZero
```

```{r, fig.height = 2, fig.width = 7}
suppressMessages(library(ggplot2))
suppressMessages(library(cowplot))

#MT2CMm------------------------------------
MT2CMm <- read.table("./R_input/dCas9_ChIPseq/MT2C_Mm.1000flank_mat_RNAseqChIPseq_sorted.matrix", skip = 3, 
                 header = T)
MT2CMm_bound <- MT2CMm[c(1:1090),]
MT2CMm_unbound <- MT2CMm[c(1091:nrow(MT2CMm)),]

MT2CMm_bound_mean <- as.data.frame(colMeans(MT2CMm_bound))
MT2CMm_unbound_mean <- as.data.frame(colMeans(MT2CMm_unbound))

colnames(MT2CMm_bound_mean) <- "average"
colnames(MT2CMm_unbound_mean) <- "average"

MT2CMm_bound_mean$group <- rep("bound", nrow(MT2CMm_bound_mean))
MT2CMm_unbound_mean$group <- rep("unbound", nrow(MT2CMm_unbound_mean))

MT2CMm_recombine <- rbind(MT2CMm_bound_mean, MT2CMm_unbound_mean)

MT2CMm_recombine$data <- c(rep("total", 250), rep("polyA", 250),
                           rep("dCas9", 250), rep("HA", 250),
                           rep("total", 250), rep("polyA", 250),
                           rep("dCas9", 250), rep("HA", 250))
MT2CMm_recombine$bp <- c(seq(1:250), seq(1:250),
                         seq(1:250), seq(1:250),
                         seq(1:250), seq(1:250),
                         seq(1:250), seq(1:250))
MT2CMm_RNA <- MT2CMm_recombine[which(MT2CMm_recombine$data == "total"),]
MT2CMm_dCas9 <- MT2CMm_recombine[which(MT2CMm_recombine$data == "dCas9"),]

pdf("./figures/231114_metaplot_boundVSunbound.pdf", width = 6, height = 4)
fig1 <- ggplot(MT2CMm_RNA, aes(x = bp, y = average, color = group)) + 
  geom_line() + ylim(c(0,12)) + 
  scale_color_manual(values = c("black", "grey")) +
  theme_cowplot(16) + ggtitle("MT2CMm_boundvsUnbound_RNA")
fig1
fig2 <- ggplot(MT2CMm_dCas9, aes(x = bp, y = average, color = group)) + 
  geom_line() + ylim(c(0,1200)) + 
  scale_color_manual(values = c("black", "grey")) +
  theme_cowplot(16) + ggtitle("MT2CMm_boundvsUnbound_dCas9")
fig2
dev.off()
plot_grid(fig1, fig2, labels= c('A', 'B'))
```

# 4. MT2i total RNA-seq repeats analyses

## 4.1 QC
### 4.1.1 Input datasets
  
Sample           | # of embryos |     Group        
-----------------|--------------|------------------
CTRi_e2c_1       | 6            | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~20hpi)
CTRi_e2c_2       | 7            | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~20hpi)
MT2_Mmi_e2c_1    | 8            | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~28hpi)
MT2_Mmi_e2c_2    | 5            | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~28hpi)
                 |              |
noninj_2c_1      | 10           | Non-injected 2-cell embryos (~28hpi)
noninj_2c_2      | 10           | Non-injected 2-cell embryos (~28hpi)
CTRi_2c_1        | 10           | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~28hpi)
CTRi_2c_2        | 10           | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~28hpi)
CTRi_2C_3        | 10           | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~28hpi)
MT2_Mmi_2c_1     | 10           | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~28hpi)
MT2_Mmi_2c_2     | 10           | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~28hpi)
MT2_Mmi_2c_3     | 10           | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~28hpi)
                 |              |
CTRi_4c_1        | 7            | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~44hpi)
CTRi_4c_2        | 6            | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~44hpi)
MT2_Mmi_4c_1     | 6            | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~44hpi)
MT2_Mmi_4c_2     | 6            | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~44hpi)
                 |              |
CTRi_8c_1        | 7            | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~54hpi)
CTRi_8c_2        | 7            | dCas9-KRAB (100ng/ul) + CTR sgRNA (60ng/ul) (~54hpi)
MT2_Mmi_8c_1     | 7            | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~54hpi)
MT2_Mmi_8c_2     | 5            | dCas9-KRAB (100ng/ul) + MT2_Mm 6xsgRNA (60ng/ul, 10ng/ul each) (~54hpi) 

*Note that all libraries were generated using Takara SMART-Seq Stranded for total RNA-seq kit.\  

*Note that all libraries were sequenced using NovaSeq platform (2 x 150bp).\     

*For all samples, 1ul of 1/10000 ERCC mix was spiked in as external control for some of the comparisons.\ 

```{r}
source("./utils.R")

#Sample info
sampleName <- c("CTRi_e2C_RNA_rep1", "CTRi_e2C_RNA_rep2",
                "MT2Mmi_e2C_RNA_rep1", "MT2Mmi_e2C_RNA_rep2",
                "Noninj_2C_RNA_rep1", "Noninj_2C_RNA_rep2",
                "CTRi_2C_RNA_rep1", "CTRi_2C_RNA_rep2", "CTRi_2C_RNA_rep3",
                "MT2Mmi_2C_RNA_rep1", "MT2Mmi_2C_RNA_rep2", "MT2Mmi_2C_RNA_rep3",
                "CTRi_4C_RNA_rep1", "CTRi_4C_RNA_rep2",
                "MT2Mmi_4C_RNA_rep1", "MT2Mmi_4C_RNA_rep2",
                "CTRi_8C_RNA_rep1", "CTRi_8C_RNA_rep2",
                "MT2Mmi_8C_RNA_rep1", "MT2Mmi_8C_RNA_rep2")

simpleName <- c("CTRi_e2c_1", "CTRi_e2c_2",
                "MT2_Mmi_e2c_1", "MT2_Mmi_e2c_2",
                "noninj_2c_1", "noninj_2c_2",
                "CTRi_2c_1", "CTRi_2c_2", "CTRi_2c_3",
                "MT2_Mmi_2c_1", "MT2_Mmi_2c_2", "MT2_Mmi_2c_3",
                "CTRi_4c_1", "CTRi_4c_2",
                "MT2_Mmi_4c_1", "MT2_Mmi_4c_2",
                "CTRi_8c_1", "CTRi_8c_2",
                "MT2_Mmi_8c_1", "MT2_Mmi_8c_2")

#input data
counts <- inputTEcountfiles(sampleName, paste0(simpleName, ".count"), 
                            countDataPath = "./R_input/MT2i_TEcount/")

rpkm <- inputStringTieRPKMfiles(sampleName, paste0(simpleName, ".rpkm"),
                                RPKMDataPath = "./R_input/MT2i_FPKM/")

ERCC_counts <- inputTEcountfiles(sampleName, paste0(simpleName, ".count"),
                                 countDataPath = "./R_input/MT2i_ERCCcount/")

write.csv(rpkm, "./R_output/MT2i_totalRNA/231024_MT2_Mmi_fpkm.csv", quote = F, row.names = F)
```

### 4.1.2 Correlation heatmap
```{r, fig.width = 3, fig.height =3 }
suppressMessages(library(reshape2))
suppressMessages(library(ggplot2))
suppressMessages(library(cowplot))

logrpkm <- log2(rpkm[, c(3:ncol(rpkm))] + 0.1)
logrpkmCor <- round(cor(logrpkm), 3)
logrpkmMelted <- melt(logrpkmCor)

theme <- theme(
  panel.grid.major.y = element_line(),
  panel.grid.major.x = element_line(),
  panel.grid.minor.x = element_line(),
  panel.grid.minor.y = element_blank(),
  panel.background = element_rect(fill="white"),
  axis.text.x = element_text(colour = "black", size = rel(1.5), angle = 90),  #size of x axis
  axis.text.y = element_text(colour = "black", size = rel(1.5)),  #size of y axis
  axis.title.y = element_blank(),
  axis.title.x = element_blank(),
  panel.border = element_rect(colour="black", fill=NA, size=0.5)
)

fig1 <- ggplot(data = logrpkmMelted, aes(Var1, Var2, fill = value)) +
  geom_tile (color = "white") + 
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0.8, 
                       limit = c(0.6, 1), space = 'Lab', name = "Pearson\nCorrelation") + 
  theme
fig1

#focus on only 2C
logrpkmMelted_2C <- logrpkmMelted[grep("2c", logrpkmMelted$Var1),]
logrpkmMelted_2C <- logrpkmMelted_2C[grep("2c", logrpkmMelted_2C$Var2),]
fig1_1 <- ggplot(data = logrpkmMelted_2C, aes(Var1, Var2, fill = value)) +
  geom_tile (color = "white") + 
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0.9, 
                       limit = c(0.8, 1), space = 'Lab', name = "Pearson\nCorrelation") + 
  theme
fig1_1
pdf(file = paste0("./figures/231115_MT2i_totalRNAseq_correation_heatmap.pdf"), 
    width = 8, height = 6)
fig1
fig1_1
dev.off()
```

### 4.1.3 dCas9 expression
```{r, fig.height = 2, fig.width =3 }
df4dCas9 <- data.frame(
  sample = simpleName,
  libsize = colSums(counts[2:ncol(counts)]),
  #ERCC_counts = colSums(ERCC_counts[2:ncol(counts)])/2,
  ERCC_count = c(313943, 370534, 293593, 417234,
                 208429, 185830, 214729, 201724, 249430, 283126, 273946, 
                 261345, 350919, 515401, 502195, 536221, 309387, 273268, 364596, 454521),
  dCas9_count = c(211237, 242886, 241157, 225462,
                  19, 32, 99643, 102736, 126410, 175147, 141915, 148397,
                  80571, 93302, 74417, 110291, 52757, 59646, 61298, 58057)
)
df4dCas9$Ratio2ERCC <- df4dCas9$dCas9_count / df4dCas9$ERCC_count 

#df4dCas9$sample <- factor(df4dCas9$sample, levels = df4dCas9$sample)
fig3 <- ggplot(df4dCas9[c(5:12),], aes(x = (sample), y = Ratio2ERCC)) + 
  geom_bar(stat="identity") + theme_cowplot(16) +
  scale_x_discrete(limits=df4dCas9[c(5:12),]$sample,
                   labels = c("Noninj_1", "Noninj_2",
                              "CTRi_1", "CTRi_2", "CTRi_3",
                              "MT2_Mmi_1", "MT2_Mmi_2", "MT2_Mmi_3")) +
  theme(axis.text.x = element_text(angle = 30, vjust = 0.9, hjust=0.9)) +
  xlab(NULL) + ylab("Normalized to ERCC spike-in") +
  ggtitle("dCas9-KRAB read counts (2C)")
fig3

pdf(file = paste0("./figures/231115_MT2i_dCas9-KRAB_read_counts_2C.pdf"), 
    width = 6, height = 4)
fig3
dev.off()
```
### 4.1.4 Non-inject vs. CTRi

Normalized by ERCC-spike in
```{r, fig.height = 2, fig.width = 3}
suppressMessages(library(stringr))
suppressMessages(library(DESeq2))

FPKMcutoff = 0
foldchange = 2

#combine TEcounts & ERCC counts
row.names(ERCC_counts) <- ERCC_counts$id
counts_w_ERCC <- rbind(counts, ERCC_counts)
row.names(counts_w_ERCC) <- counts_w_ERCC$id
counts_w_ERCC <- counts_w_ERCC[, c("noninj_2c_1.count", "noninj_2c_2.count",
                                   "CTRi_2c_1.count", "CTRi_2c_2.count",
                                   "CTRi_2c_3.count")]

min_read <- 10
counts_w_ERCC <- counts_w_ERCC[apply(counts_w_ERCC, 1, function(x){sum(x)}) > min_read, ]
groups <- factor(c(rep("CGroup",2),rep("TGroup",3)))
sampleInfo <- data.frame(groups, row.names = colnames(counts_w_ERCC))

dds <- DESeqDataSetFromMatrix(countData = counts_w_ERCC, colData = sampleInfo, design = ~ groups)
dds$groups = relevel(dds$groups,ref="CGroup")

####
####Normalize using ERCC counts
####
ERCC_genes <- str_detect(row.names(counts_w_ERCC), regex("ERCC"))
dds <- estimateSizeFactors(dds, controlGenes= ERCC_genes)
dds <- DESeq(dds)
res <- results(dds,independentFiltering=F)
  
#normalized counts obtained based on DESeq2 "plotCounts" source code
normCounts <- counts(dds, normalized = TRUE)
results <- as.data.frame(cbind(normCounts, res))
results$id <- row.names(results)  

#merge to rpkm table
CTR_noninj_DESeq_rpkm_ERCC <- merge(results[, c("id", "noninj_2c_1.count", 
                                      "noninj_2c_2.count", "CTRi_2c_1.count", 
                                      "CTRi_2c_2.count", "CTRi_2c_3.count",
                                      "log2FoldChange", "padj")], 
                               rpkm[, c("id", "name", 
                                        "noninj_2c_1.rpkm", "noninj_2c_2.rpkm",
                                        "CTRi_2c_1.rpkm", "CTRi_2c_2.rpkm", "CTRi_2c_3.rpkm")], 
                               by = "id")
CTR_noninj_DESeq_rpkm_ERCC$padj[is.na(CTR_noninj_DESeq_rpkm_ERCC$padj)] <- 1

#classify DEGs
CTR_noninj_DESeq_rpkm_ERCC.group <- classifyDEG(
                            CTR_noninj_DESeq_rpkm_ERCC,
                            ctr.rpkm = c("noninj_2c_1.rpkm", "noninj_2c_2.rpkm"),
                            trt.rpkm = c("CTRi_2c_1.rpkm", "CTRi_2c_2.rpkm", "CTRi_2c_3.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange", 
                            RPKM = FPKMcutoff, log2FC = log2(foldchange), FDR = 0.05)

CTR_noninj_DESeq_rpkm_ERCC <- cbind(CTR_noninj_DESeq_rpkm_ERCC, CTR_noninj_DESeq_rpkm_ERCC.group)
  
CTR_noninj_DESeq_rpkm_ERCC.up <- CTR_noninj_DESeq_rpkm_ERCC[CTR_noninj_DESeq_rpkm_ERCC.group == "up-regulated",]
CTR_noninj_DESeq_rpkm_ERCC.down <- CTR_noninj_DESeq_rpkm_ERCC[CTR_noninj_DESeq_rpkm_ERCC.group == "down-regulated",]
CTR_noninj_DESeq_rpkm_ERCC.detectable <- CTR_noninj_DESeq_rpkm_ERCC[CTR_noninj_DESeq_rpkm_ERCC.group != "low_expression_level",]
 
nrow(CTR_noninj_DESeq_rpkm_ERCC.up) #3
nrow(CTR_noninj_DESeq_rpkm_ERCC.down) #0
nrow(CTR_noninj_DESeq_rpkm_ERCC.detectable) #24936

write.table(CTR_noninj_DESeq_rpkm_ERCC, "./R_output/MT2i_totalRNA/231115_MT2i_CTR_noninj_DESeq_rpkm_ERCC.csv", quote = F, row.names = F, sep = ",")

#plotting
CTR_noninj_DESeq_rpkm_ERCC$log2noninj <- log2(
        (CTR_noninj_DESeq_rpkm_ERCC$noninj_2c_1.count + CTR_noninj_DESeq_rpkm_ERCC$noninj_2c_2.count)/2 + 1)
CTR_noninj_DESeq_rpkm_ERCC$log2CTR <- log2(
        (CTR_noninj_DESeq_rpkm_ERCC$CTRi_2c_1.count + CTR_noninj_DESeq_rpkm_ERCC$CTRi_2c_2.count +
         CTR_noninj_DESeq_rpkm_ERCC$CTRi_2c_3.count)/3 + 1) 
CTR_noninj_DESeq_rpkm_ERCC.up.label <- paste0("Up-regulated:\n", nrow(CTR_noninj_DESeq_rpkm_ERCC.up),
                                  " (", format(round(nrow(CTR_noninj_DESeq_rpkm_ERCC.up)/nrow(CTR_noninj_DESeq_rpkm_ERCC.detectable)*100, 2), 
                                               nsmall = 2), "%)")
CTR_noninj_DESeq_rpkm_ERCC.down.label <- paste0("Down-regulated:\n", nrow(CTR_noninj_DESeq_rpkm_ERCC.down),
                                  " (", format(round(nrow(CTR_noninj_DESeq_rpkm_ERCC.down)/nrow(CTR_noninj_DESeq_rpkm_ERCC.detectable)*100, 2), 
                                               nsmall = 2), "%)")

fig5 <- ggScatterplot(CTR_noninj_DESeq_rpkm_ERCC, x = "log2noninj", y = "log2CTR",
                      group = "CTR_noninj_DESeq_rpkm_ERCC.group", gene = "name", xlab = "Non-injected 2C",
                      ylab = "CTRi (2C)",
                      title = "CTRi vs non-injected (2C) (spike-in normalized)",
                      label.up = CTR_noninj_DESeq_rpkm_ERCC.up.label, 
                      label.down = CTR_noninj_DESeq_rpkm_ERCC.down.label,
                      genes4Label = c(
                        "Zfp809","Zscan4c", "Zfp352", "Tpt1"),
                      FC.line = foldchange,
                       my.color=c("grey50", "red3"))
fig5

pdf(file = paste0("./figures/231115_MT2i_DEGs_analyses_CTRi_vs_noninj_spikein_normalized.pdf"), 
    width = 6, height = 5)
fig5
dev.off()
```

## 4.2 Differential Expression

### 4.2.1 Differential Repeats
```{r}
row.names(counts) <- counts$id
suppressMessages(
  ec2_DESeq <- countsToDEseq2FDR(counts = counts[, c("CTRi_e2c_1.count", "CTRi_e2c_2.count",
                                                     "MT2_Mmi_e2c_1.count", "MT2_Mmi_e2c_2.count")], 
                                 CGroup = 2, TGroup = 2)
)

suppressMessages(
  c2_DESeq <- countsToDEseq2FDR(counts = counts[, c( "noninj_2c_1.count", "noninj_2c_2.count",
                                                     "CTRi_2c_1.count", 
                                                     "CTRi_2c_2.count", "CTRi_2c_3.count",
                                                     "MT2_Mmi_2c_1.count", "MT2_Mmi_2c_2.count",
                                                     "MT2_Mmi_2c_3.count")], 
                                 CGroup = 5, TGroup = 3)
)

suppressMessages(
  c4_DESeq <- countsToDEseq2FDR(counts = counts[, c("CTRi_4c_1.count", "CTRi_4c_2.count",
                                                    "MT2_Mmi_4c_1.count", "MT2_Mmi_4c_2.count")], 
                                 CGroup = 2, TGroup = 2)
)

suppressMessages(
  c8_DESeq <- countsToDEseq2FDR(counts = counts[, c("CTRi_8c_1.count", "CTRi_8c_2.count",
                                                    "MT2_Mmi_8c_1.count", "MT2_Mmi_8c_2.count")], 
                                 CGroup = 2, TGroup = 2)
)

ec2_repeats <- ec2_DESeq[grep(":", ec2_DESeq$id),]
c2_repeats <- c2_DESeq[grep(":", c2_DESeq$id),]
c4_repeats <- c4_DESeq[grep(":", c4_DESeq$id),]
c8_repeats <- c8_DESeq[grep(":", c8_DESeq$id),]

ec2_repeats$padj[is.na(ec2_repeats$padj)] <- 1
c2_repeats$padj[is.na(c2_repeats$padj)] <- 1
c4_repeats$padj[is.na(c4_repeats$padj)] <- 1
c8_repeats$padj[is.na(c8_repeats$padj)] <- 1

ec2_repeats.group <- classifyDEG(ec2_repeats, 
                                 ctr.rpkm = c("CTRi_e2c_1.count", "CTRi_e2c_2.count"),
                                 trt.rpkm = c("MT2_Mmi_e2c_1.count", "MT2_Mmi_e2c_2.count"),
                                 FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                 log2FC = log2(foldchange), FDR = 0.05)


c2_repeats.group <- classifyDEG(c2_repeats, 
                                 ctr.rpkm = c("noninj_2c_1.count", "noninj_2c_2.count",
                                   "CTRi_2c_1.count", "CTRi_2c_2.count", "CTRi_2c_3.count"),
                                 trt.rpkm = c("MT2_Mmi_2c_1.count", "MT2_Mmi_2c_2.count", "MT2_Mmi_2c_3.count"),
                                 FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                 log2FC = log2(foldchange), FDR = 0.05)

c4_repeats.group <- classifyDEG(c4_repeats, 
                                 ctr.rpkm = c("CTRi_4c_1.count", "CTRi_4c_2.count"),
                                 trt.rpkm = c("MT2_Mmi_4c_1.count", "MT2_Mmi_4c_2.count"),
                                 FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                 log2FC = log2(foldchange), FDR = 0.05)

c8_repeats.group <- classifyDEG(c8_repeats, 
                                 ctr.rpkm = c("CTRi_8c_1.count", "CTRi_8c_2.count"),
                                 trt.rpkm = c("MT2_Mmi_8c_1.count", "MT2_Mmi_8c_2.count"),
                                 FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                 log2FC = log2(foldchange), FDR = 0.05)

ec2_repeats <- cbind(ec2_repeats, ec2_repeats.group)
c2_repeats <- cbind(c2_repeats, c2_repeats.group)
c4_repeats <- cbind(c4_repeats, c4_repeats.group)
c8_repeats <- cbind(c8_repeats, c8_repeats.group)

ec2_repeats.up <- ec2_repeats[ec2_repeats.group == "up-regulated",]
ec2_repeats.down <- ec2_repeats[ec2_repeats.group == "down-regulated",]

c2_repeats.up <- c2_repeats[c2_repeats.group == "up-regulated", ]
c2_repeats.down <- c2_repeats[c2_repeats.group == "down-regulated",]

c4_repeats.up <- c4_repeats[c4_repeats.group == "up-regulated",]
c4_repeats.down <- c4_repeats[c4_repeats.group == "down-regulated",]

c8_repeats.up <- c8_repeats[c8_repeats.group == "up-regulated",]
c8_repeats.down <- c8_repeats[c8_repeats.group == "down-regulated",]

#fc = 2, fc = 3
nrow(ec2_repeats.up) #2 #0 
nrow(ec2_repeats.down) #40 #21

nrow(c2_repeats.up) #15 #6
nrow(c2_repeats.down) #59 #33

nrow(c4_repeats.up) #12 #8
nrow(c4_repeats.down) #23 #14

nrow(c8_repeats.up) #44 #14
nrow(c8_repeats.down) #5 #3

write.table(ec2_repeats, "./R_output/MT2i_totalRNA/231115_MT2i_e2C_repeats_DE_analyses.csv",
            quote = F, sep = ",", row.names = F)
write.table(c2_repeats, "./R_output/MT2i_totalRNA/231115_MT2i_2C_repeats_DE_analyses.csv", 
            quote = F, sep = ",", row.names = F)
write.table(c4_repeats, "./R_output/MT2i_totalRNA/231115_MT2i_4C_repeats_DE_analyses.csv", 
            quote = F, sep = ",", row.names = F)
write.table(c8_repeats, "./R_output/MT2i_totalRNA/231115_MT2i_8C_repeats_DE_analyses.csv", 
            quote = F, sep = ",", row.names = F)
```

```{r repeats scatter plots, fig.height = 4, fig.width = 6}
ec2_repeats$log2CTR <- log2(
  (ec2_repeats$CTRi_e2c_1.count + ec2_repeats$CTRi_e2c_2.count) / 2 + 1
)
ec2_repeats$log2MT2Mm <- log2(
  (ec2_repeats$MT2_Mmi_e2c_1.count + ec2_repeats$MT2_Mmi_e2c_2.count) / 2 + 1
)
ec2_repeats.up.label <- paste0("Up-regulated:", nrow(ec2_repeats.up))
ec2_repeats.down.label <- paste0("Down-regulated:", nrow(ec2_repeats.down))

c2_repeats$log2CTR <- log2(
  ( c2_repeats$noninj_2c_1.count + c2_repeats$noninj_2c_2.count + 
    c2_repeats$CTRi_2c_1.count + c2_repeats$CTRi_2c_2.count + c2_repeats$CTRi_2c_3.count)/5 + 1)
c2_repeats$log2MT2Mm <- log2(
  (c2_repeats$MT2_Mmi_2c_1.count + c2_repeats$MT2_Mmi_2c_2.count + c2_repeats$MT2_Mmi_2c_3.count)/3 + 1)
c2_repeats.up.label <- paste0("Up-regulated:", nrow(c2_repeats.up))
c2_repeats.down.label <- paste0("Down-regulated:", nrow(c2_repeats.down))

c4_repeats$log2CTR <- log2(
  (c4_repeats$CTRi_4c_1.count + c4_repeats$CTRi_4c_2.count)/2 + 1)
c4_repeats$log2MT2Mm <- log2(
  (c4_repeats$MT2_Mmi_4c_1.count + c4_repeats$MT2_Mmi_4c_2.count)/2 + 1)
c4_repeats.up.label <- paste0("Up-regulated:", nrow(c4_repeats.up))
c4_repeats.down.label <- paste0("Down-regulated:", nrow(c4_repeats.down))

c8_repeats$log2CTR <- log2(
  (c8_repeats$CTRi_8c_1.count + c8_repeats$CTRi_8c_2.count)/2 + 1)
c8_repeats$log2MT2Mm <- log2(
  (c8_repeats$MT2_Mmi_8c_1.count + c8_repeats$MT2_Mmi_8c_2.count)/2 + 1)
c8_repeats.up.label <- paste0("Up-regulated:", nrow(c8_repeats.up))
c8_repeats.down.label <- paste0("Down-regulated:", nrow(c8_repeats.down))

MT_MT2 <- c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR", "MT2B:ERVL:LTR",
             "MT2B1:ERVL:LTR", "MT2B2:ERVL:LTR", "MER89:ERV1:LTR",
            "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR") 
            #"GSAT_MM:Satellite:Satellite")

fig6 <- ggScatterplot(ec2_repeats, x = "log2CTR", y = "log2MT2Mm",
                      group = "ec2_repeats.group", gene = "id", xlab = "CRISPRi CTR",
                      ylab = "CRISPRi MT2_Mm",
                      title = "MT2_Mm CRISPRi (repeats)(e2C)",
                      label.up = ec2_repeats.up.label, label.down = ec2_repeats.down.label,
                      my.color=c("blue", "grey50", "red3"),
                      genes4Label = MT_MT2,
                      FC.line = foldchange)  
fig6 <- fig6 +  geom_abline(intercept = log2(5), slope = 1, linetype = "F1") + 
    geom_abline(intercept = -log2(5), slope = 1, linetype = "F1") 
#fig6

fig7 <- ggScatterplot(c2_repeats, x = "log2CTR", y = "log2MT2Mm",
                      group = "c2_repeats.group", gene = "id", xlab = "CRISPRi CTR",
                      ylab = "CRISPRi MT2_Mm",
                      title = "MT2_Mm CRISPRi (repeats)(2C)",
                      label.up = c2_repeats.up.label, label.down = c2_repeats.down.label,
                      my.color=c("blue", "grey50", "red3"),
                      genes4Label = MT_MT2,
                      FC.line = foldchange)            
fig7 <- fig7 +  geom_abline(intercept = log2(5), slope = 1, linetype = "F1") + 
    geom_abline(intercept = -log2(5), slope = 1, linetype = "F1") 
#fig7

fig8 <- ggScatterplot(c4_repeats, x = "log2CTR", y = "log2MT2Mm",
                      group = "c4_repeats.group", gene = "id", xlab = "CRISPRi CTR",
                      ylab = "CRISPRi MT2_Mm",
                      title = "MT2_Mm CRISPRi (repeats)(4C)",
                      label.up = c4_repeats.up.label, label.down = c4_repeats.down.label,
                      my.color=c("blue", "grey50", "red3"),
                      genes4Label = MT_MT2,
                      FC.line = foldchange)  
fig8 <- fig8 + geom_abline(intercept = log2(5), slope = 1, linetype = "F1") + 
    geom_abline(intercept = -log2(5), slope = 1, linetype = "F1")
#fig8

fig9 <- ggScatterplot(c8_repeats, x = "log2CTR", y = "log2MT2Mm",
                      group = "c8_repeats.group", gene = "id", xlab = "CRISPRi CTR",
                      ylab = "CRISPRi MT2_Mm",
                      title = "MT2_Mm CRISPRi (repeats)(8C)",
                      label.up = c8_repeats.up.label, label.down = c8_repeats.down.label,
                      my.color=c("blue", "grey50", "red3"),
                      genes4Label = MT_MT2,
                      FC.line = foldchange)
fig9 <- fig9 + geom_abline(intercept = log2(5), slope = 1, linetype = "F1") + 
    geom_abline(intercept = -log2(5), slope = 1, linetype = "F1")
#fig9

plot_grid(fig6, fig7, fig8, fig9, nrow = 2)
pdf(file = paste0("./figures/231115_MT2i_repeats_DE_scatterplot.pdf"), 
    width = 6, height = 5)
fig6
fig7
fig8
fig9
dev.off()
```


```{r repeats bar plot}
suppressMessages(library(ggplot2))
suppressMessages(library(reshape2))
suppressMessages(library(tidyr))
suppressMessages(library(cowplot))

ec2_MT2Mm <- ec2_repeats[which(ec2_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR", "MT2C_Mm:ERVL:LTR")),
                         c("CTRi_e2c_1.count", "CTRi_e2c_2.count", 
                           "MT2_Mmi_e2c_1.count", "MT2_Mmi_e2c_2.count")]
ec2_MT2Mm <- t(ec2_MT2Mm)
ec2_MT2Mm <- cbind(ec2_MT2Mm,
                   group = c(rep("CTRi", 2), rep("MT2_Mmi", 2)),
                   stage = "e2C")

c2_MT2Mm <- c2_repeats[which(c2_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR", "MT2C_Mm:ERVL:LTR")), 
                        c("CTRi_2c_1.count", "CTRi_2c_2.count", "CTRi_2c_3.count",
                          "MT2_Mmi_2c_1.count", "MT2_Mmi_2c_2.count", "MT2_Mmi_2c_3.count")]
c2_MT2Mm <- t(c2_MT2Mm)

c2_MT2Mm <- cbind(c2_MT2Mm,
                  group = c(rep("CTRi", 3), rep("MT2_Mmi", 3)),
                  stage = "2C")

# c4_MT2Mm <- c4_repeats[which(c4_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR")),
#                        c("CTRi_4c_1.count", "CTRi_4c_2.count", 
#                          "MT2_Mmi_4c_1.count", "MT2_Mmi_4c_2.count")]
# c4_MT2Mm <- t(c4_MT2Mm)
# c4_MT2Mm <- cbind(c4_MT2Mm,
#                   group = c(rep("CTRi", 2), rep("MT2_Mmi", 2)),
#                   stage = "4C")
# 
# 
# c8_MT2Mm <- c8_repeats[which(c8_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR")),
#                        c("CTRi_8c_1.count", "CTRi_8c_2.count", 
#                          "MT2_Mmi_8c_1.count", "MT2_Mmi_8c_2.count")]
# c8_MT2Mm <- t(c8_MT2Mm)
# c8_MT2Mm <- cbind(c8_MT2Mm,
#                   group = c(rep("CTRi", 2), rep("MT2_Mmi", 2)),
#                   stage = "8C")

df4fig_MT2Mm <- as.data.frame(rbind(ec2_MT2Mm, c2_MT2Mm))
df4fig_MT2Mm$`MERVL-int:ERVL:LTR` <- as.numeric(df4fig_MT2Mm$`MERVL-int:ERVL:LTR`)
df4fig_MT2Mm$`MT2_Mm:ERVL:LTR` <- as.numeric(df4fig_MT2Mm$`MT2_Mm:ERVL:LTR`)
df4fig_MT2Mm$`MT2C_Mm:ERVL:LTR` <- as.numeric(df4fig_MT2Mm$`MT2C_Mm:ERVL:LTR`)

fig10 <- ggplot(df4fig_MT2Mm, aes(x = stage, y = `MT2_Mm:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("CTRi", "MT2_Mmi"),
                    values = c("grey80", "white")) +
  scale_x_discrete(limits = c("e2C", "2C")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MT2_Mm")
#fig10

fig10.1 <- ggplot(df4fig_MT2Mm, aes(x = stage, y = `MT2C_Mm:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("CTRi", "MT2_Mmi"),
                    values = c("grey80", "white")) +
  scale_x_discrete(limits = c("e2C", "2C")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MT2C_Mm")
#fig10.1

fig11 <- ggplot(df4fig_MT2Mm, aes(x = stage, y = `MERVL-int:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("CTRi", "MT2_Mmi"),
                    values = c("grey80", "white")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  scale_x_discrete(limits = c("e2C", "2C")) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MERVL-int")
#fig11
plot_grid(fig10, fig10.1, fig11,  align = "h", labels = c("A", "B", "C"))
pdf(file = "./figures/231115_MT2i_MT2_Mm_MERVVL_barplot_e2C_2C.pdf",
    width = 8, height = 8)
plot_grid(fig10, fig10.1, fig11,  align = "h", labels = c("A", "B", "C"))
dev.off()
```

```{r repeat heatmap, fig.height = 3, fig.with = 1}
ec2_repeats_MT2Mm <- ec2_repeats[which(ec2_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR",
                                                                    "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR",
                                                                    "MER89:ERV1:LTR")),
                                 c("id", "log2FoldChange")]

c2_repeats_MT2Mm <- c2_repeats[which(c2_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR",
                                                                    "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR",
                                                                    "MER89:ERV1:LTR")),
                               c("id", "log2FoldChange")]

ec2_repeats_MT2Mm_mat <- as.matrix(ec2_repeats_MT2Mm[, "log2FoldChange"])
row.names(ec2_repeats_MT2Mm_mat) <- ec2_repeats_MT2Mm$id

c2_repeats_MT2Mm_mat <- as.matrix(c2_repeats_MT2Mm[, "log2FoldChange"])
row.names(c2_repeats_MT2Mm_mat)  <- c2_repeats_MT2Mm$id
pheatmap(ec2_repeats_MT2Mm_mat, cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T, 
         display_numbers = round(ec2_repeats_MT2Mm_mat,2), main = "E2C")
pheatmap(c2_repeats_MT2Mm_mat, cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T,
         display_numbers = round(c2_repeats_MT2Mm_mat,2), main = "E2C")
pdf("./figures/231115_MT2i_repeat_heatmap.pdf", height = 6, width = 5)
pheatmap(ec2_repeats_MT2Mm_mat, cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T, 
         display_numbers = round(ec2_repeats_MT2Mm_mat,2), main = "E2C")
pheatmap(c2_repeats_MT2Mm_mat, cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T,
         display_numbers = round(c2_repeats_MT2Mm_mat,2), main = "L2C")
dev.off()
```

### 4.2.2 Differential Gene
```{r DE genes table}
ec2_DESeq_rpkm <- merge(ec2_DESeq[, c("id",
                                      "CTRi_e2c_1.count", "CTRi_e2c_2.count",
                                      "MT2_Mmi_e2c_1.count", "MT2_Mmi_e2c_2.count",
                                      "log2FoldChange", "padj")],
                        rpkm[, c("id", "name", 
                                 "CTRi_e2c_1.rpkm", "CTRi_e2c_2.rpkm", 
                                 "MT2_Mmi_e2c_1.rpkm", "MT2_Mmi_e2c_2.rpkm")],
                        by = "id"
                        )

c2_DESeq_rpkm <- merge(c2_DESeq[, c("id", 
                                    "noninj_2c_1.count", "noninj_2c_2.count",
                                    "CTRi_2c_1.count", "CTRi_2c_2.count",
                                    "CTRi_2c_3.count", 
                                    "MT2_Mmi_2c_1.count",
                                    "MT2_Mmi_2c_2.count", "MT2_Mmi_2c_3.count",
                                    "log2FoldChange", "padj")],
                       rpkm[, c("id", "name", 
                                "noninj_2c_1.rpkm", "noninj_2c_2.rpkm",
                                "CTRi_2c_1.rpkm", "CTRi_2c_2.rpkm", "CTRi_2c_3.rpkm",
                                "MT2_Mmi_2c_1.rpkm", "MT2_Mmi_2c_2.rpkm", "MT2_Mmi_2c_3.rpkm")],
                       by = "id")

c4_DESeq_rpkm <- merge(c4_DESeq[, c("id", "CTRi_4c_1.count", "CTRi_4c_2.count",
                                    "MT2_Mmi_4c_1.count", "MT2_Mmi_4c_2.count",
                                    "log2FoldChange", "padj")],
                       rpkm[, c("id", "name", 
                                "CTRi_4c_1.rpkm", "CTRi_4c_2.rpkm",
                                "MT2_Mmi_4c_1.rpkm", "MT2_Mmi_4c_2.rpkm")],
                       by = "id")

c8_DESeq_rpkm <- merge(c8_DESeq[, c("id", "CTRi_8c_1.count", "CTRi_8c_2.count",
                                    "MT2_Mmi_8c_1.count", "MT2_Mmi_8c_2.count",
                                    "log2FoldChange", "padj")],
                       rpkm[, c("id", "name", 
                                "CTRi_8c_1.rpkm", "CTRi_8c_2.rpkm",
                                "MT2_Mmi_8c_1.rpkm", "MT2_Mmi_8c_2.rpkm")],
                       by = "id")

ec2_DESeq_rpkm$padj[is.na(ec2_DESeq_rpkm$padj)] <- 1
c2_DESeq_rpkm$padj[is.na(c2_DESeq_rpkm$padj)] <- 1
c4_DESeq_rpkm$padj[is.na(c4_DESeq_rpkm$padj)] <- 1
c8_DESeq_rpkm$padj[is.na(c8_DESeq_rpkm$padj)] <- 1

ec2_DESeq_rpkm.group <- classifyDEG(
                            ec2_DESeq_rpkm,
                            ctr.rpkm = c("CTRi_e2c_1.rpkm", "CTRi_e2c_2.rpkm"),
                            trt.rpkm = c("MT2_Mmi_e2c_1.rpkm", "MT2_Mmi_e2c_2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = FPKMcutoff, log2FC = log2(foldchange), FDR = 0.05)

c2_DESeq_rpkm.group <- classifyDEG(
                            c2_DESeq_rpkm,
                            ctr.rpkm = c("noninj_2c_1.rpkm", "noninj_2c_2.rpkm",
                              "CTRi_2c_1.rpkm", "CTRi_2c_2.rpkm", "CTRi_2c_3.rpkm"),
                            trt.rpkm = c("MT2_Mmi_2c_1.rpkm", "MT2_Mmi_2c_2.rpkm", "MT2_Mmi_2c_3.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = FPKMcutoff, log2FC = log2(foldchange), FDR = 0.05)

c4_DESeq_rpkm.group <- classifyDEG(
                            c4_DESeq_rpkm,
                            ctr.rpkm = c("CTRi_4c_1.rpkm", "CTRi_4c_2.rpkm"),
                            trt.rpkm = c("MT2_Mmi_4c_1.rpkm", "MT2_Mmi_4c_2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = FPKMcutoff, log2FC = log2(foldchange), FDR = 0.05)

c8_DESeq_rpkm.group <- classifyDEG(
                            c8_DESeq_rpkm,
                            ctr.rpkm = c("CTRi_8c_1.rpkm", "CTRi_8c_2.rpkm"),
                            trt.rpkm = c("MT2_Mmi_8c_1.rpkm", "MT2_Mmi_8c_2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = FPKMcutoff, log2FC = log2(foldchange), FDR = 0.05)

ec2_DESeq_rpkm <- cbind(ec2_DESeq_rpkm, ec2_DESeq_rpkm.group)
c2_DESeq_rpkm <- cbind(c2_DESeq_rpkm, c2_DESeq_rpkm.group)
c4_DESeq_rpkm <- cbind(c4_DESeq_rpkm, c4_DESeq_rpkm.group)
c8_DESeq_rpkm <- cbind(c8_DESeq_rpkm, c8_DESeq_rpkm.group)

ec2_DESeq_rpkm.up <- ec2_DESeq_rpkm[ec2_DESeq_rpkm.group == "up-regulated",]
ec2_DESeq_rpkm.down <- ec2_DESeq_rpkm[ec2_DESeq_rpkm.group == "down-regulated",]
ec2_DESeq_rpkm.detectable <- ec2_DESeq_rpkm[ec2_DESeq_rpkm.group != "low_expression_level",]

c2_DESeq_rpkm.up <- c2_DESeq_rpkm[c2_DESeq_rpkm.group == "up-regulated",]
c2_DESeq_rpkm.down <- c2_DESeq_rpkm[c2_DESeq_rpkm.group == "down-regulated",]
c2_DESeq_rpkm.detectable <- c2_DESeq_rpkm[c2_DESeq_rpkm.group != "low_expression_level",]

c4_DESeq_rpkm.up <- c4_DESeq_rpkm[c4_DESeq_rpkm.group == "up-regulated",]
c4_DESeq_rpkm.down <- c4_DESeq_rpkm[c4_DESeq_rpkm.group == "down-regulated",]
c4_DESeq_rpkm.detectable <- c4_DESeq_rpkm[c4_DESeq_rpkm.group != "low_expression_level",]

c8_DESeq_rpkm.up <- c8_DESeq_rpkm[c8_DESeq_rpkm.group == "up-regulated",]
c8_DESeq_rpkm.down <- c8_DESeq_rpkm[c8_DESeq_rpkm.group == "down-regulated",]
c8_DESeq_rpkm.detectable <- c8_DESeq_rpkm[c8_DESeq_rpkm.group != "low_expression_level",]

nrow(ec2_DESeq_rpkm.up) #158 
nrow(ec2_DESeq_rpkm.down) #341
nrow(ec2_DESeq_rpkm.detectable) #25342

nrow(c2_DESeq_rpkm.up) #481
nrow(c2_DESeq_rpkm.down) #1111
nrow(c2_DESeq_rpkm.detectable) #28068

nrow(c4_DESeq_rpkm.up) #610
nrow(c4_DESeq_rpkm.down) #596
nrow(c4_DESeq_rpkm.detectable) #22651

nrow(c8_DESeq_rpkm.up) #740
nrow(c8_DESeq_rpkm.down) #502
nrow(c8_DESeq_rpkm.detectable) #22288

write.table(ec2_DESeq_rpkm, "./R_output/MT2i_totalRNA/231115_MT2i_e2C_genes_DE_analyses.csv",
            quote = F, sep = ",", row.names = F)
write.table(c2_DESeq_rpkm, "./R_output/MT2i_totalRNA/231115_MT2i_2C_genes_DE_analyses.csv", 
            quote = F, sep = ",", row.names = F)
write.table(c4_DESeq_rpkm, "./R_output/MT2i_totalRNA/231115_MT2i_4C_genes_DE_analyses.csv", 
            quote = F, sep = ",", row.names = F)
write.table(c8_DESeq_rpkm, "./R_output/MT2i_totalRNA/231115_MT2i_8C_genes_DE_analyses.csv", 
            quote = F, sep = ",", row.names = F)
```

```{r DE gene plotting, fig.height = 4, fig.width = 6}
#early 2-cell
ec2_DESeq_rpkm$log2CTR <- log2(
  (ec2_DESeq_rpkm$CTRi_e2c_1.count + ec2_DESeq_rpkm$CTRi_e2c_2.count) / 2 + 1
)
ec2_DESeq_rpkm$log2MT2Mm <- log2(
  (ec2_DESeq_rpkm$MT2_Mmi_e2c_1.count + ec2_DESeq_rpkm$MT2_Mmi_e2c_2.count) / 2 + 1
)
ec2_DESeq_rpkm.up.label <- paste0("Up-regulated:\n", nrow(ec2_DESeq_rpkm.up),
                                  " (", format(round(nrow(ec2_DESeq_rpkm.up)/nrow(ec2_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")
ec2_DESeq_rpkm.down.label <- paste0("Down-regulated:\n", nrow(ec2_DESeq_rpkm.down),
                                 " (", format(round(nrow(ec2_DESeq_rpkm.down)/nrow(ec2_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")

#2-cell
c2_DESeq_rpkm$log2CTR <- log2(
  ( c2_DESeq_rpkm$noninj_2c_1.count + c2_DESeq_rpkm$noninj_2c_2.count + 
    c2_DESeq_rpkm$CTRi_2c_1.count + c2_DESeq_rpkm$CTRi_2c_2.count + c2_DESeq_rpkm$CTRi_2c_3.count)/5 + 1)
c2_DESeq_rpkm$log2MT2Mm <- log2(
  (c2_DESeq_rpkm$MT2_Mmi_2c_1.count + c2_DESeq_rpkm$MT2_Mmi_2c_2.count + c2_DESeq_rpkm$MT2_Mmi_2c_3.count)/3 + 1
)

c2_DESeq_rpkm.up.label <- paste0("Up-regulated:\n", nrow(c2_DESeq_rpkm.up),
                                 " (", format(round(nrow(c2_DESeq_rpkm.up)/nrow(c2_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")

c2_DESeq_rpkm.down.label <- paste0("Down-regulated:\n", nrow(c2_DESeq_rpkm.down),
                                 " (", format(round(nrow(c2_DESeq_rpkm.down)/nrow(c2_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")

#4-cell
c4_DESeq_rpkm$log2CTR <- log2(
  (c4_DESeq_rpkm$CTRi_4c_1.count + c4_DESeq_rpkm$CTRi_4c_2.count)/2 + 1)
c4_DESeq_rpkm$log2MT2Mm <- log2(
  (c4_DESeq_rpkm$MT2_Mmi_4c_1.count + c4_DESeq_rpkm$MT2_Mmi_4c_2.count)/2 + 1)
c4_DESeq_rpkm.up.label <- paste0("Up-regulated:\n", nrow(c4_DESeq_rpkm.up),
                                 " (", format(round(nrow(c4_DESeq_rpkm.up)/nrow(c4_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")
c4_DESeq_rpkm.down.label <- paste0("Down-regulated:\n", nrow(c4_DESeq_rpkm.down),
                                 " (", format(round(nrow(c4_DESeq_rpkm.down)/nrow(c4_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")

#8-cell
c8_DESeq_rpkm$log2CTR <- log2(
  (c8_DESeq_rpkm$CTRi_8c_1.count + c8_DESeq_rpkm$CTRi_8c_2.count)/2 + 1)
c8_DESeq_rpkm$log2MT2Mm <- log2(
  (c8_DESeq_rpkm$MT2_Mmi_8c_1.count + c8_DESeq_rpkm$MT2_Mmi_8c_2.count)/2 + 1)
c8_DESeq_rpkm.up.label <- paste0("Up-regulated:\n", nrow(c8_DESeq_rpkm.up),
                                 " (", format(round(nrow(c8_DESeq_rpkm.up)/nrow(c8_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")
c8_DESeq_rpkm.down.label <- paste0("Down-regulated:\n", nrow(c8_DESeq_rpkm.down),
                                 " (", format(round(nrow(c8_DESeq_rpkm.down)/nrow(c8_DESeq_rpkm.detectable)*100, 2), nsmall = 2), "%)")

#genes to label:
#Zfp809
#Tpt1
fig12 <- ggScatterplot(ec2_DESeq_rpkm, x = "log2CTR", y = "log2MT2Mm",
                       group = "ec2_DESeq_rpkm.group", gene = "name", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       title = "MT2_Mm CRISPRi (genes)(e2C)",
                       label.up = ec2_DESeq_rpkm.up.label,
                       label.down = ec2_DESeq_rpkm.down.label,
                       genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange,
                       my.color=c("blue", "grey50", "red3"))
#fig12
fig13 <- ggScatterplot(c2_DESeq_rpkm, x = "log2CTR", y = "log2MT2Mm",
                       group = "c2_DESeq_rpkm.group", gene = "name", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       title = "MT2_Mm CRISPRi (genes)(2C)",
                       label.up = c2_DESeq_rpkm.up.label,
                       label.down = c2_DESeq_rpkm.down.label,
                       genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange,
                       my.color=c("blue", "grey50", "red3"))
#fig13

fig14 <- ggScatterplot(c4_DESeq_rpkm, x = "log2CTR", y = "log2MT2Mm",
                       group = "c4_DESeq_rpkm.group", gene = "name", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       title = "MT2_Mm CRISPRi (genes)(4C)",
                       label.up = c4_DESeq_rpkm.up.label,
                       label.down = c4_DESeq_rpkm.down.label,
                       genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange,
                       my.color=c("blue", "grey50", "red3"))
#fig14

fig15 <- ggScatterplot(c8_DESeq_rpkm, x = "log2CTR", y = "log2MT2Mm",
                       group = "c8_DESeq_rpkm.group", gene = "name", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       title = "MT2_Mm CRISPRi (genes)(8C)",
                       label.up = c8_DESeq_rpkm.up.label,
                       label.down = c8_DESeq_rpkm.down.label,
                       genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange,
                       my.color=c("blue", "grey50", "red3"))
#fig15 

pdf("./figures/231115_MT2i_genes_DE_scatterplot.pdf", width = 6, height = 5)
fig12
fig13
fig14
fig15
dev.off()
plot_grid(fig12, fig13, fig14, fig15, align = "h", labels = c("A", "B", "C", "D"))
```
###4.2.3 DEG GO term analyses
```{r GO term}
suppressMessages(library(clusterProfiler))
suppressMessages(library(org.Mm.eg.db))
suppressMessages(library(enrichplot))
suppressMessages(library(ggplot2))
suppressMessages(library(cowplot))

e2c_up_go <- enrichGO(gene = ec2_DESeq_rpkm.up$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

e2c_down_go <- enrichGO(gene = ec2_DESeq_rpkm.down$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

c2_up_go <- enrichGO(gene = c2_DESeq_rpkm.up$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

c2_down_go <- enrichGO(gene = c2_DESeq_rpkm.down$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

c4_down_go <- enrichGO(gene = c4_DESeq_rpkm.down$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

c4_up_go <- enrichGO(gene = c4_DESeq_rpkm.up$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

c8_up_go <- enrichGO(gene = c8_DESeq_rpkm.up$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

c8_down_go <- enrichGO(gene = c8_DESeq_rpkm.down$name, OrgDb = org.Mm.eg.db,
                       keyType = "SYMBOL", ont = "BP", pAdjustMethod = "BH", 
                       pvalueCutoff  = 0.01, qvalueCutoff  = 0.01)

pdf("./figures/231115_MT2i_GO_terms.pdf", width = 10, height = 5)
#plotGoBars(e2c_up_go) #no significant hits
plotGoBars(e2c_down_go) #no significant hits
plotGoBars(c2_up_go, top = 20, title = "c2_up")
plotGoBars(c2_down_go, top = 20, title = "c2_down")
plotGoBars(c4_down_go, top = 20, title = "c4_down")
plotGoBars(c4_up_go, top = 20, title = "c4_up")
plotGoBars(c8_up_go, top = 20, title = "c8_up")
plotGoBars(c8_down_go, top = 20, title = "c8_down")
dev.off()

write.table(as.data.frame(e2c_down_go), "./R_output/MT2i_totalRNA/231115_MT2i_ec2_down_GO.csv", 
            quote = F, sep = ",", row.names = F)
write.table(as.data.frame(c2_up_go), "./R_output/MT2i_totalRNA/231115_MT2i_c2_up_GO.csv", 
            quote = F, sep = ",", row.names = F)
write.table(as.data.frame(c2_down_go), "./R_output/MT2i_totalRNA/231115_MT2i_c2_down_GO.csv",
            quote = F, sep = ",", row.names = F)
write.table(as.data.frame(c4_up_go), "./R_output/MT2i_totalRNA/231115_MT2i_c4_up_GO.csv",
            quote = F, sep = ",", row.names = F)
write.table(as.data.frame(c4_down_go), "./R_output/MT2i_totalRNA/231115_MT2i_c4_down_GO.csv",
            quote = F, sep = ",", row.names = F)
write.table(as.data.frame(c8_up_go), "./R_output/MT2i_totalRNA/231115_MT2i_c8_up_GO.csv",
            quote = F, sep = ",", row.names = F)
write.table(as.data.frame(c8_down_go), "./R_output/MT2i_totalRNA/231115_MT2i_c8_down_GO.csv",
            quote = F, sep = ",", row.names = F)
```

### 4.2.4 Obox/Dux expression in MT2i
```{r Obox expression upon MT2_Mmi}
Obox <- rpkm[grep(c("Obox"), rpkm$name), ]
Dux <- rpkm[grep("Duxf3", rpkm$name),]

#aggregate Obox3-ps and Obox4-ps
Obox4_ps <- Obox[grep("Obox4-ps", Obox$name),]
Obox4_ps_mean <- colMeans(Obox4_ps[, c(3:ncol(Obox4_ps))])

Obox3_ps <- Obox[grep("Obox3-ps", Obox$name),]
Obox3_ps_mean <- colMeans(Obox3_ps[, c(3:ncol(Obox3_ps))])

Obox_cleaned <- rbind(Obox[which(Obox$name %in% c("Obox1", "Obox2", "Obox5", 
                                                  "Obox7", "Obox3",
                                                  "Obox6")),],
                      c("Obox4-ps", "Obox4-ps", Obox4_ps_mean),
                      c("Obox3-ps", "Obox3-ps", Obox3_ps_mean))
Obox_cleaned <- rbind(Obox_cleaned, Dux)
Obox_cleaned$ctr_e2c <- (as.numeric(Obox_cleaned$CTRi_e2c_1.rpkm) + 
                         as.numeric(Obox_cleaned$CTRi_e2c_2.rpkm)) / 2
Obox_cleaned$mt2_e2c <- (as.numeric(Obox_cleaned$MT2_Mmi_e2c_1.rpkm) +
                         as.numeric(Obox_cleaned$CTRi_e2c_2.rpkm)) /2

Obox_cleaned$ctr_l2c <- (as.numeric(Obox_cleaned$noninj_2c_1.rpkm) + 
                         as.numeric(Obox_cleaned$noninj_2c_2.rpkm) +
                         as.numeric(Obox_cleaned$CTRi_2c_1.rpkm) + 
                         as.numeric(Obox_cleaned$CTRi_2c_2.rpkm) +
                         as.numeric(Obox_cleaned$CTRi_2c_3.rpkm)) / 5
Obox_cleaned$mt2_l2c <- (as.numeric(Obox_cleaned$MT2_Mmi_2c_1.rpkm) + 
                         as.numeric(Obox_cleaned$MT2_Mmi_2c_2.rpkm) + 
                         as.numeric(Obox_cleaned$MT2_Mmi_2c_3.rpkm)) / 3

Obox_cleaned$e2c_log2FC <- log2(Obox_cleaned$mt2_e2c+0.01) - log2(Obox_cleaned$ctr_e2c + 0.01)
Obox_cleaned$l2c_log2FC <- log2(Obox_cleaned$mt2_l2c+0.01) - log2(Obox_cleaned$ctr_l2c + 0.01)

pdf("./figures/231115_Obox_expression_upon_MT2Mmi_heatmap.pdf", height = 5, width = 4)
Obox_cleaned_e2c_mat <- as.matrix(Obox_cleaned[, "e2c_log2FC"])
row.names(Obox_cleaned_e2c_mat) <- Obox_cleaned[, "name"]
Obox_cleaned_l2c_mat <- as.matrix(Obox_cleaned[, "l2c_log2FC"])
row.names(Obox_cleaned_l2c_mat) <- Obox_cleaned[, "name"]

pheatmap(Obox_cleaned_e2c_mat, cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("red","white", "blue")))(400),
         breaks = seq(-2, 2, by = 0.01), show_rownames = T,
         display_numbers = round(Obox_cleaned_e2c_mat,2))
pheatmap(Obox_cleaned_l2c_mat, cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("red","white", "blue")))(400),
breaks = seq(-2, 2, by = 0.01), show_rownames = T,
         display_numbers = round(Obox_cleaned_l2c_mat,2))
dev.off()
Obox_cleaned[, c("name", "e2c_log2FC", "l2c_log2FC")]
```
# 5. Define minor/major ZGA, maternal decay

```{r input zhang data}
#input GSE169632 TEcount data
#GSE169632 generated totalRNAseq data for MII, 1C (12hpi), 2C(30hpi), 4C(48hpi), morula(72hpi) and blastocyst(96hpi)
#The same library construction method was used. Only difference is 2x75bp.
#Use this public dataset as baseline to check the data quality.

#this data does not have early 2C RNAseq
#thus I downloaded early2C data from GSE207222 (same sequencing length and library preparation method)
source("./utils.R")
zhang_sampleName <- c("MII_total_rep1", "MII_total_rep2",
                      "total_1c_rep1", "total_1c_rep2",
                      "e2C_total_rep1", "e2C_total_rep2", #Wang data
                      "total_2c_rep1", "total_2c_rep2",
                      "total_4c_rep1", "total_4c_rep2",
                      "total_morula_rep1", "total_morula_rep2",
                      "total_BL_rep1", "total_BL_rep2")
zhang_simpleName <- c("MII_1", "MII_2",
                      "total_1c_1", "total_1c_2",
                      "total_e2c_1", "total_e2c_2",
                      "total_2c_1", "total_2c_2",
                      "total_4c_1", "total_4c_2",
                      "total_mor_1", "total_mor_2",
                      "total_BL_1", "total_BL_2")
zhang_counts <- inputTEcountfiles(zhang_sampleName, paste0(zhang_simpleName, ".count"), 
                            countDataPath = "./R_input/zhang_wang_TEcount/")

zhang_fpkm <- inputStringTieRPKMfiles(zhang_sampleName, paste0(zhang_simpleName, ".rpkm"),
                                RPKMDataPath = "./R_input/zhang_wang_FPKM/")
write.csv(zhang_fpkm, "./R_output/zhang_wang_RNA/231115_zhang_totalRNA_fpkm.csv", quote = F, row.names = F)

zhang_fpkm$MII <- (zhang_fpkm$MII_1.rpkm + zhang_fpkm$MII_2.rpkm) / 2
zhang_fpkm$c1 <- (zhang_fpkm$total_1c_1.rpkm + zhang_fpkm$total_1c_2.rpkm) / 2
zhang_fpkm$e2c <- (zhang_fpkm$total_e2c_1.rpkm + zhang_fpkm$total_e2c_2.rpkm ) / 2
zhang_fpkm$l2c <- (zhang_fpkm$total_2c_1.rpkm + zhang_fpkm$total_2c_2.rpkm) / 2

zhang_fpkm$l2c_MII_fc <- log2(zhang_fpkm$l2c + 0.01) - log2(zhang_fpkm$MII + 0.01)
zhang_fpkm$c1_MII_fc <- log2(zhang_fpkm$c1 + 0.01) - log2(zhang_fpkm$MII + 0.01)
zhang_fpkm$e2c_MII_fc <- log2(zhang_fpkm$e2c + 0.01) - log2(zhang_fpkm$MII + 0.01)

#230803 Define major ZGA genes 
#MII, FPKM < 5, late 2C FPKM > 5 & FC > 3
#or 
#MII, FPKM > 5, late 2C FC > 5
zhang_fpkm$majorZGA <- rep("non_majorZGA", nrow(zhang_fpkm))

for(i in 1:nrow(zhang_fpkm)){
  if(zhang_fpkm$MII[i] <= 5){
    if(zhang_fpkm$l2c[i] > 5 & zhang_fpkm$l2c_MII_fc[i] > log2(3)){
      zhang_fpkm$majorZGA[i] <- "majorZGA"
    }  
  } else if (zhang_fpkm$l2c_MII_fc[i] > log2(5)){
      zhang_fpkm$majorZGA[i] <- "majorZGA"
  }
}
nrow(zhang_fpkm[which(zhang_fpkm$majorZGA == "majorZGA"), ]) #1989

majorZGA <- zhang_fpkm[which(zhang_fpkm$majorZGA == "majorZGA"), ]
#boxplot(log2(mZGA$MII+0.01), log2(mZGA$l2c+0.01))
write.table(majorZGA, 
            "./R_output/zhang_wang_RNA/231115_zhang_totalRNA_majorZGA.csv", 
            sep = ",", quote = F, row.names = F)

#230817 Define minor ZGA genes
#MII, FPKM < 5, 1C FPKM > 5 & FC > 3
#or
#MII, FPKM > 5, 1C / e2C FC > 5
zhang_fpkm$minorZGA <- rep("non_minorZGA", nrow(zhang_fpkm))

for(i in 1:nrow(zhang_fpkm)){
  if(zhang_fpkm$MII[i] <= 5){
      if(zhang_fpkm$c1[i] > 5 & zhang_fpkm$c1_MII_fc[i] > log2(3)){
        zhang_fpkm$minorZGA[i] <- "minorZGA"
      } else if (zhang_fpkm$e2c[i] > 5 & zhang_fpkm$e2c_MII_fc[i] > log2(3)){
        zhang_fpkm$minorZGA[i] <- "minorZGA"
      }
  } else if (zhang_fpkm$e2c_MII_fc[i] > log2(5)){
      zhang_fpkm$minorZGA[i] <- "minorZGA"
  } else if (zhang_fpkm$c1_MII_fc[i] > log2(5)){
      zhang_fpkm$minorZGA[i] <- "minorZGA"
  }
}
nrow(zhang_fpkm[which(zhang_fpkm$minorZGA == "minorZGA"), ]) #717

minorZGA <- zhang_fpkm[which(zhang_fpkm$minorZGA == "minorZGA"), ]
#boxplot(log2(mZGA$MII+0.01), log2(mZGA$l2c+0.01))
write.table(minorZGA, 
            "./R_output/zhang_wang_RNA/231115_zhang_totalRNA_minorZGA.csv", 
            sep = ",", quote = F, row.names = F)

#230817 Define maternal genes
#MII FPKM > 5 & latec < -log2(3)
zhang_fpkm$maternal_gene <- rep("non_maternalGene", nrow(zhang_fpkm))

for(i in 1:nrow(zhang_fpkm)){
  if(zhang_fpkm$MII[i] > 5){
    if(zhang_fpkm$l2c_MII_fc[i] < -log2(3)){
      zhang_fpkm$maternal_gene[i] <- "maternal_gene"
    }
  }
}
nrow(zhang_fpkm[which(zhang_fpkm$maternal_gene == "maternal_gene"), ]) #2768

maternal_gene <- zhang_fpkm[which(zhang_fpkm$maternal_gene == "maternal_gene"), ]
#boxplot(log2(mZGA$MII+0.01), log2(mZGA$l2c+0.01))
write.table(maternal_gene, 
            "./R_output/zhang_wang_RNA/231115_zhang_totalRNA_maternal_gene.csv", 
            sep = ",", quote = F, row.names = F)

#make heatmap to illustrate minor, major ZGA, and maternal genes
pdf("./figures/231115_minor_majorZGA_maternalGene_heatmap.pdf", 
    height = 6, width = 5)
suppressMessages(require(pheatmap))
pheatmap(as.matrix(log2(minorZGA[, c(3:10)]+0.01)), scale = "row", 
         cluster_cols = F, treeheight_row = 0, show_rownames = F,
         color = colorRampPalette(rev(c("red", "white", "blue")))(400), 
         breaks = seq(-2, 2, by = 0.01))
pheatmap(as.matrix(log2(majorZGA[, c(3:10)]+0.01)), scale = "row", 
         cluster_cols = F, treeheight_row = 0, show_rownames = F,
         color = colorRampPalette(rev(c("red", "white", "blue")))(400), 
         breaks = seq(-2, 2, by = 0.01))
pheatmap(as.matrix(log2(maternal_gene[, c(3:10)]+0.01)), scale = "row", 
         cluster_cols = F, treeheight_row = 0, show_rownames = F,
         color = colorRampPalette(rev(c("red", "white", "blue")))(400), 
         breaks = seq(-2, 2, by = 0.01))
# boxplot(log2(majorZGA[, c(3:10)]+0.01))
# boxplot(log2(minorZGA[, c(3:10)]+0.01))
# boxplot(log2(maternal_gene[, c(3:10)]+0.01))
dev.off()
```
# 6. MT2i DEGs characterization

## 6.1 ZGA/maternal gene upon MT2i
```{r minor/major ZGA and maternal gene, fig.height = 4, fig.width = 6}
suppressMessages(library(dplyr))
minorZGA <- read.table("./R_output/zhang_wang_RNA/231115_zhang_totalRNA_minorZGA.csv", sep = ",", header = T)
majorZGA <- read.table("./R_output/zhang_wang_RNA/231115_zhang_totalRNA_majorZGA.csv", sep = ",", header = T)
maternal_gene <- read.table("./R_output/zhang_wang_RNA/231115_zhang_totalRNA_maternal_gene.csv", 
                            sep = ",", header = T)
MT2_e2c_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_e2C_genes_DE_analyses.csv", sep =",", header = T)
MT2_c2_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_2C_genes_DE_analyses.csv", sep = ",", header = T)

#for later scatter plot
MT2_e2c_RNA$logCTR <- log2((MT2_e2c_RNA$CTRi_e2c_1.count + MT2_e2c_RNA$CTRi_e2c_2.count)/2 + 1)
MT2_e2c_RNA$logMT2Mm <- log2((MT2_e2c_RNA$MT2_Mmi_e2c_1.count + MT2_e2c_RNA$MT2_Mmi_e2c_2.count)/2 + 1)
MT2_c2_RNA$logCTR <- log2((MT2_c2_RNA$noninj_2c_1.count + MT2_c2_RNA$noninj_2c_2.count + MT2_c2_RNA$CTRi_2c_1.count
                    + MT2_c2_RNA$CTRi_2c_2.count + MT2_c2_RNA$CTRi_2c_3.count)/5 + 1)
MT2_c2_RNA$logMT2Mm <- log2((MT2_c2_RNA$MT2_Mmi_2c_1.count + MT2_c2_RNA$MT2_Mmi_2c_2.count +
                        MT2_c2_RNA$MT2_Mmi_2c_3.count)/3 + 1)

nrow(minorZGA) #717
nrow(majorZGA) #1989
nrow(maternal_gene) #2768

minorZGA_e2c <- left_join(minorZGA, MT2_e2c_RNA, by = "id")
nrow(minorZGA_e2c) #717
table(minorZGA_e2c$ec2_DESeq_rpkm.group) 
# write.table(minorZGA_e2c, "./R_output/MT2i_/231024_minorZGA_e2C_MT2Mmi.csv", sep = ",",
#             quote = F, row.names = F)
#down-regulated       similar_level         up-regulated 
#                 106                                 553                  50

maternal_gene_e2c <- left_join(maternal_gene, MT2_e2c_RNA, by = "id")
nrow(maternal_gene_e2c)
table(maternal_gene_e2c$ec2_DESeq_rpkm.group)
#similar_level  up-regulated 
#         2760             3

majorZGA_l2c <- left_join(majorZGA, MT2_c2_RNA, by = "id")
nrow(majorZGA_l2c) #1989
table(majorZGA_l2c$c2_DESeq_rpkm.group) 
# write.table(majorZGA_l2c, "../R_output/231024_majorZGA_l2C_MT2Mmi.csv", sep = ",",
#             quote = F, row.names = F)
#down-regulated low_expression_level        similar_level         up-regulated 
#                 320                    8                 1632                   21 
 
majorZGA_l2c_MT2i_down <- majorZGA_l2c[which(majorZGA_l2c$c2_DESeq_rpkm.group == "down-regulated"),]

maternal_gene_l2c <- left_join(maternal_gene, MT2_c2_RNA, by = "id")
nrow(maternal_gene_l2c)
table(maternal_gene_l2c$c2_DESeq_rpkm.group)
#down-regulated low_expression_level        similar_level         up-regulated 
#                  22                    4                 2717                   20 

#Make scatter plot
pdf("./figures/231115_MT2i_minor_major_maternal_ZGA_scatterplot.pdf", 
    width = 6, height = 5)
foldchange = 2
fig1 <- ggScatterplot(minorZGA_e2c, x = "logCTR", y = "logMT2Mm",
                       group = "ec2_DESeq_rpkm.group", gene = "name.y", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       title = paste0("minor ZGA (e2C):", nrow(minorZGA)),
                       my.color=c("blue", "grey50", "red3"),
                       label.up = paste0(nrow(minorZGA_e2c[which(minorZGA_e2c$ec2_DESeq_rpkm.group == "up-regulated"),])),
                       label.down = paste0(nrow(minorZGA_e2c[which(minorZGA_e2c$ec2_DESeq_rpkm.group == "down-regulated"),])),
                       #genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange)

fig2 <-ggScatterplot(maternal_gene_e2c, x = "logCTR", y = "logMT2Mm",
                       group = "ec2_DESeq_rpkm.group", gene = "name.y", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       my.color=c("grey50", "red3"),
                       title = paste0("maternal (e2C):", nrow(maternal_gene)),
                       label.up = paste0(nrow(maternal_gene_e2c[which(maternal_gene_e2c$ec2_DESeq_rpkm.group == "up-regulated"),])),
                       label.down = paste0(nrow(maternal_gene_e2c[which(maternal_gene_e2c$ec2_DESeq_rpkm.group == "down-regulated"),])),
                       #genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange)

fig3 <- ggScatterplot(majorZGA_l2c, x = "logCTR", y = "logMT2Mm",
                       group = "c2_DESeq_rpkm.group", gene = "name.y", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       title = paste0("major ZGA (l2C):", nrow(majorZGA)),
                       my.color=c("blue", "grey50", "red3"),
                       label.up = paste0(nrow(majorZGA_l2c[which(majorZGA_l2c$c2_DESeq_rpkm.group == "up-regulated"),])),
                       label.down = paste0(nrow(majorZGA_l2c[which(majorZGA_l2c$c2_DESeq_rpkm.group == "down-regulated"),])),
                       #genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange)


fig4 <- ggScatterplot(maternal_gene_l2c, x = "logCTR", y = "logMT2Mm",
                       group = "c2_DESeq_rpkm.group", gene = "name.y", xlab = "CRISPRi CTR",
                       ylab = "CRISPRi MT2_Mm",
                       title = paste0("maternal_gene (l2C):", nrow(maternal_gene)),
                       my.color=c("blue", "grey50", "red3"),
                       label.up = paste0(nrow(maternal_gene_l2c[which(maternal_gene_l2c$c2_DESeq_rpkm.group == "up-regulated"),])),
                       label.down = paste0(nrow(maternal_gene_l2c[which(maternal_gene_l2c$c2_DESeq_rpkm.group == "down-regulated"),])),
                       #genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange)
dev.off()
plot_grid(fig1, fig2, fig3, fig4, nrow = 2)
```

## 6.2 4C-up gene characterization
```{r 4C DEG dynamic expression, fig.height = 2, fig.width = 3}
zhang_fpkm$c4 <- (zhang_fpkm$total_4c_1.rpkm + zhang_fpkm$total_4c_2.rpkm)/2
zhang_fpkm$mor <- (zhang_fpkm$total_mor_1.rpkm + zhang_fpkm$total_mor_2.rpkm)/2
zhang_fpkm$bl <- (zhang_fpkm$total_BL_1.rpkm + zhang_fpkm$total_BL_2.rpkm)/2

MT2i_c4_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_4C_genes_DE_analyses.csv", 
                          sep = ",", header = T)
MT2i_c4_RNA_up <- MT2i_c4_RNA[which(MT2i_c4_RNA$c4_DESeq_rpkm.group == "up-regulated"),]
nrow(MT2i_c4_RNA_up) #610
zhang_fpkm_4c_up <- zhang_fpkm[which(zhang_fpkm$id %in% MT2i_c4_RNA_up$id),
                                 c(17:20, 27:29)]

#perform two-sided wilcox test 
wilcox.test(zhang_fpkm_4c_up$e2c, 
            zhang_fpkm_4c_up$c1, alternative = "two.sided")$p.value
#5.59e-12
wilcox.test(zhang_fpkm_4c_up$l2c, 
            zhang_fpkm_4c_up$c4, alternative = "two.sided")$p.value
#1.057e-12

df4boxplot <- gather(log2(zhang_fpkm_4c_up+0.01), "stage", "FPKM")
pdf("./figures/231115_MT2i_4C_upregulated genes in development.pdf", height = 5, width = 6)
ggplot(df4boxplot, aes(x = stage, y = FPKM)) + 
  geom_boxplot(color = "black", alpha = 0.2, outlier.color = NA) +
  scale_x_discrete(limits = c("MII","c1", "e2c", "l2c", "c4", "mor", "bl")) +
  theme(axis.title.x = element_blank()) +
  ggtitle("4C_up-regulated genes") +theme_cowplot(16) 
dev.off()
ggplot(df4boxplot, aes(x = stage, y = FPKM)) + 
  geom_boxplot(color = "black", alpha = 0.2, outlier.color = NA) +
  scale_x_discrete(limits = c("MII","c1", "e2c", "l2c", "c4", "mor", "bl")) +
  theme(axis.title.x = element_blank()) +
  ggtitle("4C_up-regulated genes") +theme_cowplot(16) 
``` 

## 6.3 DEG distance to MT2

### 6.3.1 Define MT2 gene distance
```{bash}
#!/usr/bin/bash 

module load bedtools/2.27.0

TE="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed"
GTF="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/gencode.vM25.annotation.gtf"

grep "MT2_Mm" $TE > mm10_MT2.bed
grep "MT2C_Mm" $TE >> mm10_MT2.bed

awk 'OFS="\t" {if ($3=="gene") {print $1,$4-1,$5,$10,$14,$7}}' $GTF | tr -d '";' > mm10_gene.bed
sort -k1,1 -k2,2n mm10_gene.bed > mm10_gene_sorted.bed
sort -k1,1 -k2,2n mm10_MT2.bed > mm10_MT2_sorted.bed

chrLength="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/chrNameLength.txt"
sort -k1,1 -k2,2n $chrLength > chrNameLength_sorted.txt

bedtools closest -a mm10_gene_sorted.bed -b mm10_MT2_sorted.bed -D "b" -g chrNameLength_sorted.txt | uniq > 231115_nearest_gene2MT2.txt
```

### 6.3.2 DEG to nearest MT2
```{r}
MT2_e2c_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_e2C_genes_DE_analyses.csv", sep =",", header = T)
MT2_c2_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_2C_genes_DE_analyses.csv", sep = ",", header = T)

gene_MT2_distance <- read.table("./R_input/dCas9_ChIPseq/231115_nearest_gene2MT2.txt",
                                         header = F, sep = "\t")
nrow(MT2_e2c_RNA[which(MT2_e2c_RNA$ec2_DESeq_rpkm.group == "down-regulated"),]) #341  
nrow(MT2_c2_RNA[which(MT2_c2_RNA$c2_DESeq_rpkm.group == "down-regulated"),]) #1111

nrow(gene_MT2_distance) #55915
colnames(gene_MT2_distance) <- c("gene_chr", "gene_start", "gene_end", 
                                   "id", "name", "gene_strand", 
                                   "peak_chr", "peak_start", "peak_end", 
                                   "peak_name", "peak_score", "peak_strand",
                                   "distance")
#for genes with multiple peaks in the gene bodies, randomly pick one
gene_MT2_distance <- gene_MT2_distance[!duplicated(gene_MT2_distance$id),]
nrow(gene_MT2_distance) #55401
```

```{r Classify DEGs by distance to TEs}
source("./utils.R")
MT2_c2_RNA <- merge(MT2_c2_RNA, gene_MT2_distance[, c("id", "distance")],
                by = "id")
MT2_e2c_RNA <- merge(MT2_e2c_RNA, gene_MT2_distance[, c("id", "distance")],
                 by = "id")
MT2_c2_RNA$bin <- classify_distance(as.numeric(MT2_c2_RNA$distance), ignore_direction = T)
MT2_e2c_RNA$bin <- classify_distance(as.numeric(MT2_e2c_RNA$distance), ignore_direction = T)

MT2_c2_RNA.up <- MT2_c2_RNA[which(MT2_c2_RNA$c2_DESeq_rpkm.group == "up-regulated"),]
MT2_c2_RNA.down <- MT2_c2_RNA[which(MT2_c2_RNA$c2_DESeq_rpkm.group == "down-regulated"),]
MT2_e2c_RNA.up <- MT2_e2c_RNA[which(MT2_e2c_RNA$ec2_DESeq_rpkm.group == "up-regulated"),]
MT2_e2c_RNA.down <- MT2_e2c_RNA[which(MT2_e2c_RNA$ec2_DESeq_rpkm.group == "down-regulated"),]

table(MT2_c2_RNA.up$bin)
table(MT2_c2_RNA.down$bin)
table(MT2_e2c_RNA.up$bin)
table(MT2_e2c_RNA.down$bin)
# >50kb  0-10kb 10-30kb 30-50kb 
#     423       4      22      32

#  >50kb  0-10kb 10-30kb 30-50kb 
#    366     565     126      54 

#>50kb  0-10kb 10-30kb 30-50kb 
#    124       9       9      16 

#>50kb  0-10kb 10-30kb 30-50kb 
#    42     266      27       6 

#Make pie chart here.
pie(table(MT2_c2_RNA.up$bin), main = paste0("late2C: up-regulated genes:", nrow(MT2_c2_RNA.up)))
pie(table(MT2_c2_RNA.down$bin), main = paste0("late2C: down-regulated genes:", nrow(MT2_c2_RNA.down)))
pie(table(MT2_e2c_RNA.up$bin), main = paste0("early2C: up-regulated genes:", nrow(MT2_e2c_RNA.up)))
pie(table(MT2_e2c_RNA.down$bin), main = paste0("early2C: down-regulated genes:", nrow(MT2_e2c_RNA.down)))
pdf("./figures/231115_MT2i_genes_DE_byDisatance_pieChart_FC2.pdf", width = 5, height = 5)
pie(table(MT2_c2_RNA.up$bin), main = paste0("late2C: up-regulated genes:", nrow(MT2_c2_RNA.up)))
pie(table(MT2_c2_RNA.down$bin), main = paste0("late2C: down-regulated genes:", nrow(MT2_c2_RNA.down)))
pie(table(MT2_e2c_RNA.up$bin), main = paste0("early2C: up-regulated genes:", nrow(MT2_e2c_RNA.up)))
pie(table(MT2_e2c_RNA.down$bin), main = paste0("early2C: down-regulated genes:", nrow(MT2_e2c_RNA.down)))
dev.off()
```

#7. Sakashita data comparison

## 7.1 Sakashita differential repeats
```{r input data}
source("./utils.R")

#Sample info
sampleName <- tools::file_path_sans_ext(list.files("./R_input/Sakashita_data/", 
                                                   pattern = ".cntTable"))

counts <- inputTEcountfiles(sampleName, paste0(sampleName, ".count"), 
                            countDataPath = "./R_input/Sakashita_data/")
rpkm <- inputStringTieRPKMfiles(sampleName, paste0(sampleName, ".rpkm"),
                                RPKMDataPath = "./R_input/Sakashita_data/")

row.names(counts) <- counts$id

suppressMessages(
  KD2C_DESeq <- countsToDEseq2FDR(counts = counts[, c("Ctrl_2C_rep1.count", "Ctrl_2C_rep2.count",
                                                     "MERVLKD_2C_rep1.count", "MERVLKD_2C_rep2.count")], 
                                 CGroup = 2, TGroup = 2))
suppressMessages(  
  KD4C_DESeq <- countsToDEseq2FDR(counts = counts[, c("Ctrl_4C_rep1.count", "Ctrl_4C_rep2.count",
                                                     "MERVLKD_4C_rep1.count", "MERVLKD_4C_rep2.count")], 
                                 CGroup = 2, TGroup = 2))
suppressMessages(  
  KD8C_DESeq <- countsToDEseq2FDR(counts = counts[, c("Ctrl_8C_rep1.count", "Ctrl_8C_rep2.count",
                                                     "MERVLKD_8C_rep1.count", "MERVLKD_8C_rep2.count")], 
                                 CGroup = 2, TGroup = 2))
suppressMessages(    
  i2C_DESeq  <- countsToDEseq2FDR(counts = counts[, c("GFPi_2C_rep1.count", "GFPi_2C_rep2.count",
                                                     "MERVLi_2C_rep1.count", "MERVLi_2C_rep2.count")], 
                                 CGroup = 2, TGroup = 2))
suppressMessages(    
  i4C_DESeq  <- countsToDEseq2FDR(counts = counts[, c("GFPi_4C_rep1.count", "GFPi_4C_rep2.count",
                                                     "MERVLi_4C_rep1.count", "MERVLi_4C_rep2.count")], 
                                 CGroup = 2, TGroup = 2))
suppressMessages(    
  i8C_DESeq  <- countsToDEseq2FDR(counts = counts[, c("GFPi_8C_rep1.count", "GFPi_8C_rep2.count",
                                                     "MERVLi_8C_rep1.count", "MERVLi_8C_rep2.count")], 
                                 CGroup = 2, TGroup = 2))

KD2C_repeats <- KD2C_DESeq[grep(":", KD2C_DESeq$id),]
KD4C_repeats <- KD4C_DESeq[grep(":", KD4C_DESeq$id),]
KD8C_repeats <- KD8C_DESeq[grep(":", KD8C_DESeq$id),]
i2C_repeats <- i2C_DESeq[grep(":", i2C_DESeq$id),]
i4C_repeats <- i4C_DESeq[grep(":", i4C_DESeq$id),]
i8C_repeats <- i8C_DESeq[grep(":", i8C_DESeq$id),]

#some repeats with padj = NA (due to huge variation), replace with 1
KD2C_repeats$padj[is.na(KD2C_repeats$padj)] <- 1
KD4C_repeats$padj[is.na(KD4C_repeats$padj)] <- 1
KD8C_repeats$padj[is.na(KD8C_repeats$padj)] <- 1
i2C_repeats$padj[is.na(i2C_repeats$padj)] <- 1
i4C_repeats$padj[is.na(i4C_repeats$padj)] <- 1
i4C_repeats$padj[is.na(i4C_repeats$padj)] <- 1

KD2C_repeats.group <- classifyDEG(KD2C_repeats,
                                  ctr.rpkm = c("Ctrl_2C_rep1.count", "Ctrl_2C_rep2.count"),
                                  trt.rpkm = c("MERVLKD_2C_rep1.count", "MERVLKD_2C_rep2.count"),
                                  FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                  log2FC = log2(2), FDR = 0.05)

KD4C_repeats.group <- classifyDEG(KD4C_repeats,
                                  ctr.rpkm = c("Ctrl_4C_rep1.count", "Ctrl_4C_rep2.count"),
                                  trt.rpkm = c("MERVLKD_4C_rep1.count", "MERVLKD_4C_rep2.count"),
                                  FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                  log2FC = log2(2), FDR = 0.05)
  
KD8C_repeats.group <- classifyDEG(KD8C_repeats,
                                  ctr.rpkm = c("Ctrl_8C_rep1.count", "Ctrl_8C_rep2.count"),
                                  trt.rpkm = c("MERVLKD_8C_rep1.count", "MERVLKD_8C_rep2.count"),
                                  FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                  log2FC = log2(2), FDR = 0.05)

i2C_repeats.group <- classifyDEG(i2C_repeats,
                                  ctr.rpkm = c("GFPi_2C_rep1.count", "GFPi_2C_rep2.count"),
                                  trt.rpkm = c("MERVLi_2C_rep1.count", "MERVLi_2C_rep2.count"),
                                  FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                  log2FC = log2(2), FDR = 0.05)
  
i4C_repeats.group <- classifyDEG(i4C_repeats,
                                  ctr.rpkm = c("GFPi_4C_rep1.count", "GFPi_4C_rep2.count"),
                                  trt.rpkm = c("MERVLi_4C_rep1.count", "MERVLi_4C_rep2.count"),
                                  FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                  log2FC = log2(2), FDR = 0.05)
  
i8C_repeats.group <- classifyDEG(i8C_repeats,
                                  ctr.rpkm = c("GFPi_8C_rep1.count", "GFPi_8C_rep2.count"),
                                  trt.rpkm = c("MERVLi_8C_rep1.count", "MERVLi_8C_rep2.count"),
                                  FDR.col = "padj", log2FC.col = "log2FoldChange", RPKM = 0,
                                  log2FC = log2(2), FDR = 0.05)

KD2C_repeats <- cbind(KD2C_repeats, KD2C_repeats.group)
KD4C_repeats <- cbind(KD4C_repeats, KD4C_repeats.group)
KD8C_repeats <- cbind(KD8C_repeats, KD8C_repeats.group)
i2C_repeats  <- cbind(i2C_repeats, i2C_repeats.group)
i4C_repeats  <- cbind(i4C_repeats, i4C_repeats.group)
i8C_repeats  <- cbind(i8C_repeats, i8C_repeats.group)
```

```{r, fig.height = 3, fig.width = 6}
c2_MERVL_KD <- KD2C_repeats[which(KD2C_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR", 
                                                      "MT2C_Mm:ERVL:LTR")), 
                        c("Ctrl_2C_rep1.count", "Ctrl_2C_rep2.count", 
                          "MERVLKD_2C_rep1.count", "MERVLKD_2C_rep2.count")]
c2_MERVLi <- i2C_repeats[which(i2C_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR", 
                                                      "MT2C_Mm:ERVL:LTR")),
                         c("GFPi_2C_rep1.count", "GFPi_2C_rep2.count", 
                          "MERVLi_2C_rep1.count", "MERVLi_2C_rep2.count")]

c2_MERVL_KD <- t(c2_MERVL_KD)
c2_MERVLi <- t(c2_MERVLi)

c2_MERVL_KD <- cbind(c2_MERVL_KD,
                  group = c(rep("CTR", 2), rep("KD", 2)),
                  stage = "2C")
c2_MERVLi <- cbind(c2_MERVLi,
                   group = c(rep("GFPi", 2), rep("MERVLi", 2)),
                   stage = "2C")

df4fig_MERVLKD <- as.data.frame(c2_MERVL_KD)
df4fig_MERVLKD$`MERVL-int:ERVL:LTR` <- as.numeric(df4fig_MERVLKD$`MERVL-int:ERVL:LTR`)
df4fig_MERVLKD$`MT2_Mm:ERVL:LTR` <- as.numeric(df4fig_MERVLKD$`MT2_Mm:ERVL:LTR`)
df4fig_MERVLKD$`MT2C_Mm:ERVL:LTR` <- as.numeric(df4fig_MERVLKD$`MT2C_Mm:ERVL:LTR`)

df4fig_MERVLi<- as.data.frame(c2_MERVLi)
df4fig_MERVLi$`MERVL-int:ERVL:LTR` <- as.numeric(df4fig_MERVLi$`MERVL-int:ERVL:LTR`)
df4fig_MERVLi$`MT2_Mm:ERVL:LTR` <- as.numeric(df4fig_MERVLi$`MT2_Mm:ERVL:LTR`)
df4fig_MERVLi$`MT2C_Mm:ERVL:LTR` <- as.numeric(df4fig_MERVLi$`MT2C_Mm:ERVL:LTR`)

fig1 <- ggplot(df4fig_MERVLKD, aes(x = stage, y = `MT2_Mm:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("CTR", "MERVL_KD"),
                    values = c("grey80", "white")) +
  #scale_x_discrete(limits = c("e2C", "2C")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MT2_Mm (MERVLKD)")
#fig1

fig1.1 <- ggplot(df4fig_MERVLi, aes(x = stage, y = `MT2_Mm:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("GFPi", "MERVLi"),
                    values = c("grey80", "white")) +
  #scale_x_discrete(limits = c("e2C", "2C")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MT2_Mm (MERVLi)")
#fig1.1

fig2 <- ggplot(df4fig_MERVLKD, aes(x = stage, y = `MERVL-int:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("CTR", "MERVL_KD"),
                    values = c("grey80", "white")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  #scale_x_discrete(limits = c("e2C", "2C")) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MERVL-int (MERVLKD)")
#fig2

fig2.2 <- ggplot(df4fig_MERVLi, aes(x = stage, y = `MERVL-int:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("GFPi", "MERVLi"),
                    values = c("grey80", "white")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  #scale_x_discrete(limits = c("e2C", "2C")) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MERVL-int (MERVLi)")
#fig2.2

fig3 <- ggplot(df4fig_MERVLKD, aes(x = stage, y = `MT2C_Mm:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("CTR", "MERVL_KD"),
                    values = c("grey80", "white")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  #scale_x_discrete(limits = c("e2C", "2C")) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MT2C_Mm (MERVL KD)")
#fig3

fig3.3 <- ggplot(df4fig_MERVLKD, aes(x = stage, y = `MT2C_Mm:ERVL:LTR`, fill = group)) + 
  ylab("Normalized counts") +
  geom_bar(position = "dodge", stat = 'summary', fun.y = 'mean',color = "black", width = 0.75) +
  scale_fill_manual(breaks = c("GFPi", "MERVLi"),
                    values = c("grey80", "white")) +
  geom_point(position=position_jitterdodge(jitter.width = 0.1)) + 
  theme_cowplot(font_size = 15) +
  #scale_x_discrete(limits = c("e2C", "2C")) +
  theme(axis.title.x=element_blank(),
        legend.title=element_blank()) +
  theme(legend.position= "top") +
  ggtitle("MT2C_Mm (MERVLi)")
#fig3.3

pdf(file = "./figures/231115_MERVL_KD_i_barplot_2C.pdf",
    width = 2, height = 4)
fig1
fig1.1
fig2
fig2.2
fig3
fig3.3
dev.off()
plot_grid(fig1, fig1.1, fig2, fig2.2, fig3, fig3.3, nrow = 2)
```

### 7.1.1 Repeat deeptool heatmap
```{bash}
module load bedtools/2.30.0
source ~/.bash_profile
conda activate deeptools


#Region of interest------------------------------
Region_dir="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/output"
FL_MERVL="${Region_dir}/full_length_MERVL.bed"
FL_MERVL_3primeMT2="${Region_dir}/full_length_MERVL_downMT2.bed"
FL_MERVL_5primeMT2="${Region_dir}/full_length_MERVL_upMT2.bed"
soloMT2="${Region_dir}/soloMT2.bed"
otherMT2="${Region_dir}/othersMT2.bed"

grep "MT2_Mm" $FL_MERVL_3primeMT2 > FL_MERVL_3primeMT2_Mm.bed
grep "MT2_Mm" $FL_MERVL_5primeMT2 > FL_MERVL_5primeMT2_Mm.bed
grep "MT2_Mm" $soloMT2 > soloMT2_Mm.bed
grep "MT2_Mm" $otherMT2 > otherMT2_Mm.bed

#total RNAseq------------------------------------
totalRNA_dir1="/data/ZYChenlab/Zhiyuan/projects/TE_project/230705_MT2Mmi_totalRNAseq/bigwigs"
totalRNA_dir2="/data/ZYChenlab/Zhiyuan/projects/TE_project/230809_MT2Mmi_totalRNA_e2C/bigwigs"
sakashita_ASO_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/230308_reanalyses_of_Siomi_data/KD/bigwigs"
sakashita_CRISPRi_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/230308_reanalyses_of_Siomi_data/CRIPSRi/bigwigs"

CTRi_2c="${totalRNA_dir1}/CTRi_2C_merged.sorted.bw"
MT2i_2c="${totalRNA_dir1}/MT2Mmi_2C_merged.sorted.bw"
CTRi_e2c="${totalRNA_dir2}/CTRi_e2C_merged.sorted.bw"
MT2i_e2c="${totalRNA_dir2}/MT2Mmi_e2C_merged.sorted.bw"
ASO_ctr1="${sakashita_ASO_dir}/Ctrl_2C_rep1Aligned.out.sorted_uniq.bw"
ASO_KD1="${sakashita_ASO_dir}/MERVLKD_2C_rep1Aligned.out.sorted_uniq.bw"
ASO_ctr2="${sakashita_ASO_dir}/Ctrl_2C_rep2Aligned.out.sorted_uniq.bw"
ASO_KD2="${sakashita_ASO_dir}/MERVLKD_2C_rep2Aligned.out.sorted_uniq.bw"
CRISPRi_ctr1="${sakashita_CRISPRi_dir}/GFPi_2C_rep1Aligned.out.sorted_uniq.bw"
CRISPRi_KD1="${sakashita_CRISPRi_dir}/MERVLi_2C_rep1Aligned.out.sorted_uniq.bw"
CRISPRi_ctr2="${sakashita_CRISPRi_dir}/GFPi_2C_rep2Aligned.out.sorted_uniq.bw"
CRISPRi_KD2="${sakashita_CRISPRi_dir}/MERVLi_2C_rep2Aligned.out.sorted_uniq.bw"

#computeMatrix--------------------
computeMatrix scale-regions \
		-S $CTRi_2c $MT2i_2c $ASO_ctr1 $ASO_KD1 $CRISPRi_ctr1  $CRISPRi_KD1 \
		-R FL_MERVL_5primeMT2_Mm.bed FL_MERVL_3primeMT2_Mm.bed soloMT2_Mm.bed otherMT2_Mm.bed \
		-a 1000 -b 1000 \
		--regionBodyLength 500 \
		--outFileName MT2_Mm.1000flank_mat.gz \
		--sortRegions 'descend' \
		--samplesLabel 'CTRi_2c' 'MT2i_2c' 'ASO_CTR1' 'ASO_KD1' 'CRISPRi_c1' 'CRISPRi_KD1'  \
		--outFileSortedRegions MT2_Mm.1000flank_mat_sorted.bed \
		--outFileNameMatrix MT2_Mm.1000flank_mat_sorted.matrix \
		--missingDataAsZero \
		--sortUsingSamples 3

#plotHeatmap---------------------
plotHeatmap --matrixFile MT2_Mm.1000flank_mat.gz \
		    --outFileName MT2_Mm.1000flank.pdf \
		    --dpi 300 \
	    	    --sortRegions "descend" \
	  	    --colorMap 'GnBu' 'GnBu'  'GnBu' 'GnBu' 'GnBu' 'GnBu'\
		    --boxAroundHeatmaps no \
		    --samplesLabel  'CTRi_2C' 'MT2i_2c' 'ASO_CTR1' 'ASO_KD1' 'CRISPRi_c1' 'CRISPRi_KD1' \
	            --legendLocation "lower-center" \
		    --sortUsingSamples 3 \
		    --zMax 50 50 50 50 50 50 
```

## 7.2 Sakashita differential gene
```{r}
KD2C_DESeq_rpkm <- merge(KD2C_DESeq[, c("id", "baseMean", 
                                        "log2FoldChange", "padj", 
                                        "Ctrl_2C_rep1.count", "Ctrl_2C_rep2.count",
                                        "MERVLKD_2C_rep1.count", "MERVLKD_2C_rep2.count")],
                         rpkm[, c("id", "name", 
                                  "Ctrl_2C_rep1.rpkm", "Ctrl_2C_rep2.rpkm",
                                  "MERVLKD_2C_rep1.rpkm", "MERVLKD_2C_rep2.rpkm")])

KD4C_DESeq_rpkm <- merge(KD4C_DESeq[, c("id", "baseMean", 
                                        "log2FoldChange", "padj", 
                                        "Ctrl_4C_rep1.count", "Ctrl_4C_rep2.count",
                                        "MERVLKD_4C_rep1.count", "MERVLKD_4C_rep2.count")],
                         rpkm[, c("id", "name", 
                                  "Ctrl_4C_rep1.rpkm", "Ctrl_4C_rep2.rpkm",
                                  "MERVLKD_4C_rep1.rpkm", "MERVLKD_4C_rep2.rpkm")])

KD8C_DESeq_rpkm <- merge(KD8C_DESeq[, c("id", "baseMean", 
                                        "log2FoldChange", "padj", 
                                        "Ctrl_8C_rep1.count", "Ctrl_8C_rep2.count",
                                        "MERVLKD_8C_rep1.count", "MERVLKD_8C_rep2.count")],
                         rpkm[, c("id", "name", 
                                  "Ctrl_8C_rep1.rpkm", "Ctrl_8C_rep2.rpkm",
                                  "MERVLKD_8C_rep1.rpkm", "MERVLKD_8C_rep2.rpkm")])

i2C_DESeq_rpkm <- merge(i2C_DESeq[, c("id", "baseMean",
                                      "log2FoldChange", "padj", 
                                      "GFPi_2C_rep1.count", "GFPi_2C_rep2.count",
                                      "MERVLi_2C_rep1.count", "MERVLi_2C_rep2.count")],
                        rpkm[, c("id", "name",
                                 "GFPi_2C_rep1.rpkm", "GFPi_2C_rep2.rpkm",
                                 "MERVLi_2C_rep1.rpkm", "MERVLi_2C_rep2.rpkm")])

i4C_DESeq_rpkm <- merge(i4C_DESeq[, c("id", "baseMean",
                                      "log2FoldChange", "padj", 
                                      "GFPi_4C_rep1.count", "GFPi_4C_rep2.count",
                                      "MERVLi_4C_rep1.count", "MERVLi_4C_rep2.count")],
                        rpkm[, c("id", "name",
                                 "GFPi_4C_rep1.rpkm", "GFPi_4C_rep2.rpkm",
                                 "MERVLi_4C_rep1.rpkm", "MERVLi_4C_rep2.rpkm")])

i8C_DESeq_rpkm <- merge(i8C_DESeq[, c("id", "baseMean",
                                      "log2FoldChange", "padj", 
                                      "GFPi_8C_rep1.count", "GFPi_8C_rep2.count",
                                      "MERVLi_8C_rep1.count", "MERVLi_8C_rep2.count")],
                        rpkm[, c("id", "name",
                                 "GFPi_8C_rep1.rpkm", "GFPi_8C_rep2.rpkm",
                                 "MERVLi_8C_rep1.rpkm", "MERVLi_8C_rep2.rpkm")])

KD2C_DESeq_rpkm$padj[is.na(KD2C_DESeq_rpkm$padj)] <- 1
KD4C_DESeq_rpkm$padj[is.na(KD4C_DESeq_rpkm$padj)] <- 1
KD8C_DESeq_rpkm$padj[is.na(KD8C_DESeq_rpkm$padj)] <- 1
i2C_DESeq_rpkm$padj[is.na(i2C_DESeq_rpkm$padj)] <- 1
i4C_DESeq_rpkm$padj[is.na(i4C_DESeq_rpkm$padj)] <- 1
i8C_DESeq_rpkm$padj[is.na(i8C_DESeq_rpkm$padj)] <- 1

KD2C_DESeq_rpkm.group <- classifyDEG(
                            KD2C_DESeq_rpkm,
                            ctr.rpkm = c("Ctrl_2C_rep1.rpkm", "Ctrl_2C_rep2.rpkm"),
                            trt.rpkm = c("MERVLKD_2C_rep1.rpkm", "MERVLKD_2C_rep2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange", 
                            RPKM = 0, log2FC = log2(2), FDR = 0.05)
KD4C_DESeq_rpkm.group <- classifyDEG(
                            KD4C_DESeq_rpkm,
                            ctr.rpkm = c("Ctrl_4C_rep1.rpkm", "Ctrl_4C_rep2.rpkm"),
                            trt.rpkm = c("MERVLKD_4C_rep1.rpkm", "MERVLKD_4C_rep2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange", 
                            RPKM = 0, log2FC = log2(2), FDR = 0.05)
KD8C_DESeq_rpkm.group <- classifyDEG(
                            KD8C_DESeq_rpkm,
                            ctr.rpkm = c("Ctrl_8C_rep1.rpkm", "Ctrl_8C_rep2.rpkm"),
                            trt.rpkm = c("MERVLKD_8C_rep1.rpkm", "MERVLKD_8C_rep2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange", 
                            RPKM = 0, log2FC = log2(2), FDR = 0.05)

i2C_DESeq_rpkm.group <- classifyDEG(
                            i2C_DESeq_rpkm,
                            ctr.rpkm = c("GFPi_2C_rep1.rpkm", "GFPi_2C_rep2.rpkm"),
                            trt.rpkm = c("MERVLi_2C_rep1.rpkm", "MERVLi_2C_rep2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange", 
                            RPKM = 0, log2FC = log2(2), FDR = 0.05)

i4C_DESeq_rpkm.group <- classifyDEG(
                            i4C_DESeq_rpkm,
                            ctr.rpkm = c("GFPi_4C_rep1.rpkm", "GFPi_4C_rep1.rpkm"),
                            trt.rpkm = c("MERVLi_4C_rep1.rpkm", "MERVLi_4C_rep2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange", 
                            RPKM = 0, log2FC = log2(2), FDR = 0.05)

i8C_DESeq_rpkm.group <- classifyDEG(
                            i8C_DESeq_rpkm,
                            ctr.rpkm = c("GFPi_8C_rep1.rpkm", "GFPi_8C_rep1.rpkm"),
                            trt.rpkm = c("MERVLi_8C_rep1.rpkm", "MERVLi_8C_rep2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange", 
                            RPKM = 0, log2FC = log2(2), FDR = 0.05)


KD2C_DESeq_rpkm <- cbind(KD2C_DESeq_rpkm, KD2C_DESeq_rpkm.group)
KD4C_DESeq_rpkm <- cbind(KD4C_DESeq_rpkm, KD4C_DESeq_rpkm.group)
KD8C_DESeq_rpkm <- cbind(KD8C_DESeq_rpkm, KD8C_DESeq_rpkm.group)
i2C_DESeq_rpkm <- cbind(i2C_DESeq_rpkm, i2C_DESeq_rpkm.group)
i4C_DESeq_rpkm <- cbind(i4C_DESeq_rpkm, i4C_DESeq_rpkm.group)
i8C_DESeq_rpkm <- cbind(i8C_DESeq_rpkm, i8C_DESeq_rpkm.group)

KD2C_DESeq_rpkm$log2baseMean <- log2(KD2C_DESeq_rpkm$baseMean + 1)
KD4C_DESeq_rpkm$log2baseMean <- log2(KD4C_DESeq_rpkm$baseMean + 1)
KD8C_DESeq_rpkm$log2baseMean <- log2(KD8C_DESeq_rpkm$baseMean + 1)
i2C_DESeq_rpkm$log2baseMean <- log2(i2C_DESeq_rpkm$baseMean + 1)
i4C_DESeq_rpkm$log2baseMean <- log2(i4C_DESeq_rpkm$baseMean + 1)
i8C_DESeq_rpkm$log2baseMean <- log2(i8C_DESeq_rpkm$baseMean + 1)

write.csv(KD2C_repeats, "./R_output/Sakashita_output/231115_MERVLKD_2C_repeats_DE_analyses.csv", 
          quote = F, row.names = F)
write.csv(KD4C_repeats, "./R_output/Sakashita_output/231115_MERVLKD_4C_repeats_DE_analyses.csv",
          quote = F, row.names = F)
write.csv(KD8C_repeats, "./R_output/Sakashita_output/231115_MERVLKD_8C_repeats_DE_analyses.csv",
          quote = F, row.names = F)
write.csv(i2C_repeats, "./R_output/Sakashita_output/231115_MERVLi_2C_repeats_DE_analyses.csv",
          quote = F, row.names = F)
write.csv(i4C_repeats, "./R_output/Sakashita_output/231115_MERVLi_4C_repeats_DE_analyses.csv",
          quote = F, row.names = F)
write.csv(i8C_repeats, "./R_output/Sakashita_output/231115_MERVLi_8C_repeats_DE_analyses.csv",
          quote = F, row.names = F)

write.csv(KD2C_DESeq_rpkm, "./R_output/Sakashita_output/231115_MERVLKD_2C_genes_DE_analyses.csv", 
          quote = F, row.names = F)
write.csv(KD4C_DESeq_rpkm, "./R_output/Sakashita_output/231115_MERVLKD_4C_genes_DE_analyses.csv", 
          quote = F, row.names = F)
write.csv(KD8C_DESeq_rpkm, "./R_output/Sakashita_output/231115_MERVLKD_8C_genes_DE_analyses.csv", 
          quote = F, row.names = F)
write.csv(i2C_DESeq_rpkm, "./R_output/Sakashita_output/231115_MERVLi_2C_genes_DE_analyses.csv", 
          quote = F, row.names = F)
write.csv(i4C_DESeq_rpkm, "./R_output/Sakashita_output/231115_MERVLi_4C_genes_DE_analyses.csv", 
          quote = F, row.names = F)
write.csv(i8C_DESeq_rpkm, "./R_output/Sakashita_output/231115_MERVLi_8C_genes_DE_analyses.csv", 
          quote = F, row.names = F)
```

## 7.3 Sakashita DEG characterization

### 7.3.1 ZGA/maternal gene upon MERVL KD/CRISPRi
```{r}
MERVLi_2c <- read.table("./R_output/Sakashita_output/231115_MERVLi_2C_genes_DE_analyses.csv", 
                      header = T, sep = ",")
MERVLi_4c <- read.table("./R_output/Sakashita_output/231115_MERVLi_4C_genes_DE_analyses.csv",
                        header = T, sep = ",")
MERVLi_8c <- read.table("./R_output/Sakashita_output/231115_MERVLi_8C_genes_DE_analyses.csv",
                        header = T, sep = ",")
MERVLKD_2c <- read.table("./R_output/Sakashita_output/231115_MERVLKD_2C_genes_DE_analyses.csv", 
                      header = T, sep = ",")
MERVLKD_4c <- read.table("./R_output/Sakashita_output/231115_MERVLKD_4C_genes_DE_analyses.csv", 
                      header = T, sep = ",")
MERVLKD_8c <- read.table("./R_output/Sakashita_output/231115_MERVLKD_8C_genes_DE_analyses.csv", 
                      header = T, sep = ",")

table(MERVLi_2c$i2C_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#            81          26060             95
table(MERVLKD_2c$KD2C_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
 #          359          22024            545 

table(MERVLi_4c$i4C_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#          204          24756             50 

table(MERVLi_8c$i8C_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#           85          23762             70 

table(MERVLKD_4c$KD4C_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#           339          20667            455 

table(MERVLKD_8c$KD8C_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#          176          20105            102
```

```{r}
suppressMessages(library(dplyr))
majorZGA <- read.table("./R_output/zhang_wang_RNA/231115_zhang_totalRNA_majorZGA.csv", sep = ",", header = T)

#for later scatter plot
MERVLi_2c$logGFP <- log2((MERVLi_2c$GFPi_2C_rep1.count + MERVLi_2c$GFPi_2C_rep2.count)/2 + 1)
MERVLi_2c$logMEVLi <- log2((MERVLi_2c$MERVLi_2C_rep1.count + MERVLi_2c$MERVLi_2C_rep2.count)/2 + 1)

MERVLKD_2c$logCTR <- log2((MERVLKD_2c$Ctrl_2C_rep1.count + MERVLKD_2c$Ctrl_2C_rep2.count)/2 + 1)
MERVLKD_2c$logKD <- log2((MERVLKD_2c$MERVLKD_2C_rep1.count + MERVLKD_2c$MERVLKD_2C_rep2.count)/2 + 1)

nrow(majorZGA) #1989

majorZGA_MERVLi <- left_join(majorZGA, MERVLi_2c, by = "id")
nrow(majorZGA_MERVLi) #1989
table(majorZGA_MERVLi$i2C_DESeq_rpkm.group) 
#down-regulated  similar_level   up-regulated 
#           31           1937              4 

majorZGA_MERVLKD <- left_join(majorZGA, MERVLKD_2c, by = "id")
nrow(majorZGA_MERVLKD) #1989
table(majorZGA_MERVLKD$KD2C_DESeq_rpkm.group) 
#down-regulated  similar_level   up-regulated 
#           100           1827             36
# write.table(majorZGA_MERVLKD, "../R_output/231106_majorZGA_2C_MERVLKD.csv", sep = ",",
#             quote = F, row.names = F)


pdf("./figures/231115_Sakashita_major_ZGA_scatterplot.pdf", 
    width = 6, height = 5)
foldchange = 2
source("./utils.R")
ggScatterplot(majorZGA_MERVLKD, x = "logCTR", y = "logKD",
                       group = "KD2C_DESeq_rpkm.group", gene = "name.y", xlab = "CTR",
                       ylab = "ASO KD",
                       my.color=c("blue", "grey50",  "red3"),
                       title = paste0("major ZGA (l2C):", nrow(majorZGA)),
                       label.up = paste0(nrow(majorZGA_MERVLKD[which(majorZGA_MERVLKD$KD2C_DESeq_rpkm.group == "up-regulated"),])),
                       label.down = paste0(nrow(majorZGA_MERVLKD[which(majorZGA_MERVLKD$KD2C_DESeq_rpkm.group == "down-regulated"),])),
                       #genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange)

dev.off()
ggScatterplot(majorZGA_MERVLKD, x = "logCTR", y = "logKD",
                       group = "KD2C_DESeq_rpkm.group", gene = "name.y", xlab = "CTR",
                       ylab = "ASO KD",
                       my.color=c("blue", "grey50",  "red3"),
                       title = paste0("major ZGA (l2C):", nrow(majorZGA)),
                       label.up = paste0(nrow(majorZGA_MERVLKD[which(majorZGA_MERVLKD$KD2C_DESeq_rpkm.group == "up-regulated"),])),
                       label.down = paste0(nrow(majorZGA_MERVLKD[which(majorZGA_MERVLKD$KD2C_DESeq_rpkm.group == "down-regulated"),])),
                       #genes4Label = c("Zscan4c", "Zfp352", "Fundc1", "Ddit4l", "Obox3", "Obox6"),
                       FC.line = foldchange)
```

### 7.3.2 DEG distance to MERVL

Identify MERVL (<5kb & >500bp) but with two flanking MT2
```{perl}
#!/usr/bin/perl

#FOR THISONE: FOCUS THOSE <5kb but with 2 flanking MT2
my @LTR = ("MT2_Mm", "MT2C_Mm");
my @INSERT = ("MERVL-int", "MERVL_2A-int");
my $insert_length = 5000; #5kb
my $full_length = 10000; #10kb

my $TE_BED = "/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed";
my $chr_Length = "/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/chrNameLength.txt";
system("sort -k1,1 -k2,2n $chr_Length > ${chr_Length}_sorted.txt");

#step1 : retrieve ERVL or MERVL_2A-int < 5kb
foreach(@INSERT){
        my $tmp_insert = $_;
        system("grep $tmp_insert $TE_BED > mm10_rmsk_${tmp_insert}.bed");
        open (BED, "mm10_rmsk_${tmp_insert}.bed") or die$!;
        open (OUT, ">mm10_rmsk_${tmp_insert}_lessthan${insert_length}bp.bed") or die$!;
        while (my $line = <BED>){
                chomp $line;
                my @split = split(/\s+/, $line);
                if($split[2] - $split[1] < $insert_length & $split[2] - $split[1] > 500){
                        print OUT "$line\n";
                }
        }
        close BED;
        close OUT;
}

#combine MERVL and MERVL_2A-int
foreach(@INSERT){
        my $tmp_insert = $_;
        if (-e "mm10_rmsk_MERVL.bed"){
                system("cat mm10_rmsk_${tmp_insert}_lessthan${insert_length}bp.bed >> mm10_rmsk_MERVL.bed");    
        } else{
                system("cat mm10_rmsk_${tmp_insert}_lessthan${insert_length}bp.bed > mm10_rmsk_MERVL.bed")
        }
}
system("sort -k1,1 -k2,2n mm10_rmsk_MERVL.bed > mm10_rmsk_MERVL_sorted.bed");

#step2 : find nearest MT2_Mm of MERVL or MERVL_2A-int 
foreach(@LTR){
        my $tmp_insert = $_;
        if(-e "mm10_rmsk_MT2.bed"){
                system("grep $tmp_insert $TE_BED >> mm10_rmsk_MT2.bed");
        } else{
                system("grep $tmp_insert $TE_BED > mm10_rmsk_MT2.bed");
        }               
}
system("sort -k1,1 -k2,2n mm10_rmsk_MT2.bed > mm10_rmsk_MT2_sorted.bed");

#only downstream LTR with the same strand
system("bedtools closest -s -iu -D a -g ${chr_Length}_sorted.txt -a mm10_rmsk_MERVL_sorted.bed -b mm10_rmsk_MT2_sorted.bed > MERVL_MT2_bedclosest.txt");

#only upstream LTR with the same strand
system("bedtools closest -s -id -D a -g ${chr_Length}_sorted.txt -a mm10_rmsk_MERVL_sorted.bed -b mm10_rmsk_MT2_sorted.bed > MT2_MERVL_bedclosest.txt");        

#step3: require the distance of these two MT2s <10kb 
system("Rscript filter_nonFL_MERVL.R");

#clean out intermediate files
system("mv nonFL_MERVL* output/");
```

```{r filter_nonFL_MERVL.R}
#!/usr/bin/Rscript
#MERVL with MT2 downstream
MERVL_MT2 <- read.table("MERVL_MT2_bedclosest.txt", sep = "\t")
colnames(MERVL_MT2) <- c("MERVL_chr", "MERVL_start", "MERVL_end",
                         "MERVL", "MERVL_score", "MERVL_strand",
                         "downstream_MT2_chr", "downstream_MT2_start",
                         "downstream_MT2_end",
                         "downstream_MT2", "downstream_MT2_score",
                         "downstream_MT2_strand",
                         "MERVL_MT2_distance")
#MERVL with MT2 upstream
MT2_MERVL <- read.table("MT2_MERVL_bedclosest.txt", sep = "\t")
colnames(MT2_MERVL) <- c("MERVL_chr", "MERVL_start", "MERVL_end",
                         "MERVL", "MERVL_score", "MERVL_strand",
                         "upstream_MT2_chr", "upstream_MT2_start",
                         "upstream_MT2_end",
                         "upstream_MT2", "upstream_MT2_score",
                         "upstream_MT2_strand",
                         "MT2_MERVL_distance")
MERVL <- merge(MT2_MERVL, MERVL_MT2[, c("MERVL",
                                        "downstream_MT2_chr", "downstream_MT2_start",
                                        "downstream_MT2_end",
                                        "downstream_MT2", "downstream_MT2_score",
                                        "downstream_MT2_strand",
                                        "MERVL_MT2_distance")],
               by = "MERVL")


MERVL$length <- c()
for(i in 1:nrow(MERVL)){
  if(MERVL$MERVL_strand[i] == "+"){
    MERVL$length[i] <- MERVL$downstream_MT2_end[i] - MERVL$upstream_MT2_start[i]
  }else{
    MERVL$length[i] <- MERVL$upstream_MT2_end[i] - MERVL$downstream_MT2_start[i]
  }
}

MERVL$insert_length <- MERVL$MERVL_end - MERVL$MERVL_start
MERVL <- MERVL[which(MERVL$length < 10000 & MERVL$length >0 ),]
MERVL <- MERVL[!duplicated(MERVL$MERVL),]
nrow(MERVL) #251
write.table(MERVL[, c("MERVL_chr", "MERVL_start", "MERVL_end", "MERVL",
                                  "MERVL_score", "MERVL_strand")],
            "nonFL_MERVL.bed", sep = "\t", col.names = F, quote = F, row.names =F)
write.table(MERVL[, c("upstream_MT2_chr", "upstream_MT2_start",
                                  "upstream_MT2_end", "upstream_MT2", "upstream_MT2_score",
                                  "upstream_MT2_strand")],
            "nonFL_MERVL_upMT2.bed", sep = "\t", col.names = F, quote = F, row.names = F)
write.table(MERVL[, c("downstream_MT2_chr", "downstream_MT2_start",
                                  "downstream_MT2_end", "downstream_MT2", "downstream_MT2_score",
                                  "downstream_MT2_strand")],
            "nonFL_MERVL_downMT2.bed", sep = "\t", col.names = F, quote = F, row.names = F)
```

```{bash}
#!/usr/bin/bash

module load bedtools/2.27.0

cat full_length_MERVL.bed nonFL_MERVL.bed > all_MERVL.bed
GTF="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/gencode.vM25.annotation.gtf"

awk 'OFS="\t" {if ($3=="gene") {print $1,$4-1,$5,$10,$14,$7}}' $GTF | tr -d '";' > mm10_gene.bed
sort -k1,1 -k2,2n mm10_gene.bed > mm10_gene_sorted.bed
sort -k1,1 -k2,2n all_MERVL.bed > all_MERVL_sorted.bed

grep "chr" all_MERVL_sorted.bed > all_MERVL_sorted_standard_chr.bed
bedtools closest -a mm10_gene_sorted.bed -b all_MERVL_sorted_standard_chr.bed -D "b" | uniq > 231106_nearest_MERVL2gene.txt
```

```{r}
gene_MERVL_distance <- read.table("./R_input/dCas9_ChIPseq/231106_nearest_MERVL2gene.txt",
                                         header = F, sep = "\t")
nrow(MERVLi_2c[which(MERVLi_2c$i2C_DESeq_rpkm.group == "down-regulated"),]) #81  
nrow(MERVLKD_2c[which(MERVLKD_2c$KD2C_DESeq_rpkm.group == "down-regulated"),]) #359

nrow(gene_MERVL_distance) #55447
colnames(gene_MERVL_distance) <- c("gene_chr", "gene_start", "gene_end", 
                                   "id", "name", "gene_strand", 
                                   "peak_chr", "peak_start", "peak_end", 
                                   "peak_name", "peak_score", "peak_strand",
                                   "distance")
#for genes with multiple peaks in the gene bodies, randomly pick one
gene_MERVL_distance <- gene_MERVL_distance[!duplicated(gene_MERVL_distance$id),]
nrow(gene_MERVL_distance) #55401

source("./utils.R")

MERVLi_RNA_distance <- merge(MERVLi_2c, gene_MERVL_distance[, c("id", "distance")])
MERVLKD_RNA_distance <- merge(MERVLKD_2c, gene_MERVL_distance[, c("id", "distance")])

MERVLi_RNA_distance$bin <- classify_distance(as.numeric(MERVLi_RNA_distance$distance), ignore_direction = T)
MERVLKD_RNA_distance$bin <- classify_distance(as.numeric(MERVLKD_RNA_distance$distance), ignore_direction = T)

MERVLi_RNA_distance.up <- MERVLi_RNA_distance[which(MERVLi_RNA_distance$i2C_DESeq_rpkm.group == "up-regulated"),]
MERVLi_RNA_distance.down <- MERVLi_RNA_distance[which(MERVLi_RNA_distance$i2C_DESeq_rpkm.group == "down-regulated"),]

MERVLKD_RNA_distance.up <- MERVLKD_RNA_distance[which(MERVLKD_RNA_distance$KD2C_DESeq_rpkm.group == "up-regulated"),]
MERVLKD_RNA_distance.down <- MERVLKD_RNA_distance[which(MERVLKD_RNA_distance$KD2C_DESeq_rpkm.group == "down-regulated"),]


table(MERVLi_RNA_distance.up$bin)
#>50kb 10-30kb 30-50kb 
#     90       3       2 
table(MERVLi_RNA_distance.down$bin)
#>50kb  0-10kb 10-30kb 30-50kb 
#    26      44       9       2

table(MERVLKD_RNA_distance.up$bin)
#>50kb  0-10kb 10-30kb 30-50kb 
#    515       8       8      14 
table(MERVLKD_RNA_distance.down$bin)
#>50kb  0-10kb 10-30kb 30-50kb 
#    258      77      14      10 

pdf("./figures/231115_Sakashita_genes_DE_byDisatance_pieChart_FC2.pdf", width = 5, height = 5)
pie(table(MERVLi_RNA_distance.up$bin), main = paste0("MERVLi_UP: up-regulated genes:", nrow(MERVLi_RNA_distance.up)))
pie(table(MERVLi_RNA_distance.down$bin), main = paste0("MERVLi_down: down-regulated genes:", nrow(MERVLi_RNA_distance.down)))
pie(table(MERVLKD_RNA_distance.up$bin), main = paste0("MERVL_KD_up: up-regulated genes:", nrow(MERVLKD_RNA_distance.up)))
pie(table(MERVLKD_RNA_distance.down$bin), main = paste0("MERVL_KD_down: down-regulated genes:", nrow(MERVLKD_RNA_distance.down)))
dev.off()
#pie(table(MERVLi_RNA_distance.up$bin), main = paste0("MERVLi_UP: up-regulated genes:", nrow(MERVLi_RNA_distance.up)))
pie(table(MERVLi_RNA_distance.down$bin), main = paste0("MERVLi_down: down-regulated genes:", nrow(MERVLi_RNA_distance.down)))
#pie(table(MERVLKD_RNA_distance.up$bin), main = paste0("MERVL_KD_up: up-regulated genes:", nrow(MERVLKD_RNA_distance.up)))
pie(table(MERVLKD_RNA_distance.down$bin), main = paste0("MERVL_KD_down: down-regulated genes:", nrow(MERVLKD_RNA_distance.down)))
```

### 7.3.3 Sashita 2C down overlap with MT2i
```{r}
MT2i_2c <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_2C_genes_DE_analyses.csv", 
                      header = T, sep = ",")
MT2i_2c_up <- MT2i_2c[which(MT2i_2c$c2_DESeq_rpkm.group == "up-regulated"),]
MT2i_2c_down <- MT2i_2c[which(MT2i_2c$c2_DESeq_rpkm.group == "down-regulated"),]
nrow(MT2i_2c_up) #481
nrow(MT2i_2c_down) #1111

MERVLKD_2c_up <- MERVLKD_2c[which(MERVLKD_2c$KD2C_DESeq_rpkm.group == "up-regulated"),]
MERVLKD_2c_down <- MERVLKD_2c[which(MERVLKD_2c$KD2C_DESeq_rpkm.group == "down-regulated"),]
nrow(MERVLKD_2c_up) #545
nrow(MERVLKD_2c_down) #359

MERVLi_2c_up <- MERVLi_2c[which(MERVLi_2c$i2C_DESeq_rpkm.group == "up-regulated"),]
MERVLi_2c_down <- MERVLi_2c[which(MERVLi_2c$i2C_DESeq_rpkm.group == "down-regulated"),]
 
nrow(MERVLi_2c_up) #95
nrow(MERVLi_2c_down) #81

suppressMessages(library(VennDiagram))
suppressMessages(library(ggsci))

venn_2c_down <- venn.diagram(x = list(
                      "MERVLi_down" = unique(MERVLi_2c_down$id),
                      "MERVLKD_down" = unique(MERVLKD_2c_down$id),
                      "MT2i_down" = unique(MT2i_2c_down$id)
                      ),
                      filename = NULL,
                           main="2c_down",
                            resolution = 300,
                           #fill=ggsci::pal_futurama(3),
                           width = 800,height = 800,
                           euler.d = TRUE, cex=2,
                           cat.cex=0.8,scaled = TRUE,
                           offset=0.5, lwd=1,margin=0)
grid.newpage()
grid.draw(venn_2c_down)
```

```{r}
suppressMessages(library("eulerr"))

venn_2c_down2 <- euler(c("MERVLi" = 8,
                      "MERVLKD" = 202,
                      "MT2i" = 934,
                      "MERVLi&MERVLKD" = 2,
                      "MERVLi&MT2i" = 22,
                      "MT2i&MERVLKD" = 106,
                      "MERVLi&MERVLKD&MT2i"=49))
pdf("./figures/231115_Sakashita_MT2i_2C_venn.pdf", width = 5, height = 5)
plot(venn_2c_down2, key = TRUE, counts = TRUE)
grid.newpage()
grid.draw(venn_2c_down)
dev.off()
plot(venn_2c_down2, key = TRUE, counts = TRUE)
```

### 7.3.4 characterize overlap/specific gene set
For genes overlapped bewteen MT2i and MERVL_KD
identify following genes:
1) MT2i-specific
2) MERVLKD-specific
3) shared

To check their distance to nearest soloMT2 or MERVL_KD
if <30kb, considered to be associated with soloMT2 or MERVL

```{bash}
#!/usr/bin/bash

module load bedtools/2.27.0

GTF="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/gencode.vM25.annotation.gtf"

awk 'OFS="\t" {if ($3=="gene") {print $1,$4-1,$5,$10,$14,$7}}' $GTF | tr -d '";' > mm10_gene.bed
sort -k1,1 -k2,2n mm10_gene.bed > mm10_gene_sorted.bed
sort -k1,1 -k2,2n soloMT2.bed > soloMT2_sorted.bed
grep "chr" soloMT2_sorted.bed > soloMT2_sorted_standard_chr.bed
bedtools closest -a mm10_gene_sorted.bed -b soloMT2_sorted_standard_chr.bed -D "b" | uniq > 231108_nearest_soloMT2togene.txt
```

```{r}
gene_MERVL_distance <- read.table("./R_input/dCas9_ChIPseq/231106_nearest_MERVL2gene.txt",
                                         header = F, sep = "\t")
gene_soloMT2_distance <- read.table("./R_input/dCas9_ChIPseq/231108_nearest_soloMT2togene.txt", 
                                    header = F, sep = "\t")
nrow(gene_MERVL_distance) #55447 
nrow(gene_soloMT2_distance) #55541

colnames(gene_MERVL_distance) <- c("gene_chr", "gene_start", "gene_end", 
                                   "id", "name", "gene_strand", 
                                   "peak_chr", "peak_start", "peak_end", 
                                   "peak_name", "peak_score", "peak_strand",
                                   "distance_MERVL")
colnames(gene_soloMT2_distance) <- c("gene_chr", "gene_start", "gene_end", 
                                   "id", "name", "gene_strand", 
                                   "peak_chr", "peak_start", "peak_end", 
                                   "peak_name", "peak_score", "peak_strand",
                                   "distance_soloMT2")
#for genes with multiple peaks in the gene bodies, randomly pick one
gene_MERVL_distance <- gene_MERVL_distance[!duplicated(gene_MERVL_distance$id),]
nrow(gene_MERVL_distance) #55401
gene_soloMT2_distance <- gene_soloMT2_distance[!duplicated(gene_soloMT2_distance$id),]
nrow(gene_soloMT2_distance) #55401

gene_distance <- merge(gene_MERVL_distance, gene_soloMT2_distance, by = "id")
nrow(gene_distance) #55401

gene_classification <- function(df){
  df$group <- rep("others", nrow(df))
  distance_cutoff = 30000
  for(i in 1:nrow(df)){
    if(abs(df$distance_MERVL[i]) <= distance_cutoff & abs(df$distance_soloMT2[i]) <= distance_cutoff){
      if (abs(df$distance_MERVL[i]) <= abs(df$distance_soloMT2[i])){
          df$group[i] <- "MERVL-associated"
      } else {
          df$group[i] <- "soloMT2-associated"
      } 
    } else if(abs(df$distance_MERVL[i]) <= distance_cutoff & abs(df$distance_soloMT2[i]) > distance_cutoff){
      #associated with MERVL
      df$group[i] <- "MERVL-associated"
    } else if (abs(df$distance_MERVL[i]) > distance_cutoff & abs(df$distance_soloMT2[i]) <= distance_cutoff){
      #asscoated with soloMT2
      df$group[i] <- "soloMT2-associated"
    } else {
      df$group[i] <- "neither"
    }
  }
  return(df$group)
}
gene_distance$group <-  gene_classification(gene_distance)

table(gene_distance$group)
#MERVL-associated            neither soloMT2-associated 
#              1428              49675               4298 
```

```{r}
#genes shared by MT2i and MERVL KD
MT2i_MERVLi_2c_commmon_down <- MT2i_2c_down[which(MT2i_2c_down$id %in% MERVLi_2c_down$id),]
nrow(MT2i_MERVLi_2c_commmon_down) #71
MT2i_MERVLKD_2c_commmon_down <- MT2i_2c_down[which(MT2i_2c_down$id %in% MERVLKD_2c_down$id),]
nrow(MT2i_MERVLKD_2c_commmon_down) #155

MT2i_MERVLi_KD_2c_common_down <- rbind(MT2i_MERVLi_2c_commmon_down, MT2i_MERVLKD_2c_commmon_down)
MT2i_MERVLi_KD_2c_common_down <- MT2i_MERVLi_KD_2c_common_down[!duplicated(MT2i_MERVLi_KD_2c_common_down$id),]
nrow(MT2i_MERVLi_KD_2c_common_down) #177

MT2i_MERVLi_KD_2c_common_down <- left_join(MT2i_MERVLi_KD_2c_common_down, gene_distance, by = "id")
table(MT2i_MERVLi_KD_2c_common_down$group)
#MERVL-associated            neither soloMT2-associated 
#                92                 51                 34

MT2i_2c_unique_down <- MT2i_2c_down[!(MT2i_2c_down$id %in% MT2i_MERVLi_KD_2c_common_down$id),]
MT2i_2c_unique_down <- left_join(MT2i_2c_unique_down, gene_distance, by = "id")
nrow(MT2i_2c_unique_down) #934
table(MT2i_2c_unique_down$group) 
#MERVL-associated            neither soloMT2-associated 
#               165                415                354 

MERVLKD_2c_unique_down <- MERVLKD_2c_down[!(MERVLKD_2c_down$id %in% MT2i_MERVLi_KD_2c_common_down$id),]
nrow(MERVLKD_2c_unique_down)
#204
MERVLKD_2c_unique_down <- left_join(MERVLKD_2c_unique_down, gene_distance, by = "id")
table(MERVLKD_2c_unique_down$group)
#MERVL-associated            neither soloMT2-associated 
#                11                170                 23
df4fig <- data.frame(rbind(table(gene_distance$group), 
                     table(MT2i_MERVLi_KD_2c_common_down$group),
                     table(MT2i_2c_unique_down$group),
                     table(MERVLKD_2c_unique_down$group)))
df4fig$group <- c("all", "common", "MT2i_specific", "MERVLKD_specific")
#df4fig

suppressMessages(library(ggplot2))
suppressMessages(library(cowplot))
suppressMessages(library(tidyr))
suppressMessages(library(purrr))
suppressMessages(library(ggsci))

df4fig <- gather(df4fig, key = "classification", value = "count", -group)
df4fig <- df4fig %>% 
  group_by(group) %>%
  mutate(Freq = count / sum(count) * 100)

df4fig$group <- factor(df4fig$group, 
                       levels = c("all", "MT2i_specific", 
                                  "common", "MERVLKD_specific"))
df4fig$classification <- factor(df4fig$classification,
                                levels = c("neither",
                                           "MERVL.associated",
                                           "soloMT2.associated"
                                           ))
pdf("./figures/231115_overlap_soloMT2vsMERVL.pdf", width = 6, height = 6)
ggplot(df4fig, aes(x= group, y = Freq, fill = classification)) +
  geom_bar(stat = "identity", color = "black")  +
  scale_fill_manual(values = c("white", "grey", "black")) + theme_cowplot(16)
dev.off()
ggplot(df4fig, aes(x= group, y = Freq, fill = classification)) +
  geom_bar(stat = "identity", color = "black")  +
  scale_fill_manual(values = c("white", "grey", "black")) + theme_cowplot(16)
#run some stats here

#chi-square test
#MT2i only
MT2i_only <- data.frame(
  "MT2-associated" = c(4298, 354),
  "nonMT2-associated" = c(55401-4298, 934-354),
  row.names = c("all genes", "MT2i_only")
)
chisq.test(MT2i_only)$p.value #<2.2e-16

shared <- data.frame(
  "MERVL-assocaited" = c(1428, 92),
  "nonMERVL-associated" = c(55401-1428, 177-92),
  row.names = c("all genes", "shared")
)
chisq.test(shared)$p.value #<2.2e-16

MERVLKD_only <- data.frame(
  "MERVL-associated" = c(1428, 11),
  "nonMERVL-associated" = c(55401-1428, 212-11),
  row.names = c("all genes", "MERVLKD_only")
)
chisq.test(MERVLKD_only) #0.02
```
# 8. cis-function of MT2/MERVL

## 8.1 MT2/MERVL splicing analyses

### 8.1.1 extract SJ using regtools
```{bash}
regtools junctions extract -o total_2c_SJ.bed -s RF all_2C_merged.sorted.bam
#Note that this bam is a merge of polyA and total RNAseq of 2-cell embryos
```

### 8.1.2 Annotate SJ by repeats
To identify SJ that overlap with MT2/MERVL

```{perl annoate SJ overlap with MT2}
#!/usr/bin/perl -w
use strict;

#1) extract SJs from RNA-seq
#2) identify LTR overlap SJs
#3) annotate LTR overlap SJs

#2)-------------------------------
my $mm10="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/mm10.fa";
my $GTF="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/gencode.vM25.annotation.gtf";
my $dCas9MT2="/data/ZYChenlab/Zhiyuan/projects/TE_project/230829_MT2_Mm_Integrative_analyses/R_output/231030_dCas9_MT2Mm_MT2CMm.bed";
my $SJ = shift @ARGV;;
my $LTR = shift @ARGV;

my $sj_input;
if($SJ =~ /(.+)_SJ\.bed/){
        $sj_input =  $1;
}

my $region_dir="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/output";
my $FL_MERVL="${region_dir}/full_length_MERVL.bed";
my $FL_MERVL_3primeMT2="${region_dir}/full_length_MERVL_downMT2.bed";
my $FL_MERVL_5primeMT2="${region_dir}/full_length_MERVL_upMT2.bed";
my $soloMT2="${region_dir}/soloMT2.bed";
my $otherMT2="${region_dir}/othersMT2.bed";

system("grep '${LTR}' $FL_MERVL_3primeMT2 > FL_MERVL_3prime${LTR}.bed");
system("grep '${LTR}' $FL_MERVL_5primeMT2 > FL_MERVL_5prime${LTR}.bed");
system("grep '${LTR}' $soloMT2 > solo${LTR}.bed");
system("grep '${LTR}' $otherMT2 > other${LTR}.bed");

#Split bed12 into regular bed for intersection
#Only keep the splicing donor exons
#Also keep the original bed12 records
system("bed12ToBed6 -i $SJ > ${sj_input}_SJ_splited.bed");
system("grep '+' ${sj_input}_SJ_splited.bed > ${sj_input}_SJ_splited_plus.bed");
system("grep '-' ${sj_input}_SJ_splited.bed > ${sj_input}_SJ_splited_minus.bed");

#for + strand, keep the 1st record, which is the donor
system("awk 'NR == 1 || NR % 2 == 1' ${sj_input}_SJ_splited_plus.bed > ${sj_input}_SJ_splited_plus_filtered.bed");
#for - strand, keep the 2nd record, which is the donor
system("awk 'NR % 2 == 0' ${sj_input}_SJ_splited_minus.bed > ${sj_input}_SJ_splited_minus_filtered.bed");

my @array= ("FL_MERVL_3prime${LTR}.bed", "FL_MERVL_5prime${LTR}.bed", "solo${LTR}.bed", "other${LTR}.bed");
foreach(@array){
        if($_ =~ /(.+)\.bed/){
                my $name = $1;
                system("bedtools intersect -a ${sj_input}_SJ_splited_plus_filtered.bed -b $_ -wa -s > ${name}_overlap.bed");
                system("bedtools intersect -a ${sj_input}_SJ_splited_minus_filtered.bed -b $_ -wa -s >> ${name}_overlap.bed");
                
                #only keep MT2s that are targeted by dCas9
                system("bedtools intersect -a $_ -b $dCas9MT2 -wa > ${name}_dCas9.bed");
                system("bedtools intersect -a ${name}_overlap.bed -b ${name}_dCas9.bed  -wa -wb -s > ${name}_${sj_input}_SJ_intersect.txt");:wq
                my %JUNC;
                open(IN, "${name}_overlap.bed") or die$!;
                while (my $line = <IN>){
                        chomp $line;
                        my @split = split (/\s+/, $line);
                        $JUNC{$split[3]} = 1;
                } 
                close IN;
                open(IN2, $SJ) or die$!;
                open(OUT, ">${name}_${sj_input}_SJ.bed");
                while (my $line = <IN2>){
                        chomp $line;
                        my @split = split (/\s+/, $line);
                        if($JUNC{$split[3]}){
                                print OUT $line . "\n";
                        }
                }
                close IN2;
                close OUT;
        }
}
system("rm *splited*");
system("rm *overlap*");

#3)-----------------------------------------------
foreach(@array){
        if($_ =~ /(.+)\.bed/){
                my $name = $1;
                system("regtools junctions annotate ${name}_${sj_input}_SJ.bed $mm10 $GTF -o ${name}_${sj_input}_SJ_annotate.txt");
                #system("bedtools intersect -a ${name}_CTRi_2C_SJ.bed -b $_ -wa -wb > ${name}_CTRi_2C_SJ_intersect.txt");
        }
}
```

```{perl annotate SJ overlap with MERVL}
#!/usr/bin/perl -w
use strict;

#1) extract SJs from RNA-seq
#2) identify MERVL overlap SJs
#3) annotate MERVL overlap SJs

#1)-------------------------------
#source ~/.bash_profile
#conda activate regtools 
#module load samtools/1.9.0 bedtools/2.27.0 
#regtools junctions extract -o CTRi_2C_SJ.bed -s RF /data/ZYChenlab/Zhiyuan/projects/TE_project/230705_MT2Mmi_totalRNAseq/merged_Bam/CTRi_2C_merged.sorted.bam 

#2)-------------------------------
my $mm10="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/mm10.fa";
my $GTF="/data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/gencode.vM25.annotation.gtf";
my $dCas9MT2="/data/ZYChenlab/Zhiyuan/projects/TE_project/230829_MT2_Mm_Integrative_analyses/R_output/231030_dCas9_MT2Mm_MT2CMm.bed";
my $SJ = shift @ARGV;;
my $MERVL = shift @ARGV;

my $sj_input;
if($SJ =~ /(.+)_SJ\.bed/){
        $sj_input =  $1;
}

my $region_dir="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/output";
my $FL_MERVL="${region_dir}/full_length_MERVL.bed";
my $nonFL_MERVL="${region_dir}/nonFL_MERVL.bed";

system("grep '${MERVL}' $FL_MERVL > FL_${MERVL}.bed");
system("grep '${MERVL}' $nonFL_MERVL > nonFL_${MERVL}.bed");

#Split bed12 into regular bed for intersection
#Only keep the splicing donor exons
#Also keep the original bed12 records

system("bed12ToBed6 -i $SJ > ${sj_input}_SJ_splited.bed");
system("grep '+' ${sj_input}_SJ_splited.bed > ${sj_input}_SJ_splited_plus.bed");
system("grep '-' ${sj_input}_SJ_splited.bed > ${sj_input}_SJ_splited_minus.bed");

#for + strand, keep the 1st record
system("awk 'NR == 1 || NR % 2 == 1' ${sj_input}_SJ_splited_plus.bed > ${sj_input}_SJ_splited_plus_filtered.bed");
#for - strand, keep the 2nd record
system("awk 'NR % 2 == 0' ${sj_input}_SJ_splited_minus.bed > ${sj_input}_SJ_splited_minus_filtered.bed");

my @array = ("FL_${MERVL}.bed", "nonFL_${MERVL}.bed");
foreach(@array){
        if($_ =~ /(.+)\.bed/){
                my $name = $1;
                system("bedtools intersect -a ${sj_input}_SJ_splited_plus_filtered.bed -b $_ -wa -s > ${name}_overlap.bed");
                system("bedtools intersect -a ${sj_input}_SJ_splited_minus_filtered.bed -b $_ -wa -s >> ${name}_overlap.bed");
        
                #only keep MT2s that are targeted by dCas9
                #system("bedtools intersect -a $_ -b $dCas9MT2 -wa > ${name}_dCas9.bed");
                system("bedtools intersect -a ${name}_overlap.bed -b $_  -wa -wb -s > ${name}_${sj_input}_SJ_intersect.txt");
                my %JUNC;
                open(IN, "${name}_overlap.bed") or die$!;
                while (my $line = <IN>){
                        chomp $line;
                        my @split = split (/\s+/, $line);
                        $JUNC{$split[3]} = 1;
                } 
                close IN;
                open(IN2, $SJ) or die$!;
                open(OUT, ">${name}_${sj_input}_SJ.bed");
                while (my $line = <IN2>){
                        chomp $line;
                        my @split = split (/\s+/, $line);
                        if($JUNC{$split[3]}){
                                print OUT $line . "\n";
                        }
                }
                close IN2;
                close OUT;
        }
}
system("rm *splited*");
system("rm *overlap*");

#3)-----------------------------------------------
foreach(@array){
        if($_ =~ /(.+)\.bed/){
                my $name = $1;
                system("regtools junctions annotate ${name}_${sj_input}_SJ.bed $mm10 $GTF -o ${name}_${sj_input}_SJ_annotate.txt");
                #system("bedtools intersect -a ${name}_CTRi_2C_SJ.bed -b $_ -wa -wb > ${name}_CTRi_2C_SJ_intersect.txt");
        }
}
```


### 8.1.3 Overview SJ datasets
```{r}
annotate.file.list <- list(
  FL_3MT2_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_3primeMT2_Mm_total_polyA_2c_SJ_annotate.txt",
  FL_5MT2_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_5primeMT2_Mm_total_polyA_2c_SJ_annotate.txt",
  soloMT2_Mm_2C = "./R_input/TE_chimeric/soloMT2_Mm_total_polyA_2c_SJ_annotate.txt",
  otherMT2_Mm_2C = "./R_input/TE_chimeric/otherMT2_Mm_total_polyA_2c_SJ_annotate.txt",
  FL_3MT2C_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_3primeMT2C_Mm_total_polyA_2c_SJ_annotate.txt",
  FL_5MT2C_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_5primeMT2C_Mm_total_polyA_2c_SJ_annotate.txt",
  soloMT2C_Mm_2C = "./R_input/TE_chimeric/soloMT2C_Mm_total_polyA_2c_SJ_annotate.txt",
  otherMT2C_Mm_2C = "./R_input/TE_chimeric/otherMT2C_Mm_total_polyA_2c_SJ_annotate.txt",
  FL_MERVL_2C = "./R_input/TE_chimeric/FL_MERVL_total_polyA_2c_SJ_annotate.txt",
  nonFL_MERVL_2C = "./R_input/TE_chimeric/nonFL_MERVL_total_polyA_2c_SJ_annotate.txt"
)
intersect.file.list <- list(
  FL_3MT2_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_3primeMT2_Mm_total_polyA_2c_SJ_intersect.txt",
  FL_5MT2_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_5primeMT2_Mm_total_polyA_2c_SJ_intersect.txt",
  soloMT2_Mm_2C = "./R_input/TE_chimeric/soloMT2_Mm_total_polyA_2c_SJ_intersect.txt",
  otherMT2_Mm_2C = "./R_input/TE_chimeric/otherMT2_Mm_total_polyA_2c_SJ_intersect.txt",
  FL_3MT2C_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_3primeMT2C_Mm_total_polyA_2c_SJ_intersect.txt",
  FL_5MT2C_Mm_2C = "./R_input/TE_chimeric/FL_MERVL_5primeMT2C_Mm_total_polyA_2c_SJ_intersect.txt",
  soloMT2C_Mm_2C = "./R_input/TE_chimeric/soloMT2C_Mm_total_polyA_2c_SJ_intersect.txt",
  otherMT2C_Mm_2C = "./R_input/TE_chimeric/otherMT2C_Mm_total_polyA_2c_SJ_intersect.txt",
  FL_MERVL_2C = "./R_input/TE_chimeric/FL_MERVL_total_polyA_2c_SJ_intersect.txt",
  nonFL_MERVL_2C = "./R_input/TE_chimeric/nonFL_MERVL_total_polyA_2c_SJ_intersect.txt"
)

annotate.list <- list()
intersect.list <- list()
for (i in 1:length(annotate.file.list)){
  annotate.list[[i]] <- read.table(annotate.file.list[[i]], header = T)
  intersect.list[[i]] <- read.table(intersect.file.list[[i]], header = F)
  colnames(intersect.list[[i]]) <- c("SJ_chrom", "SJ_start", "SJ_end", "name", "score", 
                                     "strand",
                                     "TE_chrom", "TE_start", "TE_end", "TE_name", 
                                     "TE_score", "TE_strand")
}
names(annotate.list) <- names(annotate.file.list)
names(intersect.list) <- names(annotate.file.list)
sapply(annotate.list, nrow)
#FL_3MT2_Mm_2C   FL_5MT2_Mm_2C   soloMT2_Mm_2C  otherMT2_Mm_2C  FL_3MT2C_Mm_2C  FL_5MT2C_Mm_2C  soloMT2C_Mm_2C otherMT2C_Mm_2C 
#           1187            2322            3283            1077               5              15            1728             239 
#    FL_MERVL_2C  nonFL_MERVL_2C 
#           3587             511 
sapply(intersect.list, nrow)
# FL_3MT2_Mm_2C   FL_5MT2_Mm_2C   soloMT2_Mm_2C  otherMT2_Mm_2C  FL_3MT2C_Mm_2C  FL_5MT2C_Mm_2C  soloMT2C_Mm_2C otherMT2C_Mm_2C 
#            1187            2322            3283            1077               5              15            1728             239 
#     FL_MERVL_2C  nonFL_MERVL_2C 
#            3587             535 

#merge annotation and intersection file
df.list <- list()
for (i in 1:length(annotate.list)){
  df.list[[i]] <- merge(annotate.list[[i]], intersect.list[[i]], by = "name")
  
}
names(df.list) <- names(annotate.file.list)

SJ_cutoff <- 1 #minimum SJ reads 

df.filter.list <- list()
for (i in 1:length(df.list)){
  df.filter.list[[i]] <- subset(df.list[[i]], score.x >= SJ_cutoff)
}
names(df.filter.list) <- names(df.list)
#sapply(df.filter.list, nrow)

#output for sequencing logo analyses
for (i in 1:length(df.filter.list)){
  tmp.df1 <- df.filter.list[[i]][!duplicated(df.filter.list[[i]]$name),]
  write.table(tmp.df1, paste0("./R_input/sequenceLogo_analyses/", names(df.filter.list)[i], "_total_2c_SJ_filtered.txt"), sep = "\t", 
                             quote = F, row.names = F, col.names = F)
  #nrow(tmp.df1)
}
#output BED format for deeptool analyses
for (i in 1:length(df.filter.list)){
  tmp.df1 <- df.filter.list[[i]][!duplicated(df.filter.list[[i]]$TE_name),]
  tmp.df <- data.frame()
  tmp.df <- tmp.df1[, c("chrom", "TE_start", "TE_end", "TE_name", "score.y",
                                    "TE_strand")]
  write.table(tmp.df, paste0("./R_output/TE_chimeric/", names(df.filter.list)[i], "_total_2c_SJ_filtered.bed"), sep = "\t", 
                             quote = F, row.names = F, col.names = F)
  nrow(tmp.df)
}
```

### 8.1.4 SJ sequence logo analyses

To make sure that these are real splicing junctions
```{perl retrieve_seq_at_SJ.pl}
#!/usr/bin/perl -w 

use strict;

#1) defeine sub to retrieve sequences

#my $seq  = &retrieve_SJ_donor_seq('chr1', 58935940, 58938295, 5, 10,  "-");
my $in = shift @ARGV;
open (IN, $in) or die$!;
my $sample;
if($in =~ /(.+)\_total_2c/){
        $sample = $1;
}
open (DONOROUT, ">${sample}_donor_SJ.txt") or die$!;
open (ACCEPTOUT, ">${sample}_accept_SJ.txt") or die$!;

my %hash;
while (my $line = <IN>){
        chomp $line;
        next if $line =~ /splice_site/;

        my @split = split (/\s+/, $line);
        my $chr = $split[1];
        my $SJ_start = $split[2];
        my $SJ_end = $split[3];
        my $name = $split[0];
        my $strand = $split[5];

        #do not repeat the same junction
        unless($hash{$name}){
                $hash{$name} = 1;
                my $donor = &retrieve_SJ_donor_seq($chr, $SJ_start, $SJ_end, 5, 10, $strand);
                print DONOROUT "$donor\n";
                my $accept = &retrieve_SJ_acceptor_seq($chr, $SJ_start, $SJ_end, 5, 10, $strand);
                print ACCEPTOUT "$accept\n";
        }
}

close IN;
close DONOROUT;
close ACCEPTOUT;
exit;

sub retrieve_SJ_acceptor_seq{
        my ($chr, $SJ_start, $SJ_end, $upstream, $downstream, $strand) = @_;
        my ($range, $range_start, $range_end, $seq);
        if($strand eq "+"){
                $range_start = $SJ_end - $downstream + 1;
                $range_end = $SJ_end + $upstream;
                $range = $chr . ":" . $range_start . "-". "$range_end";
                $seq = `samtools faidx /data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/mm10.fa $range`;
                my @split = split(/\n/, $seq);
                $split[1] =~ s/\n//g;
                return $split[1];
        }
        else{ #negative strand
                $range_start = $SJ_start - $upstream;
                $range_end = $SJ_start + $downstream - 1;
                $range = $chr . ":" . $range_start . "-" . $range_end;
                $seq = `samtools faidx /data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/mm10.fa $range`;
                my @split = split(/\n/, $seq);
                $split[1] =~ s/\n//g;
                #reverse complementary
                my $revcomp = reverse $split[1];
                $revcomp =~ tr/ATGCatgc/TACGtacg/;
                return $revcomp;
                }
}

sub retrieve_SJ_donor_seq{
        my ($chr, $SJ_start, $SJ_end, $upstream, $downstream, $strand) = @_;
        my ($range, $range_start, $range_end, $seq);
        if($strand eq "+"){
                $range_start = $SJ_start - $upstream + 1;
                $range_end = $SJ_start + $downstream;
                $range = $chr . ":" . $range_start . "-". "$range_end";
                $seq = `samtools faidx /data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/mm10.fa $range`;
                my @split = split(/\n/, $seq);
                $split[1] =~ s/\n//g;
                return $split[1];
        }
        else{ #negative strand
                $range_start = $SJ_end - $downstream;
                $range_end = $SJ_end + $upstream - 1;
                $range = $chr . ":" . $range_start . "-" . $range_end;
                $seq = `samtools faidx /data/ZYChenlab/Zhiyuan/genomes_annotations/mm10/genomes/mm10.fa $range`;
                my @split = split(/\n/, $seq);
                $split[1] =~ s/\n//g;
                #reverse complementary
                my $revcomp = reverse $split[1];
                $revcomp =~ tr/ATGCatgc/TACGtacg/;
                return $revcomp;
        }
}
```

```{bash}
perl retrieve_seq_at_SJ.pl FL_3MT2_Mm_2C_total_2c_SJ_filtered.txt 
perl retrieve_seq_at_SJ.pl FL_5MT2_Mm_2C_total_2c_SJ_filtered.txt
perl retrieve_seq_at_SJ.pl soloMT2_Mm_2C_total_2c_SJ_filtered.txt
perl retrieve_seq_at_SJ.pl otherMT2_Mm_2C_total_2c_SJ_filtered.txt
perl retrieve_seq_at_SJ.pl FL_3MT2C_Mm_2C_total_2c_SJ_filtered.txt 
perl retrieve_seq_at_SJ.pl FL_5MT2C_Mm_2C_total_2c_SJ_filtered.txt
perl retrieve_seq_at_SJ.pl soloMT2C_Mm_2C_total_2c_SJ_filtered.txt
perl retrieve_seq_at_SJ.pl otherMT2C_Mm_2C_total_2c_SJ_filtered.txt 
```

```{r, fig.width = 3, fig.height = 2}
require(ggplot2)
require(ggseqlogo)

FL_3MT2_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/FL_3MT2_Mm_2C_donor_SJ.txt", header = F)
FL_5MT2_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/FL_5MT2_Mm_2C_donor_SJ.txt", header = F)
soloMT2_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/soloMT2_Mm_2C_donor_SJ.txt", header = F)
otherMT2_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/otherMT2_Mm_2C_donor_SJ.txt", header = F)
FL_3MT2C_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/FL_3MT2C_Mm_2C_donor_SJ.txt", header = F)
FL_5MT2C_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/FL_5MT2C_Mm_2C_donor_SJ.txt", header = F)
soloMT2C_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/soloMT2C_Mm_2C_donor_SJ.txt", header = F)
otherMT2C_Mm_donor <- read.table("./R_input/sequenceLogo_analyses/otherMT2C_Mm_2C_donor_SJ.txt", header = F)

# ggseqlogo(FL_3MT2_Mm_donor$V1)
# ggseqlogo(FL_5MT2_Mm_donor$V1)
# ggseqlogo(soloMT2_Mm_donor$V1)
# ggseqlogo(otherMT2_Mm_donor$V1)
# ggseqlogo(FL_3MT2C_Mm_donor$V1)
# ggseqlogo(FL_5MT2C_Mm_donor$V1)
# ggseqlogo(soloMT2C_Mm_donor$V1)
# ggseqlogo(otherMT2C_Mm_donor$V1)
#No big difference bewtween each group, just combine

ggseqlogo(c(FL_3MT2C_Mm_donor$V1, FL_5MT2C_Mm_donor$V1, 
            soloMT2C_Mm_donor$V1, otherMT2C_Mm_donor$V1,
            FL_3MT2_Mm_donor$V1, FL_5MT2_Mm_donor$V1, 
            soloMT2_Mm_donor$V1, otherMT2_Mm_donor$V1))
```

```{r, fig.width = 3, fig.height = 2}
FL_3MT2_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/FL_3MT2_Mm_2C_accept_SJ.txt", header = F)
FL_5MT2_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/FL_5MT2_Mm_2C_accept_SJ.txt", header = F)
soloMT2_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/soloMT2_Mm_2C_accept_SJ.txt", header = F)
otherMT2_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/otherMT2_Mm_2C_accept_SJ.txt", header = F)
FL_3MT2C_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/FL_3MT2C_Mm_2C_accept_SJ.txt", header = F)
FL_5MT2C_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/FL_5MT2C_Mm_2C_accept_SJ.txt", header = F)
soloMT2C_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/soloMT2C_Mm_2C_accept_SJ.txt", header = F)
otherMT2C_Mm_acceptor <- read.table("./R_input/sequenceLogo_analyses/otherMT2C_Mm_2C_accept_SJ.txt", header = F)

# ggseqlogo(FL_3MT2_Mm_acceptor$V1)
# ggseqlogo(FL_5MT2_Mm_acceptor$V1)
# ggseqlogo(soloMT2_Mm_acceptor$V1)
# ggseqlogo(otherMT2_Mm_acceptor$V1)
# ggseqlogo(FL_3MT2C_Mm_acceptor$V1)
# ggseqlogo(FL_5MT2C_Mm_acceptor$V1)
# ggseqlogo(soloMT2C_Mm_acceptor$V1)
# ggseqlogo(otherMT2C_Mm_acceptor$V1)

#No big difference bewtween each group, just combine
ggseqlogo(c(FL_3MT2_Mm_acceptor$V1, FL_5MT2_Mm_acceptor$V1, 
            soloMT2_Mm_acceptor$V1, otherMT2_Mm_acceptor$V1,
            FL_3MT2C_Mm_acceptor$V1, FL_5MT2C_Mm_acceptor$V1, 
            soloMT2C_Mm_acceptor$V1, otherMT2C_Mm_acceptor$V1))

pdf("./figures/231115_MT2_SJ_seqLogo.pdf", height = 3, width = 6)
ggseqlogo(c(FL_3MT2_Mm_donor$V1, FL_5MT2_Mm_donor$V1, 
            soloMT2_Mm_donor$V1, otherMT2_Mm_donor$V1,
            FL_3MT2C_Mm_donor$V1, FL_5MT2C_Mm_donor$V1, 
            soloMT2C_Mm_donor$V1, otherMT2C_Mm_donor$V1))
ggseqlogo(c(FL_3MT2_Mm_acceptor$V1, FL_5MT2_Mm_acceptor$V1, 
            soloMT2_Mm_acceptor$V1, otherMT2_Mm_acceptor$V1,
            FL_3MT2C_Mm_acceptor$V1, FL_5MT2C_Mm_acceptor$V1, 
            soloMT2C_Mm_acceptor$V1, otherMT2C_Mm_acceptor$V1))
dev.off()
```

### 8.1.5 annotate SJ by genes
```{r, fig.height = 3, fig.width = 3}
suppressMessages(library(dplyr))
suppressMessages(library(tidyr))
df.known.annotation <- list()
MT2_known_SJ <- 0
MT2_unknown_SJ <- 0
for (i in 1:length(df.filter.list)){
  if (i == 5){ next } #skip FL_3MT2C_Mm, which has no known junctions
  df.known.annotation[[i]] <- subset(df.filter.list[[i]], anchor %in% c("DA", "NDA", "A", "D"))
  df.known.annotation[[i]]$group <- names(df.filter.list)[i] 
  slice4pie <- c(
    length(unique(df.known.annotation[[i]]$TE_name)),
    length(unique(df.filter.list[[i]]$TE_name))-length(unique(df.known.annotation[[i]]$TE_name))
  )
  label <- c(
    paste0("known:", length(unique(df.known.annotation[[i]]$TE_name))),
    paste0("unknown:", length(unique(df.filter.list[[i]]$TE_name))-length(unique(df.known.annotation[[i]]$TE_name)))
  )
  pie(slice4pie, labels = label, main = names(df.filter.list)[i])
  MT2_known_SJ <- MT2_known_SJ + length(unique(df.known.annotation[[i]]$TE_name))
  MT2_unknown_SJ <- MT2_unknown_SJ + length(unique(df.filter.list[[i]]$TE_name))-length(unique(df.known.annotation[[i]]$TE_name))
}
names(df.known.annotation) <- names(df.filter.list)

# this should the list of MT2_Mm-chimeric splicing junction
# combine as a single file and label each MT2_Mm catogory.
MT2_chimeric <- rbind(df.known.annotation[[1]], 
                         df.known.annotation[[2]],
                         df.known.annotation[[3]],
                         df.known.annotation[[4]],
                         df.known.annotation[[5]],
                         df.known.annotation[[6]],
                         df.known.annotation[[7]],
                         df.known.annotation[[8]],
                         df.known.annotation[[9]],
                         df.known.annotation[[10]])

#some SJs cover more than one gene, split these
#genes into seperate rows.

MT2_chimeric_cleaned <- MT2_chimeric[, c("name", "chrom", "start", "end", "score.x",
                                       "strand.x", "splice_site", "anchor", 
                                       "gene_ids", "TE_start",
                                       "TE_end", "TE_name", "TE_strand", "group")]
colnames(MT2_chimeric_cleaned)[9] <- "id" 
#start: junction start, end: junction end
#score.x: # of SJ reads

MT2_chimeric <- MT2_chimeric_cleaned %>% 
                  separate_rows(id, sep = ",")
length(unique(MT2_chimeric$id)) #665
length(unique(MT2_chimeric$TE_name)) #699

#For one gene it may have mutiple splicing junction
#for simplicity, peak the one with the most reads support
suppressMessages(library(dplyr))

MT2_chimeric_simple <- MT2_chimeric %>% 
                          group_by(id) %>%
                          slice_max(order_by = score.x)

  
write.table(MT2_chimeric,"./R_output/TE_chimeric/231115_MT2_chimeric.txt", sep = "\t",
            quote = F, row.names = F)
pie(c(MT2_known_SJ, MT2_unknown_SJ), 
    labels = c(paste0("MT2_known: ", MT2_known_SJ),
              paste0("MT2_unknown:", MT2_unknown_SJ)))
```
Check gene type
```{r, fig.height = 3, fig.width = 3}
gene_type <- read.table("/Volumes/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/gencode.vM25.id.type.txt",
                        header = F, sep = "\t")
colnames(gene_type) <- c("id", "type")
MT2_chimeric <- left_join(MT2_chimeric, gene_type, by = "id")
colnames(MT2_chimeric)

df4GeneTypePie <- MT2_chimeric[!duplicated(MT2_chimeric$TE_name),]
df4GeneTypePie <- as.data.frame(table(df4GeneTypePie$type))

tmp.df <- data.frame(Var1 = "Pseudogenes",
                     Freq = sum(subset(df4GeneTypePie, !(df4GeneTypePie$Var1 %in% c("protein_coding", "lincRNA")))[, "Freq"]))

tmp.df <- rbind(tmp.df, df4GeneTypePie[which(df4GeneTypePie$Var1 %in% c("protein_coding", "lincRNA")),])
tmp.df <- rbind(tmp.df, data.frame(Var1 = "Unknown", Freq = MT2_unknown_SJ))
tmp.df
pie(tmp.df$Freq, 
    labels = c(paste0("Pseudogenes:", tmp.df$Freq[1]), 
               paste0("LincRNA:", tmp.df$Freq[2]),
               paste0("protein_coding", tmp.df$Freq[3]),
               paste0("Unknown", tmp.df$Freq[4])))

pdf("./figures/231115_annotate_MT2_splicing.pdf", height = 6, width = 6)
pie(c(MT2_known_SJ, MT2_unknown_SJ), 
    labels = c(paste0("MT2_known: ", MT2_known_SJ),
              paste0("MT2_unknown:", MT2_unknown_SJ)))
pie(tmp.df$Freq, 
    labels = c(paste0("Pseudogenes:", tmp.df$Freq[1]), 
               paste0("LincRNA:", tmp.df$Freq[2]),
               paste0("protein_coding", tmp.df$Freq[3]),
               paste0("Unknown", tmp.df$Freq[4])))
dev.off()
```
## 8.2 MT2 promoter vs. distal

### 8.2.1 TE-centric gene expression
```{r, fig.height = 4, fig.width = 8}
source("./utils.R")
suppressMessages(library(ggplot2))
suppressMessages(library(cowplot))
suppressMessages(library(tidyr))

MT2Mm_distance <- read.table("./R_input/dCas9_ChIPseq/231030_MT2Mm_nearest_ChIPseqPeak2Gene.txt",
                             header = F, sep = "\t")
offtarget_distance <- read.table("./R_input/dCas9_ChIPseq/231030_offtarget_nearest_ChIPseqPeak2Gene.txt")

nrow(MT2Mm_distance) #55821
nrow(offtarget_distance) #55408
colnames <- c("gene_chr", "gene_start", "gene_end", 
                                   "id", "name", "gene_strand", 
                                   "peak_chr", "peak_start", "peak_end", 
                                   "peak_name", "peak_score", "peak_strand",
                                   "distance") 

colnames(MT2Mm_distance) <- colnames
colnames(offtarget_distance) <- colnames

MT2Mm_distance <- MT2Mm_distance[!duplicated(MT2Mm_distance$id),]
offtarget_distance <- offtarget_distance[!duplicated(offtarget_distance$id),]

nrow(MT2Mm_distance) #55401
nrow(offtarget_distance) #55401

MT2Mm_distance$bin <- classify_distance(MT2Mm_distance$distance, ignore_direction = F)
offtarget_distance$bin <- classify_distance(offtarget_distance$distance, ignore_direction = F)

MT2_c2_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_2C_genes_DE_analyses.csv", sep = ",", header = T)

df4boxplot_offtarget_l2c <- merge(offtarget_distance,
                           MT2_c2_RNA[, c("id", "log2FoldChange")],
                           by = "id")
df4boxplot_MT2Mm_l2c <- merge(MT2Mm_distance,
                            MT2_c2_RNA[, c("id", "log2FoldChange", "c2_DESeq_rpkm.group")],
                            by = "id")
pdf("./figures/231115_MT2Mm_ChIPpeaks_centric_expression_boxplot_noGroups.pdf")
fig1 <- ggplot(df4boxplot_MT2Mm_l2c, aes(x = bin, y = log2FoldChange)) +
  geom_boxplot(outlier.shape = NA) + theme_cowplot(16) +
  ylim(-10,10) + ylab("Log2FC (MT2_Mmi / CTRi") +
  xlab("Distance from MT2Mm (ChIPseq peaks)") +
  theme(axis.text.x = element_text(angle = 30, vjust = 0.9, hjust=0.9)) +
  ggtitle("CRISPRi gene expression\n(MT2Mm ChIPseq peaks)(L2C)") +
  geom_hline(yintercept= 0, linetype="dashed", color = "red") +
  scale_x_discrete(limits = c("< -50kb", "-30-50kb", "-10-30kb", "-0-10kb"
                              ,"0-10kb", "10-30kb", 
                              "30-50kb", ">50kb")) 

fig2 <- ggplot(df4boxplot_offtarget_l2c, aes(x = bin, y = log2FoldChange)) +
  geom_boxplot(outlier.shape = NA) + theme_cowplot(16) +
  ylim(-10,10) + ylab("Log2FC (MT2_Mmi / CTRi") +
  xlab("Distance from off-targets (ChIPseq peaks)") +
  theme(axis.text.x = element_text(angle = 30, vjust = 0.9, hjust=0.9)) +
  ggtitle("CRISPRi gene expression\n(off-target ChIPseq peaks), (L2C)") +
  geom_hline(yintercept= 0, linetype="dashed", color = "red") +
  scale_x_discrete(limits = c("< -50kb", "-30-50kb", "-10-30kb", "-0-10kb"
                              ,"0-10kb", "10-30kb", 
                              "30-50kb", ">50kb"))
plot_grid(fig1,fig2)
dev.off()
plot_grid(fig1,fig2)
```
run some stats here
```{r}
#get p-value here. 
wilcox.test(df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "0-10kb"), 
                                 c("log2FoldChange")],
            df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == ">50kb"),
                                 c("log2FoldChange")],)$p.value
wilcox.test(df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "10-30kb"), 
                                 c("log2FoldChange")],
            df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == ">50kb"),
                                 c("log2FoldChange")],)$p.value
wilcox.test(df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "30-50kb"), 
                                 c("log2FoldChange")],
            df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == ">50kb"),
                                 c("log2FoldChange")],)$p.value
wilcox.test(df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "-0-10kb"), 
                                 c("log2FoldChange")],
            df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "< -50kb"),
                                 c("log2FoldChange")],)$p.value
wilcox.test(df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "-10-30kb"), 
                                 c("log2FoldChange")],
            df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "< -50kb"),
                                 c("log2FoldChange")],)$p.value
wilcox.test(df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "-30-50kb"), 
                                 c("log2FoldChange")],
            df4boxplot_MT2Mm_l2c[which(df4boxplot_MT2Mm_l2c$bin == "< -50kb"),
                                 c("log2FoldChange")],)$p.value
wilcox.test(df4boxplot_offtarget_l2c[which(df4boxplot_offtarget_l2c$bin == "0-10kb"), 
                                 c("log2FoldChange")],
            df4boxplot_offtarget_l2c[which(df4boxplot_offtarget_l2c$bin == ">50kb"),
                                 c("log2FoldChange")],)$p.value
wilcox.test(df4boxplot_offtarget_l2c[which(df4boxplot_offtarget_l2c$bin == "-0-10kb"), 
                                 c("log2FoldChange")],
            df4boxplot_offtarget_l2c[which(df4boxplot_offtarget_l2c$bin == "< -50kb"),
                                 c("log2FoldChange")],)$p.value
# [1] 6.708161e-150
# [1] 1.274874e-35
# [1] 4.514251e-08
# [1] 1.763113e-15
# [1] 2.086582e-05
# [1] 0.07372972
# [1] 0.1244676
# [1] 0.5339557
```

### 8.2.2 MT2 Proximal/chimeric vs. distal
```{r}
source("./utils.R")
gene_MT2_distance <- read.table("./R_input/dCas9_ChIPseq/231115_nearest_gene2MT2.txt",
                                         header = F, sep = "\t")
colnames(gene_MT2_distance) <- c("gene_chr", "gene_start", "gene_end", 
                                   "id", "name", "gene_strand", 
                                   "peak_chr", "peak_start", "peak_end", 
                                   "peak_name", "peak_score", "peak_strand",
                                   "distance")
gene_MT2_distance <- gene_MT2_distance[!duplicated(gene_MT2_distance$id),]
gene_MT2_distance$bin <- classify_distance(gene_MT2_distance$distance, ignore_direction = F)
gene_MT2_distance$group <- classify_repeat_gene_pair(gene_MT2_distance)
nrow(gene_MT2_distance) #55401

MT2_e2c_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_e2C_genes_DE_analyses.csv", sep =",", header = T)
MT2_c2_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_2C_genes_DE_analyses.csv", sep = ",", header = T)

MT2_RNA <- merge(MT2_e2c_RNA, MT2_c2_RNA, by = "id", all = T)
nrow(MT2_RNA) #29803

MT2_RNA <- left_join(MT2_RNA, gene_MT2_distance, by = "id")

MT2_RNA_down <- MT2_RNA[which(MT2_RNA$c2_DESeq_rpkm.group == "down-regulated" |
                            MT2_RNA$ec2_DESeq_rpkm.group == "down-regulated"),]
nrow(MT2_RNA_down) #1155

MT2_RNA_down_30kb <- MT2_RNA_down[which(MT2_RNA_down$bin %in%
                                            c("-0-10kb", "-10-30kb", "0-10kb",
                                              "10-30kb"#, 
                                              #"30-50kb", 
                                              #"-30-50kb"
                                              )),]
nrow(MT2_RNA_down_30kb) #717
MT2_RNA_down_30kb_SJ <- merge(MT2_RNA_down_30kb[, c("id", "gene_chr", "gene_start",
                                                    "gene_end", "name", "gene_strand")], 
                              MT2_chimeric_simple[, c("id", "TE_start", "TE_end", "TE_name",
                                                      "TE_strand")], by = "id")
MT2_RNA_down_30kb_SJ$group <- c("MT2 proximal/chimeric")
nrow(MT2_RNA_down_30kb_SJ) #331
MT2_RNA_down_30kb_SJ <- MT2_RNA_down_30kb_SJ[!duplicated(MT2_RNA_down_30kb_SJ$id),]
nrow(MT2_RNA_down_30kb_SJ) #329
MT2_RNA_down_30kb_nonSJ <- subset(MT2_RNA_down_30kb[, c("id", "gene_chr", "gene_start",
                                                    "gene_end", "name", "gene_strand", 
                                                    "peak_start", "peak_end", "peak_name",
                                                    "peak_strand", "group")], 
                                 !(MT2_RNA_down_30kb$id %in% MT2_RNA_down_30kb_SJ$id))
nrow(MT2_RNA_down_30kb_nonSJ) #388

```
For the E2C/L2C MT2i down genes (n=1155), 717 genes are <30kb to MT2.
Of these 717 genes, 388 of them do not have splicing, whereas 329 of them 
form splicing.\ 

Need to ensure that the 361 genes are not near TSS (-1kb, +500bp) regions before 
considering them as distal elements.\ 

```{r}
MT2_RNA_down_30kb_nonSJ$annotation <- rep("Distal", nrow(MT2_RNA_down_30kb_nonSJ))

for(i in 1:nrow(MT2_RNA_down_30kb_nonSJ)){
  #Proximal critera:
  #1) same direction
  #2) distance < 1kb from the TSS
  if(MT2_RNA_down_30kb_nonSJ$gene_strand[i] == MT2_RNA_down_30kb_nonSJ$peak_strand[i]){
    if(MT2_RNA_down_30kb_nonSJ$gene_strand[i] == "+"){
       tmp_tss_lower <- MT2_RNA_down_30kb_nonSJ$gene_start[i] - 1000
       tmp_tss_upper <- MT2_RNA_down_30kb_nonSJ$gene_start[i] + 500
       if (tmp_tss_lower <= MT2_RNA_down_30kb_nonSJ$peak_end[i] &
           tmp_tss_upper >= MT2_RNA_down_30kb_nonSJ$peak_start[i]){
              MT2_RNA_down_30kb_nonSJ$annotation[i] <- "MT2 proximal/chimeric"
           }
    } else if (MT2_RNA_down_30kb_nonSJ$gene_strand[i] == "-"){
       tmp_tss_lower <- MT2_RNA_down_30kb_nonSJ$gene_end[i] - 500
       tmp_tss_upper <- MT2_RNA_down_30kb_nonSJ$gene_end[i] + 1000
       if (tmp_tss_lower <= MT2_RNA_down_30kb_nonSJ$peak_end[i] &
           tmp_tss_upper >= MT2_RNA_down_30kb_nonSJ$peak_start[i]){
              MT2_RNA_down_30kb_nonSJ$annotation[i] <- "MT2 proximal/chimeric"
       }
    }
  }
}
table(MT2_RNA_down_30kb_nonSJ$annotation)
#Distal Proximal 
#     301       87 

table(MT2_RNA_down_30kb_nonSJ[which(MT2_RNA_down_30kb_nonSJ$annotation == "MT2 proximal/chimeric"), c("group")])
#sense_down sense_up/intragenic 
 #                 1                  86 
table(MT2_RNA_down_30kb_nonSJ[which(MT2_RNA_down_30kb_nonSJ$annotation == "Distal"), c("group")])
#anti_down/intragenic              anti_up           sense_down  sense_up/intragenic 
#                  61                   59                   26                  155

pdf("./figures/231115_MT2_Proximal_distal_grouop.pdf", width = 6, height = 6)
pie(c(87+329, 301), 
    labels = c("MT2_chimeric/promoter", "Distal"))
pie(table(MT2_RNA_down_30kb_nonSJ[which(MT2_RNA_down_30kb_nonSJ$annotation == "Distal"), c("group")]))
dev.off()

write.table(MT2_RNA_down_30kb_nonSJ, file = "./R_output/MT2i_totalRNA/231115_MT2_2c_down_nonSJ_annotation.txt", 
            quote = F, row.names = F, sep = "\t")
write.table(MT2_RNA_down_30kb_SJ, file = "./R_output/MT2i_totalRNA/231115_MT2_2c_down_SJ_annotation.txt", 
            quote = F, row.names = F, sep = "\t")

#nrow(MT2_RNA_down_30kb[!duplicated(MT2_RNA_down_30kb$id),]) #717
#nrow(MT2_RNA_down_30kb[!duplicated(MT2_RNA_down_30kb$peak_name),]) #627
```

### 8.2.3 MERVL Proximal/chimeric vs. distal
```{r}
# #input gene list
# MERVLi_2c <- read.table("./R_output/Sakashita_output/231115_MERVLi_2C_genes_DE_analyses.csv", 
#                       header = T, sep = ",")
# MERVLKD_2c <- read.table("./R_output/Sakashita_output/231115_MERVLKD_2C_genes_DE_analyses.csv", 
#                       header = T, sep = ",")
# nrow(MERVLi_2c[which(MERVLi_2c$i2C_DESeq_rpkm.group == "down-regulated"),]) #81  
# nrow(MERVLKD_2c[which(MERVLKD_2c$KD2C_DESeq_rpkm.group == "down-regulated"),]) #359
# 
# #input MERVL gene distance
# gene_MERVL_distance <- read.table("./R_input/dCas9_ChIPseq/231106_nearest_MERVL2gene.txt",
#                                          header = F, sep = "\t")
# nrow(gene_MERVL_distance) #55447
# colnames(gene_MERVL_distance) <- c("gene_chr", "gene_start", "gene_end", 
#                                    "id", "name", "gene_strand", 
#                                    "peak_chr", "peak_start", "peak_end", 
#                                    "peak_name", "peak_score", "peak_strand",
#                                    "distance")
# #for genes with multiple peaks in the gene bodies, randomly pick one
# gene_MERVL_distance <- gene_MERVL_distance[!duplicated(gene_MERVL_distance$id),]
# nrow(gene_MERVL_distance) #55401
# gene_MERVL_distance$bin <- classify_distance(gene_MERVL_distance$distance, ignore_direction = F)
# 
# #input chimeric splicing reads
# MT2_chimeric <- read.table("./R_output/TE_chimeric/231115_MT2_chimeric.txt", 
#                            header = T, sep = "\t")
# colnames(MT2_chimeric)[13] <- "id" 
# 
# #combine MERVLi and MERVLKD
# MERVL_2c <- merge(MERVLi_2c, MERVLKD_2c, by = "id", all = T)
# nrow(MERVL_2c) #26666
# MERVL_2c_down <- MERVL_2c[which(MERVL_2c$KD2C_DESeq_rpkm.group == "down-regulated" |
#                                 MERVL_2c$i2C_DESeq_rpkm.group == "down-regulated"), ]
# nrow(MERVL_2c_down) #389
# 
# 
# #merge gene and gene/MERVL distance
# MERVL_2c_down <- merge(gene_MERVL_distance,
#                               MERVL_2c_down[, c("id", "log2FoldChange.x", "log2FoldChange.y",
#                                               "KD2C_DESeq_rpkm.group", "i2C_DESeq_rpkm.group")],
#                               by = "id")
# MERVL_2c_down_30kb <- MERVL_2c_down[which(MERVL_2c_down$bin %in%
#                                             c("-0-10kb", "-10-30kb", "0-10kb",
#                                               "10-30kb"#, 
#                                               #"30-50kb", 
#                                               #"-30-50kb"
#                                               )),]
# nrow(MERVL_2c_down_30kb) #106
# 
# MERVL_2c_down_30kb_nonSJ <- subset(MERVL_2c_down_30kb, 
#                                  !(MERVL_2c_down_30kb$id %in% MT2_chimeric$id))
# 
# nrow(MERVL_2c_down_30kb_nonSJ) #48
# 
# MERVL_2c_down_30kb_SJ <- subset(MERVL_2c_down_30kb, 
#                               MERVL_2c_down_30kb$id %in% MT2_chimeric$id)
# nrow(MERVL_2c_down_30kb_SJ) #58
# 
# #manually annotate peak
# #this is because that package ChIPpeakAnno may annotate peak to other genes
# #but not the down-regulated genes
# 
# MERVL_2c_down_30kb_nonSJ$annotation <- rep("Distal", nrow(MERVL_2c_down_30kb_nonSJ))
# for(i in 1:nrow(MERVL_2c_down_30kb_nonSJ)){
#   #Proximal critera:
#   #1) same direction
#   #2) distance < 1kb from the TSS
#   if(MERVL_2c_down_30kb_nonSJ$gene_strand[i] == MERVL_2c_down_30kb_nonSJ$peak_strand[i]){
#     if(MERVL_2c_down_30kb_nonSJ$gene_strand[i] == "+"){
#        tmp_tss_lower <- MERVL_2c_down_30kb_nonSJ$gene_start[i] - 1000
#        tmp_tss_upper <- MERVL_2c_down_30kb_nonSJ$gene_start[i] + 500
#        if (tmp_tss_lower <= MERVL_2c_down_30kb_nonSJ$peak_end[i] &
#            tmp_tss_upper >= MERVL_2c_down_30kb_nonSJ$peak_start[i]){
#               MERVL_2c_down_30kb_nonSJ$annotation[i] <- "Proximal"
#            }
#     } else if (MERVL_2c_down_30kb_nonSJ$gene_strand[i] == "-"){
#        tmp_tss_lower <- MERVL_2c_down_30kb_nonSJ$gene_end[i] - 500
#        tmp_tss_upper <- MERVL_2c_down_30kb_nonSJ$gene_end[i] + 1000
#        if (tmp_tss_lower <= MERVL_2c_down_30kb_nonSJ$peak_end[i] &
#            tmp_tss_upper >= MERVL_2c_down_30kb_nonSJ$peak_start[i]){
#               MERVL_2c_down_30kb_nonSJ$annotation[i] <- "Proximal"
#        }
#     }
#   }
# }
# table(MERVL_2c_down_30kb_nonSJ$annotation)
# #Distal Proximal 
# #     30       18 
# 
# pdf("./figures/231115_MERVL_Proximal_distal_grouop.pdf", width = 6, height = 6)
# pie(c(18+58, 30), 
#     labels = c("MERVL_chimeric/promoter", "Distal"))
# dev.off()
```

### 8.2.4 K4/K27/ATAC at proximal/distal MT2
```{r}
# source("./utils.R")
# suppressMessages(library(rtracklayer))
# proximal_MT2 <- rbind(MT2_RNA_down_30kb_SJ[, c("peak_chr", "peak_start", "peak_end",
#                                                "peak_name", "peak_score", "peak_strand", "id")],
#                       MT2_RNA_down_30kb_nonSJ[which(MT2_RNA_down_30kb_nonSJ$annotation == "Proximal"),
#                                               c("peak_chr", "peak_start", "peak_end",
#                                                "peak_name", "peak_score", "peak_strand", "id")])
# distal_MT2 <- MT2_RNA_down_30kb_nonSJ[which(MT2_RNA_down_30kb_nonSJ$annotation == "Distal"),
#                                       c("peak_chr", "peak_start", "peak_end",
#                                                "peak_name", "peak_score", "peak_strand", "id")]
# nrow(proximal_MT2) #416
# nrow(proximal_MT2[!duplicated(proximal_MT2$peak_name),]) #393
# nrow(distal_MT2) #301
# nrow(distal_MT2[!duplicated(distal_MT2$peak_name),]) #272
# 
# bw_list <- list(
#   H3K4me3 = "/Volumes/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231031_remap_zhang_H3K4me3_embryo/bigwigs/K4me3_2c_merged.bw",
#   ATAC = "/Volumes/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231029_remap_L2C_ESC_ATAC/bigwigs/L2C_ATAC_merged.bw",
#   H3K27ac = "/Volumes/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231021_remap_H3K27ac_embryo/bigwigs/K27ac_l2c_mm10.sorted.multi.dedup.bw",
#   K4_ESC = "/Volumes/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231031_remap_zhang_H3K4me3_embryo/bigwigs/K4me3_ESC_mm10.sorted.multi.dedup.bw"
# )
# 
# df_list <- list(
#   proximal_MT2 =  proximal_MT2,
#   distal_MT2 = distal_MT2
# )
# 
# gr_list <- list()
# for (i in names(df_list)){
#   gr_list[[i]] <- GRanges(
#     seqnames = df_list[[i]]$peak_chr,
#     ranges = IRanges(start = df_list[[i]]$peak_start-500,
#                      end = df_list[[i]]$peak_end+500),
#     strand = df_list[[i]]$peak_strand,
#     peak_name = df_list[[i]]$peak_name, 
#     id = df_list[[i]]$id
#   )
# }
# 
# tmp_df <- data.frame()
# for (i in names(gr_list)){
#   tmp_matrix <- matrix(0, nrow = length(gr_list[[i]]),
#                           ncol = length(bw_list))
#   colnames(tmp_matrix) <- names(bw_list)
#   rownames(tmp_matrix) <- gr_list[[i]]$id
#   for(bw in names(bw_list)){
#     bw_file <- bw_list[[bw]]
#     tmp_matrix[,bw] <- calcPeaksSignal(gr_list[[i]], bw_file)$meanScore
#   }
#   tmp_df <- rbind(tmp_df, data.frame(tmp_matrix))
# }
# tmp_df$id <- row.names(tmp_df)
# 
# for (i in names(df_list)){
#   df_list[[i]] <- left_join(df_list[[i]], tmp_df, by = "id")
# }
# View(df_list$proximal_MT2)
# 
# boxplot(log2(df_list$proximal_MT2$H3K4me3+1),
#         log2(df_list$distal_MT2$H3K4me3+1), 
#         log2(df_list$distal_MT2$K4_ESC+1))
# boxplot(log2(df_list$proximal_MT2$H3K27ac+1),
#         log2(df_list$distal_MT2$H3K27ac+1))
# boxplot(log2(df_list$proximal_MT2$ATAC+1),
#         log2(df_list$distal_MT2$ATAC+1))
```


```{r}

# boxplot(log2(df_list$Obox_Dux_down$obox1+0.01),
#         #log2(df_list$Dux_down$obox1+0.01),
#         log2(df_list$Obox_down$obox1+0.01),
#         log2(df_list$Neither$obox1+0.01))
# boxplot(log2(df_list$Obox_Dux_down$obox3+0.01),
#         log2(df_list$Dux_down$obox3+0.01),
#         log2(df_list$Obox_down$obox3+0.01),
#         log2(df_list$Neither$obox3+0.01))
# boxplot(log2(df_list$Obox_Dux_down$obox5+0.01),
#         log2(df_list$Dux_down$obox5+0.01),
#         log2(df_list$Obox_down$obox5+0.01),
#         log2(df_list$Neither$obox5+0.01))
# boxplot(log2(df_list$Obox_Dux_down$dux1+0.01),
#         log2(df_list$Dux_down$dux1+0.01),
#         log2(df_list$Obox_down$dux1+0.01),
#         log2(df_list$Neither$dux1+0.01))
# boxplot(log2(df_list$Obox_Dux_down$dux2+0.01),
#         log2(df_list$Dux_down$dux2+0.01),
#         log2(df_list$Obox_down$dux2+0.01),
#         log2(df_list$Neither$dux2+0.01))
# boxplot(log2(df_list$Obox_Dux_down$dux2+0.01),
#         log2(df_list$Dux_down$dux3+0.01),
#         log2(df_list$Obox_down$dux3+0.01),
#         log2(df_list$Neither$dux3+0.01))
```

## 8.3 MT2 RNA/ATAC/K4/K27/Obox/Dux deeptool
```{bash}
module load bedtools/2.30.0
source ~/.bash_profile
conda activate deeptools

#Region of interest------------------------------
Region_dir="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/output"
FL_MERVL="${Region_dir}/full_length_MERVL.bed"
FL_MERVL_3primeMT2="${Region_dir}/full_length_MERVL_downMT2.bed"
FL_MERVL_5primeMT2="${Region_dir}/full_length_MERVL_upMT2.bed"
soloMT2="${Region_dir}/soloMT2.bed"
otherMT2="${Region_dir}/othersMT2.bed"

grep "MT2_Mm" $FL_MERVL_3primeMT2 > FL_MERVL_3primeMT2_Mm.bed
grep "MT2_Mm" $FL_MERVL_5primeMT2 > FL_MERVL_5primeMT2_Mm.bed
grep "MT2_Mm" $soloMT2 > soloMT2_Mm.bed
grep "MT2_Mm" $otherMT2 > otherMT2_Mm.bed

#splicing bed
Splice_dir="/users/chend4/ZYChenlab/Zhiyuan/projects/TE_project/230829_MT2_Mm_Integrative_analyses/deeptools/bed_input"
splice_3primeMT2="${Splice_dir}/FL_3MT2_Mm_2C_total_2c_SJ_filtered.bed"
splice_5primeMT2="${Splice_dir}/FL_5MT2_Mm_2C_total_2c_SJ_filtered.bed"
splice_soloMT2="${Splice_dir}/soloMT2_Mm_2C_total_2c_SJ_filtered.bed"
splice_otherMT2="${Splice_dir}/otherMT2_Mm_2C_total_2c_SJ_filtered.bed"

bedtools intersect -a FL_MERVL_3primeMT2_Mm.bed -b $splice_3primeMT2 -v > nonsplice_3primeMT2.bed
bedtools intersect -a FL_MERVL_5primeMT2_Mm.bed -b $splice_5primeMT2 -v > nonsplice_5primeMT2.bed
bedtools intersect -a soloMT2_Mm.bed -b $splice_soloMT2 -v > nonsplice_soloMT2.bed
bedtools intersect -a otherMT2_Mm.bed -b $splice_otherMT2 -v > nonsplice_otherMT2.bed
 
#total RNAseq------------------------------------
totalRNA_dir1="/data/ZYChenlab/Zhiyuan/projects/TE_project/230705_MT2Mmi_totalRNAseq/bigwigs"
total_polyA_2c_RNA="${totalRNA_dir1}/total_polyA_2C_merged.sorted.bw"
totalRNA_dir2="/data/ZYChenlab/Zhiyuan/projects/TE_project/231026_MT2splicing_analyses"
total_polyA_2c_splice="${totalRNA_dir2}/total_polyA_2c_spliced2.sorted.bw"

#K27ac--------------------------------------------
K27ac_dir="/users/chend4/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231021_remap_H3K27ac_embryo/bigwigs"
K27ac_l2c="${K27ac_dir}/K27ac_l2c_mm10.sorted.multi.dedup.bw"
K27ac_ESC="${K27ac_dir}/K27ac_ESC_mm10.sorted.multi.dedup.bw"

#K4me3--------------------------------------------
K4me3_dir="/users/chend4/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231031_remap_zhang_H3K4me3_embryo/bigwigs"
K4me3_2c="${K4me3_dir}/K4me3_2c_merged.bw"
K4me3_ESC="${K4me3_dir}/K4me3_ESC_mm10.sorted.multi.dedup.bw"

#ATAC---------------------------------------------
ATAC_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231029_remap_L2C_ESC_ATAC/bigwigs"
ATAC_2c="${ATAC_dir}/L2C_ATAC_merged.bw"
ATAC_ESC="${ATAC_dir}/ATAC_ESC_merged.bw"

#Obox--------------------------------------------
Obox_dir="/users/chend4/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231101_remap_Obox_stacc_seq/bigwigs"
obox1="${Obox_dir}/L2C_Obox1_mm10.sorted.multi.dedup.bw"
obox3="${Obox_dir}/L2C_Obox3_mm10.sorted.multi.dedup.bw"
obox5="${Obox_dir}/L2C_Obox5_mm10.sorted.multi.dedup.bw"
obox5mut="${Obox_dir}/L2C_Obox5R98E_mm10.sorted.multi.dedup.bw"

#Dux---------------------------------------------
Dux_dir="/users/chend4/ZYChenlab/Zhiyuan/projects/TE_project/drafts/231023_remap_Dux_ChIP/bigwigs"
Dux="${Dux_dir}/Dux_18hr_rep1_mm10.sorted.multi.dedup.bw"

computeMatrix scale-regions \
		-S $total_polyA_2c_RNA $total_polyA_2c_splice $K4me3_2c $K4me3_ESC $ATAC_2c $ATAC_ESC $K27ac_l2c $K27ac_ESC $obox1 $obox3 $obox5 $obox5mut $Dux \
		-R $splice_5primeMT2 $splice_3primeMT2 $splice_soloMT2 $splice_otherMT2 nonsplice_5primeMT2.bed nonsplice_3primeMT2.bed nonsplice_soloMT2.bed nonsplice_otherMT2.bed \
                -a 1000 -b 1000 \
                --regionBodyLength 500 \
		--outFileName MT2_Mm.1000flank_mat.gz \
		--sortRegions 'descend' \
		--samplesLabel 'RNA' 'splice' 'K4me3' 'K4me3_ESC' 'ATAC' 'ATAC_ESC' 'K27ac' 'K27ac_ESC' 'obox1' 'obox3' 'obox5' 'obox5mut' 'Dux'\
		--outFileSortedRegions MT2_Mm.1000flank_mat_sorted.bed \
		--outFileNameMatrix MT2_Mm.1000flank_mat.matrix \
		--missingDataAsZero \
                --sortUsingSamples 1

plotHeatmap --matrixFile MT2_Mm.1000flank_mat.gz \
                    --outFileName MT2_Mm.1000flank.pdf \
                    --dpi 300 \
                    --sortRegions "descend" \
                    --colorMap 'GnBu' 'GnBu'  'Blues' 'Blues' 'Blues' 'Blues' 'Blues' 'Blues' 'Blues' 'Blues' 'Blues' 'Blues' 'Blues' \
                    --boxAroundHeatmaps no \
                    --samplesLabel 'RNA' 'splice' 'K4me3' 'K4me3_ESC' 'ATAC' 'ATAC_ESC' 'K27ac' 'K27ac_ESC' 'obox1' 'obox3' 'obox5' 'obox5mut' 'Dux' \
                    --legendLocation "lower-center" \
                    --sortUsingSamples 3 \
		    --zMax 40 10 30 30 30 30 40 40 600 300 600 600 300 
exit;
```

## 8.4 ATAC-metaplot at MT2/MERVL
```{bash}
module load bedtools/2.30.0
source ~/.bash_profile
conda activate deeptools

#Region of interest------------------------------
Region_dir1="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/soloMT2_vs_MERVL/output"
FL_MERVL="${Region_dir1}/full_length_MERVL.bed"
soloMT2="${Region_dir1}/soloMT2.bed"

mm10_TE="/users/chend4/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed"
grep "L1_Mm" $mm10_TE > L1_Mm.bed

ATAC_dir="/data/ZYChenlab/Zhiyuan/projects/TE_project/drafts/230805_remap_p300_ATAC/bigwigs"
dmso_e2c="${ATAC_dir}/early2C_DMSO_ATAC_merged_multi.bw"
A485_e2c="${ATAC_dir}/early2C_A485_ATAC_merged_multi.bw"

computeMatrix scale-regions \
        -S $dmso_e2c $A485_e2c \
        -R $FL_MERVL \
        -a 5000 -b 5000 \
        --regionBodyLength 5000 \
        --outFileName MERVL.5000.flank_mat_ATAC.gz \
        --sortRegions 'descend' \
        --samplesLabel 'DMSO_e2c' 'A485_e2c' \
        --outFileSortedRegions MERVL.5000.flank_mat_sorted_ATAC.bed \
        --outFileNameMatrix  MERVL.5000.flank_mat_sorted_ATAC.matrix \
        --missingDataAsZero \
        --sortUsingSamples 1

plotHeatmap --matrixFile MERVL.5000.flank_mat_ATAC.gz \
        --outFileName MERVL.5000.flank_mat_ATAC.pdf \
        --dpi 300 \
        --sortRegions "descend" \
        --colorMap 'Blues' \
        --boxAroundHeatmaps no \
        --samplesLabel 'DMSO_e2c' 'A485_e2c' \
        --legendLocation "lower-center" \
        --sortUsingSamples 1
        --zMax 90 90

computeMatrix scale-regions \
        -S $dmso_e2c $A485_e2c \
        -R L1_Mm.bed \
        -a 5000 -b 5000 \
        --regionBodyLength 5000 \
        --outFileName L1Mm.5000.flank_mat_ATAC.gz \
        --sortRegions 'descend' \
        --samplesLabel 'DMSO_e2c' 'A485_e2c' \
        --outFileSortedRegions L1Mm.5000.flank_mat_sorted_ATAC.bed \
        --outFileNameMatrix  L1Mm.5000.flank_mat_sorted_ATAC.matrix \
        --missingDataAsZero \
        --sortUsingSamples 1

plotHeatmap --matrixFile L1Mm.5000.flank_mat_ATAC.gz \
        --outFileName L1Mm.5000.flank_mat_ATAC.pdf \
        --dpi 300 \
        --sortRegions "descend" \
        --colorMap 'Blues' \
        --boxAroundHeatmaps no \
        --samplesLabel 'DMSO_e2c' 'A485_e2c' \
        --legendLocation "lower-center" \
        --sortUsingSamples 1
        --zMax 90 90

computeMatrix scale-regions \
        -S $dmso_e2c $A485_e2c \
        -R $soloMT2 \
        -a 1000 -b 1000 \
        --regionBodyLength 500 \
        --outFileName MT2.1000.flank_mat_ATAC.gz \
        --sortRegions 'descend' \
        --samplesLabel 'DMSO_e2c' 'A485_e2c' \
        --outFileSortedRegions MT2.1000.flank_mat_sorted_ATAC.bed \
        --outFileNameMatrix  MT2.1000.flank_mat_sorted_ATAC.matrix \
        --missingDataAsZero \
        --sortUsingSamples 1

plotHeatmap --matrixFile MT2.1000.flank_mat_ATAC.gz \
        --outFileName MT2.5000.flank_mat_ATAC.pdf \
        --dpi 300 \
        --sortRegions "descend" \
        --colorMap 'Blues' \
        --boxAroundHeatmaps no \
        --samplesLabel 'DMSO_e2c' 'A485_e2c' \
        --legendLocation "lower-center" \
        --sortUsingSamples 1
        --zMax 90 90


exit;

```

Use Deeptool Matrix output as input for meta-plot analyses
```{r}
suppressMessages(library(ggplot2))
suppressMessages(library(cowplot))

#FL MERVL------------------------------------
MERVL <- read.table("MERVL.5000.flank_mat_sorted_ATAC.matrix", skip = 3, 
                 header = T)

MERVL_mean <- as.data.frame(colMeans(MERVL))
colnames(MERVL_mean) <- "average"
MERVL_mean$group <- c(rep("DMSO", 1500), rep("A485", 1500))
MERVL_mean$bp <- c(seq(1:1500), seq(1:1500))

L1Mm <- read.table("L1Mm.5000.flank_mat_sorted_ATAC.matrix", skip = 3, 
                    header = T)
L1Mm_mean <- as.data.frame(colMeans(L1Mm))
colnames(L1Mm_mean) <- "average"
L1Mm_mean$group <- c(rep("DMSO", 1500), rep("A485", 1500))
L1Mm_mean$bp <- c(seq(1:1500), seq(1:1500))


MT2 <- read.table("MT2.1000.flank_mat_sorted_ATAC.matrix", skip = 3, 
                   header = T)
MT2_mean <- as.data.frame(colMeans(MT2))
colnames(MT2_mean) <- "average"
MT2_mean$group <- c(rep("DMSO", 250), rep("A485", 250))
MT2_mean$bp <- c(seq(1:250), seq(1:250))



fig1 <- ggplot(MERVL_mean, aes(x = bp, y = average, color = group)) + 
  geom_line() + ylim(c(0,100)) + 
  scale_color_manual(values = c("grey", "black")) +
  theme_cowplot(16) + ggtitle("ATAC at FL MERVL (E2C)")
fig1

fig2 <- ggplot(L1Mm_mean, aes(x = bp, y = average, color = group)) + 
  geom_line() + ylim(c(0,100)) + 
  scale_color_manual(values = c("grey", "black")) +
  theme_cowplot(16) + ggtitle("ATAC at FL L1Mm (E2C)")
fig2

fig3 <- ggplot(MT2_mean, aes(x = bp, y = average, color = group)) + 
  geom_line() + ylim(c(0,100)) + 
  scale_color_manual(values = c("grey", "black")) +
  theme_cowplot(16) + ggtitle("ATAC at MT2 (E2C)")
fig3

pdf("231102_metaplot_ATAC_A485_E2C.pdf", width = 6, height = 4)
fig1
fig2
fig3
dev.off()

wilcox.test(x = MERVL_mean[which(MERVL_mean$group == "DMSO" & 
                                   MERVL_mean$bp >= 500 &
                                   MERVL_mean$bp <= 1000), "average"],
            y = MERVL_mean[which(MERVL_mean$group == "A485" &
                                   MERVL_mean$bp >= 500 &
                                   MERVL_mean$bp <= 1000), "average"])$p.value
#1.028703e-138

wilcox.test(x = MT2_mean[which(MT2_mean$group == "DMSO" & 
                                 MT2_mean$bp >= 100 &
                                 MT2_mean$bp <= 150), "average"],
            y = MT2_mean[which(MT2_mean$group == "A485" &
                                 MT2_mean$bp >= 100 &
                                 MT2_mean$bp <= 150), "average"])$p.value
#3.303682e-18    
```

# 9. Dux/Obox ananlyses

## 9.1 Dux/Obox RNAseq analyses
### 9.1.1 input count/FPKM
```{r}
suppressMessages(library(dplyr))
suppressMessages(library(rtracklayer))
source("./utils.R")

Obox_sampleName <- c("RNA_WT_e2c_rep1", "RNA_WT_e2c_rep2",
                      "RNA_mzKO_e2c_rep1", "RNA_mzKO_e2c_rep2",
                      "RNA_WT_l2c_rep1", "RNA_WT_l2c_rep2", 
                      "RNA_WT_l2c_rep3", "RNA_mzKO_l2c_rep1",
                      "RNA_mzKO_l2c_rep2", "RNA_mzKO_l2c_rep3")
Dux_sampleName <- c("Dux_WT_L2C_1", "Dux_WT_L2C_2", "Dux_WT_L2C_3",
                    "Dux_KO_L2C_1", "Dux_KO_L2C_2", "Dux_KO_L2C_3")
Obox_simpleName <- c("Obox_WT_e2c_1", "Obox_WT_e2c_2",
                     "Obox_mzKO_e2c_1", "Obox_mzKO_e2c_2",
                     "Obox_WT_l2c_1", "Obox_WT_l2c_2", "Obox_WT_l2c_3",
                     "Obox_mzKO_l2c_1", "Obox_mzKO_l2c_2", "Obox_mzKO_l2c_3")
Dux_simpleName <- Dux_sampleName

GuoDux_sampleName <- c("e2c_wt_1", "e2c_wt_2", "e2c_wt_3", "e2c_wt_4", 
                       "e2c_wt_5", "e2c_wt_6", 
                       "e2c_ko_1", "e2c_ko_2", "e2c_ko_3", "e2c_ko_4",
                       "l2c_wt_1", "l2c_wt_2", "l2c_wt_3", "l2c_wt_4",
                       "l2c_wt_5",
                       "l2c_ko_1", "l2c_ko_2", "l2c_ko_3", "l2c_ko_4",
                       "l2c_ko_5")
Guo_Dux_simpleName <- GuoDux_sampleName

Obox_counts <- inputTEcountfiles(Obox_sampleName, paste0(Obox_simpleName, ".count"), 
                            countDataPath = "./R_input/Obox_FPKM_TEcount/")
row.names(Obox_counts) <- Obox_counts$id

Dux_counts <- inputTEcountfiles(Dux_sampleName, paste0(Dux_sampleName, ".count"), 
                            countDataPath = "./R_input/Chen_Dux_FPKM_TEcount/")
row.names(Dux_counts) <- Dux_counts$id

GuoDux_counts <- inputTEcountfiles(GuoDux_sampleName, paste0(GuoDux_sampleName, ".count"), 
                            countDataPath = "./R_input/Guo_Dux_FPKM_TEcount/")
row.names(GuoDux_counts) <- GuoDux_counts$id

GuoDux_fpkm <- inputStringTieRPKMfiles(GuoDux_sampleName, paste0(GuoDux_sampleName, ".rpkm"),
                                RPKMDataPath = "./R_input/Guo_Dux_FPKM_TEcount/")
Obox_fpkm <- inputStringTieRPKMfiles(Obox_sampleName, paste0(Obox_simpleName, ".rpkm"),
                                RPKMDataPath = "./R_input/Obox_FPKM_TEcount/")
write.csv(Obox_fpkm, "./R_output/Obox_Dux_RNA/231116_Obox_fpkm.csv", quote = F, row.names = F)
write.csv(GuoDux_fpkm, "./R_output/Obox_Dux_RNA/231116_GuoDux_fpkm.csv", quote = F, row.names = F)
```

### 9.1.2 DESeq2 analyses
```{r DEG identification}
foldchange = 2

suppressMessages(
  GuoDux_e2c_DESeq <- countsToDEseq2FDR(
    counts = GuoDux_counts[, c("e2c_wt_1.count", "e2c_wt_2.count", "e2c_wt_3.count",
                               "e2c_wt_4.count", "e2c_wt_5.count", "e2c_wt_6.count",
                               "e2c_ko_1.count", "e2c_ko_2.count", "e2c_ko_3.count",
                               "e2c_ko_4.count")], 
                                 CGroup = 6, TGroup = 4))
suppressMessages(
  GuoDux_l2c_DESeq <- countsToDEseq2FDR(
    counts = GuoDux_counts[, c("l2c_wt_1.count", "l2c_wt_2.count", "l2c_wt_3.count",
                               "l2c_wt_4.count", "l2c_wt_5.count", "l2c_ko_1.count",
                               "l2c_ko_2.count", "l2c_ko_3.count", "l2c_ko_4.count",
                               "l2c_ko_5.count")], 
                                 CGroup = 5, TGroup = 5))

suppressMessages(
  Obox_e2c_DESeq <- countsToDEseq2FDR(counts = Obox_counts[, c("Obox_WT_e2c_1.count", "Obox_WT_e2c_2.count",
                                                     "Obox_mzKO_e2c_1.count", "Obox_mzKO_e2c_2.count")], 
                                 CGroup = 2, TGroup = 2)
)

suppressMessages(
  Obox_l2c_DESeq <- countsToDEseq2FDR(counts = Obox_counts[, c( "Obox_WT_l2c_1.count", "Obox_WT_l2c_2.count",
                                                     "Obox_WT_l2c_3.count", 
                                                     "Obox_mzKO_l2c_1.count", "Obox_mzKO_l2c_2.count",
                                                     "Obox_mzKO_l2c_3.count")], 
                                 CGroup = 3, TGroup = 3)
)

suppressMessages(
  Dux_l2c_DESeq <- countsToDEseq2FDR(counts = Dux_counts[, c( "Dux_WT_L2C_1.count", "Dux_WT_L2C_2.count",
                                                     "Dux_WT_L2C_3.count", 
                                                     "Dux_KO_L2C_1.count", "Dux_KO_L2C_2.count",
                                                     "Dux_KO_L2C_3.count")], 
                                 CGroup = 3, TGroup = 3)
)

#classify DEGs 
GuoDux_e2c_DESeq_rpkm <- merge(
  GuoDux_e2c_DESeq[, c("id", "e2c_wt_1.count", "e2c_wt_2.count", "e2c_wt_3.count",
                               "e2c_wt_4.count", "e2c_wt_5.count", "e2c_wt_6.count",
                               "e2c_ko_1.count", "e2c_ko_2.count", "e2c_ko_3.count",
                               "e2c_ko_4.count", "log2FoldChange", "padj")],
  GuoDux_fpkm[, c("id", "name",
                  "e2c_wt_1.rpkm", "e2c_wt_2.rpkm", "e2c_wt_3.rpkm", "e2c_wt_4.rpkm",
                  "e2c_wt_5.rpkm", "e2c_wt_6.rpkm", "e2c_ko_1.rpkm", "e2c_ko_2.rpkm",
                  "e2c_ko_3.rpkm", "e2c_ko_4.rpkm")]
)

GuoDux_l2c_DESeq_rpkm <- merge(
  GuoDux_l2c_DESeq[, c("id", "l2c_wt_1.count", "l2c_wt_2.count",
                       "l2c_wt_3.count", "l2c_wt_4.count", "l2c_wt_5.count",
                       "l2c_ko_1.count", "l2c_ko_2.count", "l2c_ko_3.count",
                       "l2c_ko_4.count", "l2c_ko_5.count", "log2FoldChange", "padj")],
  GuoDux_fpkm[, c("id", "name",
                  "l2c_wt_1.rpkm", "l2c_wt_2.rpkm", "l2c_wt_3.rpkm", "l2c_wt_4.rpkm",
                  "l2c_wt_5.rpkm", "l2c_ko_1.rpkm", "l2c_ko_2.rpkm", "l2c_ko_3.rpkm",
                  "l2c_ko_4.rpkm", "l2c_ko_5.rpkm")])

Obox_e2c_DESeq_rpkm <- merge(Obox_e2c_DESeq[, c("id",
                                      "Obox_WT_e2c_1.count", "Obox_WT_e2c_2.count",
                                      "Obox_mzKO_e2c_1.count", "Obox_mzKO_e2c_2.count",
                                      "log2FoldChange", "padj")],
                        Obox_fpkm[, c("id", "name", 
                                 "Obox_WT_e2c_1.rpkm", "Obox_WT_e2c_2.rpkm", 
                                 "Obox_mzKO_e2c_1.rpkm", "Obox_mzKO_e2c_2.rpkm")],
                        by = "id"
                        )

Obox_l2c_DESeq_rpkm <- merge(Obox_l2c_DESeq[, c("id", 
                                    "Obox_WT_l2c_1.count", "Obox_WT_l2c_2.count",
                                    "Obox_WT_l2c_3.count", "Obox_mzKO_l2c_1.count",
                                    "Obox_mzKO_l2c_2.count", 
                                    "Obox_mzKO_l2c_3.count",
                                    "log2FoldChange", "padj")],
                       Obox_fpkm[, c("id", "name", 
                                "Obox_WT_l2c_1.rpkm", "Obox_WT_l2c_2.rpkm",
                                "Obox_WT_l2c_3.rpkm", "Obox_mzKO_l2c_1.rpkm", "Obox_mzKO_l2c_2.rpkm",
                                "Obox_mzKO_l2c_3.rpkm")],
                       by = "id")

#since I'm not gonna to use RPKM for cutoff, i use Obox_fpkm to allow function running
Dux_l2c_DESeq_rpkm <- merge(Dux_l2c_DESeq[, c("id", "Dux_WT_L2C_1.count", "Dux_WT_L2C_2.count",
                                                     "Dux_WT_L2C_3.count", 
                                                     "Dux_KO_L2C_1.count", "Dux_KO_L2C_2.count",
                                                     "Dux_KO_L2C_3.count",
                                         "log2FoldChange", "padj")],
                       Obox_fpkm[, c("id", "name", "Obox_WT_l2c_1.rpkm", "Obox_WT_l2c_2.rpkm")])

Obox_e2c_DESeq_rpkm$padj[is.na(Obox_e2c_DESeq_rpkm$padj)] <- 1
Obox_l2c_DESeq_rpkm$padj[is.na(Obox_l2c_DESeq_rpkm$padj)] <- 1
Dux_l2c_DESeq_rpkm$padj[is.na(Dux_l2c_DESeq_rpkm$padj)] <- 1
GuoDux_e2c_DESeq_rpkm$padj[is.na(GuoDux_e2c_DESeq_rpkm$padj)] <- 1
GuoDux_l2c_DESeq_rpkm$padj[is.na(GuoDux_l2c_DESeq_rpkm$padj)] <- 1

GuoDux_e2c_DESeq_rpkm.group <- classifyDEG(
  GuoDux_e2c_DESeq_rpkm,
  ctr.rpkm = c("e2c_wt_1.rpkm", "e2c_wt_2.rpkm", "e2c_wt_3.rpkm", 
               "e2c_wt_5.rpkm", "e2c_wt_6.rpkm"),
  trt.rpkm = c("e2c_ko_1.rpkm", "e2c_ko_2.rpkm", "e2c_ko_3.rpkm",
               "e2c_ko_4.rpkm"),
   FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = 0, log2FC = log2(foldchange), FDR = 0.05)
table(GuoDux_e2c_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#           370          20963            164 

GuoDux_l2c_DESeq_rpkm.group <- classifyDEG(
  GuoDux_l2c_DESeq_rpkm,
  ctr.rpkm = c("l2c_wt_1.rpkm", "l2c_wt_2.rpkm", "l2c_wt_3.rpkm", 
               "l2c_wt_5.rpkm"),
  trt.rpkm = c("l2c_ko_1.rpkm", "l2c_ko_2.rpkm", "l2c_ko_3.rpkm",
               "l2c_ko_4.rpkm", "l2c_ko_5.rpkm"),
   FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = 0, log2FC = log2(foldchange), FDR = 0.05)
table(GuoDux_l2c_DESeq_rpkm.group)
#GuoDux_l2c_DESeq_rpkm.group
#down-regulated  similar_level   up-regulated 
#          1200          21999           1420 

Obox_e2c_DESeq_rpkm.group <- classifyDEG(
                            Obox_e2c_DESeq_rpkm,
                            ctr.rpkm = c("Obox_WT_e2c_1.rpkm", "Obox_WT_e2c_2.rpkm"),
                            trt.rpkm = c("Obox_mzKO_e2c_1.rpkm", "Obox_mzKO_e2c_2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = 0, log2FC = log2(foldchange), FDR = 0.05)

table(Obox_e2c_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#          434          16963              2 

Obox_l2c_DESeq_rpkm.group <- classifyDEG(Obox_l2c_DESeq_rpkm,
                            ctr.rpkm = c("Obox_WT_l2c_1.rpkm", "Obox_WT_l2c_2.rpkm", "Obox_WT_l2c_3.rpkm"),
                            trt.rpkm = c("Obox_mzKO_l2c_1.rpkm", "Obox_mzKO_l2c_2.rpkm", "Obox_mzKO_l2c_3.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = 0, log2FC = log2(foldchange), FDR = 0.05)
table(Obox_l2c_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#         2589          15924           1923

#since I'm not gonna  use RPKM for cutoff, i use Obox_fpkm to allow function running
Dux_l2c_DESeq_rpkm.group <- classifyDEG(Dux_l2c_DESeq_rpkm,
                            ctr.rpkm = c("Obox_WT_l2c_1.rpkm"),
                            trt.rpkm = c("Obox_WT_l2c_2.rpkm"),
                            FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = 0, log2FC = log2(foldchange), FDR = 0.05)
table(Dux_l2c_DESeq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#           625          30199            211

Obox_e2c_DESeq_rpkm <- cbind(Obox_e2c_DESeq_rpkm, Obox_e2c_DESeq_rpkm.group)
Obox_l2c_DESeq_rpkm <- cbind(Obox_l2c_DESeq_rpkm, Obox_l2c_DESeq_rpkm.group)
Dux_l2c_DESeq_rpkm <- cbind(Dux_l2c_DESeq_rpkm, Dux_l2c_DESeq_rpkm.group)
GuoDux_e2c_DESeq_rpkm <- cbind(GuoDux_e2c_DESeq_rpkm, GuoDux_e2c_DESeq_rpkm.group)
GuoDux_l2c_DESeq_rpkm <- cbind(GuoDux_l2c_DESeq_rpkm, GuoDux_l2c_DESeq_rpkm.group)

Obox_e2c_DESeq_rpkm.up <- Obox_e2c_DESeq_rpkm[Obox_e2c_DESeq_rpkm.group == "up-regulated",]
Obox_e2c_DESeq_rpkm.down <- Obox_e2c_DESeq_rpkm[Obox_e2c_DESeq_rpkm.group == "down-regulated",]
Obox_e2c_DESeq_rpkm.detectable <- Obox_e2c_DESeq_rpkm[Obox_e2c_DESeq_rpkm.group != "low_expression_level",]

Obox_l2c_DESeq_rpkm.up <- Obox_l2c_DESeq_rpkm[Obox_l2c_DESeq_rpkm.group == "up-regulated",]
Obox_l2c_DESeq_rpkm.down <- Obox_l2c_DESeq_rpkm[Obox_l2c_DESeq_rpkm.group == "down-regulated",]
Obox_l2c_DESeq_rpkm.detectable <- Obox_l2c_DESeq_rpkm[Obox_l2c_DESeq_rpkm.group != "low_expression_level",]

Dux_l2c_DESeq_rpkm.up <- Dux_l2c_DESeq_rpkm[Dux_l2c_DESeq_rpkm.group == "up-regulated",]
Dux_l2c_DESeq_rpkm.down <- Dux_l2c_DESeq_rpkm[Dux_l2c_DESeq_rpkm.group == "down-regulated",]
Dux_l2c_DESeq_rpkm.detectable <- Dux_l2c_DESeq_rpkm[Dux_l2c_DESeq_rpkm.group != "low_expression_level",]

nrow(Obox_e2c_DESeq_rpkm.up) #2
nrow(Obox_e2c_DESeq_rpkm.down) #434
nrow(Obox_e2c_DESeq_rpkm.detectable) #17399

nrow(Obox_l2c_DESeq_rpkm.up) #19233
nrow(Obox_l2c_DESeq_rpkm.down) #2589
nrow(Obox_l2c_DESeq_rpkm.detectable) #20436

nrow(Dux_l2c_DESeq_rpkm.up) #211
nrow(Dux_l2c_DESeq_rpkm.down) #625
nrow(Dux_l2c_DESeq_rpkm.detectable) #31035

Obox_e2c_DESeq_rpkm$log2WT <- log2(
  (Obox_e2c_DESeq_rpkm$Obox_WT_e2c_1.count + Obox_e2c_DESeq_rpkm$Obox_WT_e2c_2.count) / 2 + 1
)

Obox_e2c_DESeq_rpkm$log2mzKO <- log2(
  (Obox_e2c_DESeq_rpkm$Obox_mzKO_e2c_1.count + Obox_e2c_DESeq_rpkm$Obox_mzKO_e2c_2.count) /2 + 1
)

Obox_e2c_DESeq_rpkm.up.label <- paste0("Up-regulated:\n", nrow(Obox_e2c_DESeq_rpkm.up))
Obox_e2c_DESeq_rpkm.down.label <- paste0("Down-regulated:\n", nrow(Obox_e2c_DESeq_rpkm.down))

Obox_l2c_DESeq_rpkm$log2WT <- log2(
  (Obox_l2c_DESeq_rpkm$Obox_WT_l2c_1.count + Obox_l2c_DESeq_rpkm$Obox_WT_l2c_2.count + Obox_l2c_DESeq_rpkm$Obox_WT_l2c_3.count) / 3 + 1
)

Obox_l2c_DESeq_rpkm$log2mzKO <- log2(
  (Obox_l2c_DESeq_rpkm$Obox_mzKO_l2c_1.count + Obox_l2c_DESeq_rpkm$Obox_mzKO_l2c_2.count + Obox_l2c_DESeq_rpkm$Obox_mzKO_l2c_3.count) / 3 + 1
)

Obox_l2c_DESeq_rpkm.up.label <- paste0("Up-regulated:\n", nrow(Obox_l2c_DESeq_rpkm.up))
Obox_l2c_DESeq_rpkm.down.label <- paste0("Down-regulated:\n", nrow(Obox_l2c_DESeq_rpkm.down))

Dux_l2c_DESeq_rpkm$log2WT <- log2(
  (Dux_l2c_DESeq_rpkm$Dux_WT_L2C_1.count + Dux_l2c_DESeq_rpkm$Dux_WT_L2C_2.count + Dux_l2c_DESeq_rpkm$Dux_WT_L2C_3.count) / 3 + 1
)
Dux_l2c_DESeq_rpkm$log2KO <- log2(
  (Dux_l2c_DESeq_rpkm$Dux_KO_L2C_1.count + Dux_l2c_DESeq_rpkm$Dux_KO_L2C_2.count + Dux_l2c_DESeq_rpkm$Dux_KO_L2C_3.count) / 3 + 1
)

Dux_l2c_DESeq_rpkm.up.label <- paste0("Up-regulated:\n", nrow(Dux_l2c_DESeq_rpkm.up))
Dux_l2c_DESeq_rpkm.down.label <- paste0("Down-regulated:\n", nrow(Dux_l2c_DESeq_rpkm.down))

# pdf("../figures/231109_Obox_Dux_genes_DE_scatterplot.pdf", width = 6, height = 5)
# ggScatterplot(Obox_e2c_DESeq_rpkm, x = "log2WT", y = "log2mzKO",
#                        group = "Obox_e2c_DESeq_rpkm.group", gene = "name", xlab = "Obox WT",
#                        ylab = "Obox mzKO",
#                        title = "Obox (genes)(e2C)",
#                        my.color = c("blue", "grey50", "red3"),
#                        label.up = Obox_e2c_DESeq_rpkm.up.label,
#                        label.down = Obox_e2c_DESeq_rpkm.down.label,
#                        genes4Label = c("Zfp352", "Zfp54", "Gm8994", "Fundc1", "Ddit4l", "Gm11517"),
#                        FC.line = foldchange)
# 
# ggScatterplot(Obox_l2c_DESeq_rpkm, x = "log2WT", y = "log2mzKO",
#                        group = "Obox_l2c_DESeq_rpkm.group", gene = "name", xlab = "Obox WT",
#                        ylab = "Obox mzKO",
#                        title = "Obox (genes)(l2C)",
#                        my.color = c("blue", "grey50", "red3"),
#                        label.up = Obox_l2c_DESeq_rpkm.up.label,
#                        label.down = Obox_l2c_DESeq_rpkm.down.label,
#                        genes4Label = c("Gm11517", "Gm3139", "Rnf11", "Spic", "Npl", "Hipk1"),
#                        FC.line = foldchange)
# ggScatterplot(Dux_l2c_DESeq_rpkm, x = "log2WT", y = "log2KO",
#                        group = "Dux_l2c_DESeq_rpkm.group", gene = "name", xlab = "Dux WT",
#                        ylab = "Dux mzKO",
#                        title = "Dux (genes)(l2C)",
#                        my.color = c("blue", "grey50", "red3"),
#                        label.up = Dux_l2c_DESeq_rpkm.up.label,
#                        label.down = Dux_l2c_DESeq_rpkm.down.label,
#                        genes4Label = c("Gm11517", "Gm3139", "Rnf11", "Spic", "Npl", "Hipk1"),
#                        FC.line = foldchange)
# dev.off()
write.table(Obox_e2c_DESeq_rpkm, "./R_output/Obox_Dux_RNA/231109_Obox_E2C_DEG_analyses.csv", 
            quote = F, row.names = F, sep = ",")
write.table(Obox_l2c_DESeq_rpkm, "./R_output/Obox_Dux_RNA/231109_Obox_L2C_DEG_analyses.csv", 
            quote = F, row.names = F, sep = ",")
write.table(Dux_l2c_DESeq_rpkm, "./R_output/Obox_Dux_RNA/231109_Dux_L2C_DEG_analyses.csv", 
            quote = F, row.names = F, sep = ",")
write.table(GuoDux_e2c_DESeq_rpkm, "./R_output/Obox_Dux_RNA/231110_GuoDux_E2C_DEG_analyses.csv",
            quote = F, row.names = F, sep = ",")
write.table(GuoDux_l2c_DESeq_rpkm, "./R_output/Obox_Dux_RNA/231110_GuoDux_L2C_DEG_analyses.csv",
            quote = F, row.names = F, sep = ",")
table(Obox_e2c_DESeq_rpkm$Obox_e2c_DESeq_rpkm.group)
table(Obox_l2c_DESeq_rpkm$Obox_l2c_DESeq_rpkm.group)
table(Dux_l2c_DESeq_rpkm$Dux_l2c_DESeq_rpkm.group)
table(GuoDux_e2c_DESeq_rpkm$GuoDux_e2c_DESeq_rpkm.group)
table(GuoDux_l2c_DESeq_rpkm$GuoDux_l2c_DESeq_rpkm.group)
```

### 9.1.3 MT2/MERVL expression upon Obox/Dux KO
```{r}
suppressMessages(library(ggplot2))
suppressMessages(library(reshape2))
suppressMessages(library(tidyr))
suppressMessages(library(cowplot))
suppressMessages(library(pheatmap))

Obox_e2c_repeats <- Obox_e2c_DESeq[grep(":", Obox_e2c_DESeq$id),]
Obox_l2c_repeats <- Obox_l2c_DESeq[grep(":", Obox_l2c_DESeq$id),]
Dux_l2c_repeats <- Dux_l2c_DESeq[grep(":", Dux_l2c_DESeq$id),]
GuoDux_e2c_repeats <- GuoDux_e2c_DESeq[grep(":",GuoDux_e2c_DESeq$id),]
GuoDux_l2c_repeats <- GuoDux_l2c_DESeq[grep(":", GuoDux_l2c_DESeq$id),]

GuoDux_e2c_MT2Mm <- GuoDux_e2c_repeats[which(GuoDux_e2c_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR",
                                                                    "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR",
                                                                    "MER89:ERV1:LTR")),
                                       c("id","log2FoldChange")]
GuoDux_l2c_MT2Mm <- GuoDux_l2c_repeats[which(GuoDux_l2c_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR",
                                                                    "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR",
                                                                    "MER89:ERV1:LTR")),
                                       c("id","log2FoldChange")]

Obox_e2c_MT2Mm <- Obox_e2c_repeats[which(Obox_e2c_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR",
                                                                    "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR",
                                                                    "MER89:ERV1:LTR")),
                         c("id","log2FoldChange")]

Obox_l2c_MT2Mm <- Obox_l2c_repeats[which(Obox_l2c_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR",
                                                                    "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR",
                                                                    "MER89:ERV1:LTR")),
                        c("id","log2FoldChange")]

Dux_l2c_MT2Mm <- Dux_l2c_repeats[which(Dux_l2c_repeats$id %in% c("MT2_Mm:ERVL:LTR", "MERVL-int:ERVL:LTR",
                                                                    "MT2C_Mm:ERVL:LTR", "MERVL_2A-int:ERVL:LTR",
                                                                    "MER89:ERV1:LTR")),
                        c("id", "log2FoldChange")]

pdf("./figures/231116_Obox_Dux_repeat_heatmap.pdf", height = 6, width = 5)
pheatmap(as.matrix(Obox_e2c_MT2Mm[, "log2FoldChange"], row.names = Obox_e2c_MT2Mm$id), cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T)
pheatmap(as.matrix(Obox_l2c_MT2Mm[, "log2FoldChange"], row.names = Obox_l2c_MT2Mm$id), cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T)
pheatmap(as.matrix(Dux_l2c_MT2Mm[, "log2FoldChange"], row.names = Dux_l2c_MT2Mm$id), cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T)
pheatmap(as.matrix(GuoDux_e2c_MT2Mm[, "log2FoldChange"], row.names = GuoDux_e2c_MT2Mm$id),
         cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T)
pheatmap(as.matrix(GuoDux_l2c_MT2Mm[, "log2FoldChange"], row.names = GuoDux_l2c_MT2Mm$id),
         cluster_rows = F, cluster_cols = F,
         color = colorRampPalette(rev(c("white", "blue")))(100), 
         breaks = seq(-5, 0, by = 0.05), show_rownames = T)
dev.off()
```
### 9.1.4 Obox/Dux/MT2i overlap analyses

Since Dux/Obox not only bind to MT2 but also other places, here only consider 
genes that <30kb to MT2

```{r}
source("./utils.R")
gene_MT2_distance <- read.table("./R_input/dCas9_ChIPseq/231115_nearest_gene2MT2.txt",
                                         header = F, sep = "\t")
nrow(gene_MT2_distance) #55915
colnames(gene_MT2_distance) <- c("gene_chr", "gene_start", "gene_end",
                                   "id", "name", "gene_strand",
                                   "peak_chr", "peak_start", "peak_end",
                                   "peak_name", "peak_score", "peak_strand",
                                   "distance")
#for genes with multiple peaks in the gene bodies, randomly pick one
gene_MT2_distance <- gene_MT2_distance[!duplicated(gene_MT2_distance$id),]
nrow(gene_MT2_distance) #55401

gene_MT2_distance$bin <- classify_distance(as.numeric(gene_MT2_distance$distance), ignore_direction = T)
```

```{r}
#input MT2 down genes
MT2_e2c_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_e2C_genes_DE_analyses.csv", sep = ",", header = T)
MT2_l2c_RNA <- read.table("./R_output/MT2i_totalRNA/231115_MT2i_2C_genes_DE_analyses.csv", sep = ",", header = T)

MT2_RNA <- merge(MT2_e2c_RNA, MT2_l2c_RNA, by = "id", all = T)
MT2_RNA <- left_join(MT2_RNA, gene_MT2_distance, by = "id")

MT2_down <- MT2_RNA[which(MT2_RNA$c2_DESeq_rpkm.group == "down-regulated" |
                          MT2_RNA$ec2_DESeq_rpkm.group == "down-regulated"),]
nrow(MT2_down) #1155
length(unique(MT2_down$id))


MT2_down_30kb <- MT2_down[which(MT2_down$bin %in% c("0-10kb", "10-30kb")),]
nrow(MT2_down_30kb) #717


```

```{r}
Obox_e2c_RNA <- read.table("./R_output/Obox_Dux_RNA/231109_Obox_E2C_DEG_analyses.csv", sep = ",",
                           header = T)
Obox_l2c_RNA <- read.table("./R_output/Obox_Dux_RNA/231109_Obox_L2C_DEG_analyses.csv", sep = ",",
                          header = T)
Obox_RNA <- merge(Obox_e2c_RNA, Obox_l2c_RNA, by = "id", all = T)
Obox_RNA <- left_join(Obox_RNA, gene_MT2_distance, by = "id")
Obox_down <- Obox_RNA[which(Obox_RNA$Obox_l2c_DESeq_rpkm.group == "down-regulated" |
                            Obox_RNA$Obox_e2c_DESeq_rpkm.group == "down-regulated"),]
nrow(Obox_down) #2777
length(unique(Obox_down$id))
Obox_down_30kb <- Obox_down[which(Obox_down$bin %in% c("0-10kb", "10-30kb")),]
nrow(Obox_down_30kb) #647

Dux_RNA <- read.table("./R_output/Obox_Dux_RNA/231109_Dux_L2C_DEG_analyses.csv", sep = ",",
                         header = T)
GuoDux_e2c_RNA <- read.table("./R_output/Obox_Dux_RNA/231110_GuoDux_E2C_DEG_analyses.csv", sep = ",",
                             header = T)
GuoDux_l2c_RNA <- read.table("./R_output/Obox_Dux_RNA/231110_GuoDux_L2C_DEG_analyses.csv", sep = ",",
                             header = T)
Dux_RNA_all <- merge(Dux_RNA, GuoDux_e2c_RNA, by = "id", all = T)
Dux_RNA_all <- merge(Dux_RNA_all, GuoDux_l2c_RNA, by = "id", all = T)
Dux_RNA_all <- left_join(Dux_RNA_all, gene_MT2_distance, by = "id")

Dux_down <-  Dux_RNA_all[which(Dux_RNA_all$GuoDux_l2c_DESeq_rpkm.group == "down-regulated" |
                               Dux_RNA_all$GuoDux_e2c_DESeq_rpkm.group == "down-regulated" |
                               Dux_RNA_all$Dux_l2c_DESeq_rpkm.group == "down-regulated"),]
nrow(Dux_down) #1982
length(unique(Dux_down$id)) #1982
Dux_down_30kb <- Dux_down[which(Dux_down$bin %in% c("0-10kb", "10-30kb")),]
nrow(Dux_down_30kb) #411
```

```{r}
suppressMessages(library(dplyr))
suppressMessages(library(VennDiagram))
suppressMessages(library(ggsci))
suppressMessages(library(eulerr))
Obox_Dux_MT2_30kb_venn <- venn.diagram(x = list(
                      "Dux down" = unique(Dux_down_30kb$id),
                      "Obox down" = unique(Obox_down_30kb$id),
                      "MT2 down" = unique(MT2_down_30kb$id)
                      ),
                      filename = NULL,
                           main="minorZGA",
                            resolution = 300,
                           #fill=ggsci::pal_futurama(3),
                           width = 800,height = 800,
                           euler.d = TRUE, cex=2,
                           cat.cex=0.8,scaled = TRUE,
                           offset=0.5, lwd=1,margin=0)
grid.newpage()
grid.draw(Obox_Dux_MT2_30kb_venn)

Obox_Dux_MT2_ZGA_venn2 <- euler(c(Dux = 159,
                      Obox = 216,
                      MT2 = 347,
                      "Dux&Obox" = 102,
                      "Dux&MT2" = 41,
                      "MT2&Obox" =220,
                      "Obox&Dux&MT2"=109))
plot(Obox_Dux_MT2_ZGA_venn2)
pdf("./figures/231116_MT2i_Obox_Dux_venn.pdf", width = 6, height = 6)
grid.newpage()
grid.draw(Obox_Dux_MT2_30kb_venn)
plot(Obox_Dux_MT2_ZGA_venn2)
dev.off()

MT2_Dux_down_30kb <- MT2_down_30kb[which(MT2_down_30kb$id %in% Dux_down_30kb$id),]
MT2_Dux_Obox_30kb <- MT2_Dux_down_30kb[which(MT2_Dux_down_30kb$id %in% Obox_down_30kb$id),]
nrow(MT2_Dux_Obox_30kb) #109

MT2_Dux_down_30kb_only <- MT2_Dux_down_30kb[!(MT2_Dux_down_30kb$id %in% MT2_Dux_Obox_30kb$id),]
nrow(MT2_Dux_down_30kb_only) #41

MT2_Obox_down_30kb <- MT2_down_30kb[which(MT2_down_30kb$id %in% Obox_down_30kb$id),]
MT2_Obox_down_30kb_only <- MT2_Obox_down_30kb[!(MT2_Obox_down_30kb$id %in% MT2_Dux_Obox_30kb$id),]
nrow(MT2_Obox_down_30kb_only) #220

Neither <- MT2_down_30kb[!(MT2_down_30kb$id %in% MT2_Dux_down_30kb$id),]
Neither <- Neither[!(Neither$id %in% MT2_Obox_down_30kb$id),]
nrow(Neither) #347

Obox_down_only <- Obox_down_30kb[!(Obox_down_30kb$id %in% MT2_down_30kb$id),]
Obox_down_only <- Obox_down_only[!(Obox_down_only$id %in% Dux_down_30kb$id),]
nrow(Obox_down_only) #216
Dux_down_only <- Dux_down_30kb[!(Dux_down_30kb$id %in% MT2_down_30kb$id),]
Dux_down_only <- Dux_down_only[!(Dux_down_only$id %in% Obox_down_30kb$id),]
nrow(Dux_down_only) #159

Dux_Obox_down_notMT2 <- Dux_down_30kb[which(Dux_down_30kb$id %in% Obox_down_30kb$id),]
nrow(Dux_Obox_down_notMT2) #211
Dux_Obox_down_notMT2 <- Dux_Obox_down_notMT2[!(Dux_Obox_down_notMT2$id %in% MT2_down_30kb$id),]
nrow(Dux_Obox_down_notMT2) #102
```

## 9.2 Dux/Obox binding analyses

### 9.2.1 input mm10 annotation
```{r input mm10 repeats}
suppressMessages(library("rtracklayer"))

# import mm10 repeat masker bed file
mm10_rmsk <- import.bed("/Volumes/ZYChenlab/Zhiyuan/genomes_annotations/mm10/annotations/mm10_rmsk_TE.bed")
length(mm10_rmsk) #3725827

# get MT2_Mm & MT2C_Mm
MT2Mm <- mm10_rmsk[grep("MT2_Mm", mm10_rmsk$name),]
MT2CMm <- mm10_rmsk[grep("MT2C_Mm", mm10_rmsk$name),]

length(MT2Mm) #2667
length(MT2CMm) #1982
MT2Mm$length <- end(MT2Mm) - start(MT2Mm)
MT2CMm$length <- end(MT2CMm) - start(MT2CMm)
```

### 9.2.2 input narrowPeak
```{r input narrowPeak}
# declare names and types for the extra BED columns
extraCols_narrowPeak <- c(FoldChange="numeric", pVal="numeric",
qVal="numeric", summit="integer")

#following peak files were input
peak_file_list <- list(
  obox1 = "./R_input/Obox_Dux_Zscan4c_binding/L2C_Obox1_multi_peaks.narrowPeak",
  obox3 = "./R_input/Obox_Dux_Zscan4c_binding/L2C_Obox3_multi_peaks.narrowPeak",
  obox5 = "./R_input/Obox_Dux_Zscan4c_binding/L2C_Obox5_multi_peaks.narrowPeak",
  dux = "./R_input/Obox_Dux_Zscan4c_binding/Dux_18hr_1_multi_peaks.narrowPeak",
  zscan4c = "./R_input/Obox_Dux_Zscan4c_binding/Zscan4c_2CLC_multi_peaks.narrowPeak"
)

peak_list <- list()
for (i in names(peak_file_list)){
  peak_list[[i]] <- import.bed(peak_file_list[[i]],
                               extraCols=extraCols_narrowPeak)
}

sapply(peak_list, length)
```
### 9.2.3 Obox/Dux overlap with MT2
```{r}
obox1_overlap <- findOverlaps(MT2Mm, peak_list$obox1)
MT2Mm$obox1 <- "NotBound"
MT2Mm$obox1[queryHits(obox1_overlap)] <- "obox"
length(MT2Mm[which(MT2Mm$obox1 == "obox"),]) #2461

obox1_overlap2 <- findOverlaps(MT2CMm, peak_list$obox1)
MT2CMm$obox1 <- "NotBound"
MT2CMm$obox1[queryHits(obox1_overlap2)] <- "obox"
length(MT2CMm[which(MT2CMm$obox1 == "obox"),]) #857

obox3_overlap <- findOverlaps(MT2Mm, peak_list$obox3)
MT2Mm$obox3 <- "NotBound"
MT2Mm$obox3[queryHits(obox3_overlap)] <- "obox"
length(MT2Mm[which(MT2Mm$obox3 == "obox"),]) #2154

obox3_overlap2 <- findOverlaps(MT2CMm, peak_list$obox3)
MT2CMm$obox3 <- "NotBound"
MT2CMm$obox3[queryHits(obox3_overlap2)] <- "obox"
length(MT2CMm[which(MT2CMm$obox3 == "obox"),]) #417

obox5_overlap <- findOverlaps(MT2Mm, peak_list$obox5)
MT2Mm$obox5 <- "NotBound"
MT2Mm$obox5[queryHits(obox5_overlap)] <- "obox"
length(MT2Mm[which(MT2Mm$obox5 == "obox"),]) #2319

obox5_overlap2 <- findOverlaps(MT2CMm, peak_list$obox5)
MT2CMm$obox5 <- "NotBound"
MT2CMm$obox5[queryHits(obox5_overlap2)] <- "obox"
length(MT2CMm[which(MT2CMm$obox5 == "obox"),]) #691

dux_overlap <- findOverlaps(MT2Mm, peak_list$dux)
MT2Mm$dux <- "NotBound"
MT2Mm$dux[queryHits(dux_overlap)] <- "dux"
length(MT2Mm[which(MT2Mm$dux == "dux"),]) #2278

dux_overlap2 <- findOverlaps(MT2CMm, peak_list$dux)
MT2CMm$dux <- "NotBound"
MT2CMm$dux[queryHits(dux_overlap2)] <- "dux"
length(MT2CMm[which(MT2CMm$dux == "dux"),]) #203

zscan4_overlap <- findOverlaps(MT2Mm, peak_list$zscan4c)
MT2Mm$zscan4 <- "NotBound"
MT2Mm$zscan4[queryHits(zscan4_overlap)] <- "zscan4"
length(MT2Mm[which(MT2Mm$zscan4 == "zscan4"),]) #803

zscan4_overlap2 <- findOverlaps(MT2CMm, peak_list$zscan4c)
MT2CMm$zscan4 <- "NotBound"
MT2CMm$zscan4[queryHits(zscan4_overlap2)] <- "zscan4"
length(MT2CMm[which(MT2CMm$zscan4 == "zscan4"),]) #7
```

```{r}
MT2Mm_noObox <- subset(MT2Mm, obox1 == "NotBound" &
                              obox3 == "NotBound" &
                              obox5 == "NotBound")
MT2CMm_noObox <- subset(MT2CMm, obox1 == "NotBound" &
                              obox3 == "NotBound" &
                              obox5 == "NotBound")
length(MT2Mm_noObox) #203
length(MT2CMm_noObox) #1027

MT2Mm_allObox <- subset(MT2Mm, obox1 == "obox" &
                               obox3 == "obox" &
                               obox5 == "obox")
MT2CMm_allObox <- subset(MT2CMm, obox1 == "obox" &
                              obox3 == "obox" &
                              obox5 == "obox")
length(MT2Mm_allObox) #2135
length(MT2CMm_allObox) #361

#boxplot(MT2Mm_noObox$length, MT2Mm_allObox$length)
#boxplot(MT2CMm_noObox$length, MT2CMm_allObox$length)

pie(c(length(MT2Mm_noObox),
    length(MT2Mm_allObox),
    length(MT2Mm)-length(MT2Mm_noObox)-length(MT2Mm_allObox)),
    main = "MT2_Mm_Obox")
pie(c(length(MT2CMm_noObox),
    length(MT2CMm_allObox),
    length(MT2CMm)-length(MT2CMm_noObox)-length(MT2CMm_allObox)),
    main = "MT2C_Mm_Obox")

MT2Mm_Dux <- subset(MT2Mm, dux == "NotBound")
MT2CMm_Dux <- subset(MT2CMm, dux == "NotBound")
length(MT2Mm_Dux) #389
length(MT2CMm_Dux) #1779

pie(c(length(MT2Mm_Dux),
      length(MT2Mm)-length(MT2Mm_Dux)),
    main = "MT2Mm_Dux")
pie(c(length(MT2CMm_Dux),
      length(MT2CMm)-length(MT2CMm_Dux)),
    main = "MT2CMm_Dux")


MT2Mm_zscan4 <- subset(MT2Mm, zscan4 == "NotBound")
MT2CMm_zscan4 <- subset(MT2CMm, zscan4 == "NotBound")
length(MT2Mm_zscan4) #1864
length(MT2CMm_zscan4) #1975

pie(
  c(length(MT2Mm_zscan4),
    length(MT2Mm)-length(MT2Mm_zscan4)),
    main = "MT2Mm_Zscan4")
pie(
  c(length(MT2CMm_zscan4),
    length(MT2CMm)-length(MT2CMm_zscan4)),
    main = "MT2CMm_Zscan4")

pdf("./figures/231116_piechart_Dux_Obox_Zscann4_bound.pdf",
    width = 6, height = 6)
pie(c(length(MT2Mm_noObox),
    length(MT2Mm_allObox),
    length(MT2Mm)-length(MT2Mm_noObox)-length(MT2Mm_allObox)),
    main = "MT2_Mm_Obox")
pie(c(length(MT2CMm_noObox),
    length(MT2CMm_allObox),
    length(MT2CMm)-length(MT2CMm_noObox)-length(MT2CMm_allObox)),
    main = "MT2C_Mm_Obox")
pie(c(length(MT2Mm_Dux),
      length(MT2Mm)-length(MT2Mm_Dux)),
    main = "MT2Mm_Dux")
pie(c(length(MT2CMm_Dux),
      length(MT2CMm)-length(MT2CMm_Dux)),
    main = "MT2CMm_Dux")
pie(
  c(length(MT2Mm_zscan4),
    length(MT2Mm)-length(MT2Mm_zscan4)),
    main = "MT2Mm_Zscan4")
pie(
  c(length(MT2CMm_zscan4),
    length(MT2CMm)-length(MT2CMm_zscan4)),
    main = "MT2CMm_Zscan4")
dev.off()
```

# 10 MT2a analyses

To understand why MT2 CRISPRa activates Zscan4c whereas MTi does not affect
Zscan4c.

## 10.1 input data
```{r}
suppressMessages(library(dplyr))
suppressMessages(library(rtracklayer))
source("./utils.R")

MT2a_sampleName <- c("EV1", "EV2", "MT2a1", "MT2a2")
MT2a_simpleName <- MT2a_sampleName

MT2a_counts <- inputTEcountfiles(MT2a_sampleName, paste0(MT2a_simpleName, ".count"),
                                 countDataPath = "./R_input/MT2a_FPKM_TEcount/")
row.names(MT2a_counts) <- MT2a_counts$id

MT2a_fpkm <- inputStringTieRPKMfiles(MT2a_sampleName, paste0(MT2a_sampleName, ".rpkm"),
                                     RPKMDataPath = "./R_input/MT2a_FPKM_TEcount/")
#dir.create("./R_output/MT2a_RNA/")
write.csv(MT2a_fpkm, "./R_output/MT2a_RNA/231116_MT2a_fpkm.csv", quote = F, row.names = F)
#View(MT2a_fpkm)
```

## 10.2 DESeq2 analyses
```{r DEG identification}
foldchange = 2

suppressMessages(
  MT2a_DEseq <- countsToDEseq2FDR(
    counts = MT2a_counts[, c("EV1.count", "EV2.count", 
                             "MT2a1.count", "MT2a2.count")], 
     CGroup = 2, TGroup = 2))


MT2a_DEseq_rpkm <- merge(
  MT2a_DEseq[, c("id", "EV1.count", "EV2.count", 
                             "MT2a1.count", "MT2a2.count",
                 "log2FoldChange", "padj")],
  MT2a_fpkm[, c("id", "name",
                "EV1.rpkm", "EV2.rpkm", "MT2a1.rpkm", "MT2a2.rpkm")], 
  by = "id"
)

MT2a_DEseq_rpkm$padj[is.na(MT2a_DEseq_rpkm$padj)] <- 1

MT2a_DEseq_rpkm.group <- classifyDEG(
  MT2a_DEseq_rpkm,
  ctr.rpkm = c("EV1.rpkm", "EV2.rpkm"),
  trt.rpkm = c("MT2a1.rpkm", "MT2a2.rpkm"),
   FDR.col = "padj", log2FC.col = "log2FoldChange",
                            RPKM = 0, log2FC = log2(foldchange), FDR = 0.05)
table(MT2a_DEseq_rpkm.group)
#down-regulated  similar_level   up-regulated 
#           163          20813            735  

MT2a_DEseq_rpkm <- cbind(MT2a_DEseq_rpkm, MT2a_DEseq_rpkm.group)
write.table(MT2a_DEseq_rpkm, "./R_output/MT2a_RNA/231116_MT2A_DEG.csv", 
            sep = ",", quote = F, row.names = F)
```

```{r}
```

```{r}
```

```{r}
```
