simplify_sg_id <- function(x){
  tmp <- paste(unlist(strsplit(x, split = "[.|,]"))[1:3], collapse = "_")
  tmp <- gsub(pattern = "X", x = tmp, replacement = "")
  return (tmp)
}

Stack_plot <- function(df0, df1, df2, df3, sgRNA, title, max = 7){
  
  tmp_df<- data.frame(mis0 = rowSums(df0[, sgRNA]))
  tmp_df$mis1 <- rowSums(df1[, sgRNA])
  tmp_df$mis2 <- rowSums(df2[, sgRNA])
  tmp_df$mis3 <- rowSums(df3[, sgRNA])
  
  #Now calculate fractions of repeats that are targeted by # of sgRNAs
  tmp0 <- as.data.frame(table(tmp_df$mis0))
  if (nrow(tmp0) < max+1){
    tmp0 <- rbind(tmp0, data.frame(Var1 = paste(c(nrow(tmp0):max)), 
                                   Freq = rep(0, max-nrow(tmp0)+1)))
  }
  tmp0$Pct <- tmp0$Freq / nrow(tmp_df)
  tmp0$grp <- "mis0"
  
  tmp1 <- as.data.frame(table(tmp_df$mis1))
  if (nrow(tmp1) < max+1){
    tmp1 <- rbind(tmp1, data.frame(Var1 = paste(c(nrow(tmp1):max)), 
                                   Freq = rep(0, max-nrow(tmp1)+1)))
  }
  tmp1$Pct <- tmp1$Freq / nrow(tmp_df)
  tmp1$grp <- "mis1"
  
  tmp2 <- as.data.frame(table(tmp_df$mis2))
  if (nrow(tmp2) < max+1){
    tmp2 <- rbind(tmp2, data.frame(Var1 = paste(c(nrow(tmp2):max)), 
                                   Freq = rep(0, max-nrow(tmp2)+1)))
  }
  tmp2$Pct <- tmp2$Freq / nrow(tmp_df)
  tmp2$grp <- "mis2"
  
  tmp3 <- as.data.frame(table(tmp_df$mis3))
  if (nrow(tmp3) < max+1){
    tmp3 <- rbind(tmp3, data.frame(Var1 = paste(c(nrow(tmp3):max)), 
                                   Freq = rep(0, max-nrow(tmp3)+1)))
  }
  tmp3$Pct <- tmp3$Freq / nrow(tmp_df)
  tmp3$grp <- "mis3"
  
  tmp4fig <- rbind(tmp0, tmp1, tmp2, tmp3)
  
  suppressMessages(library(ggplot2))
  suppressMessages(library(cowplot))
  
  fig <- ggplot(tmp4fig, aes(x = grp, y = Pct, 
                             fill = factor(Var1, levels = c(max:0)))) + 
    geom_bar(stat = "identity", position = "fill", color = "black") + 
    scale_fill_grey(start = 0, end = 1) +
    ggtitle(paste0("sgRNA alignments to ", title)) + 
    xlab("Number of mismatches allowed") +
    ylab(paste0("Fraction of ", title)) + 
    labs(fill='sgRNAs aligned\nper LTR') +
    scale_x_discrete(limits = c("mis0", "mis1", "mis2", "mis3"),
                     labels = c("0", "1", "2", "3")) +
    theme_cowplot(16) 
  
  return(fig)
}

simplifyAnnotation <- function(peaks_anno) {
  #input is a vector of annotation generated by annotatePeak fro ChIPseeker
  #output is a vector of simplified
  tmp <- peaks_anno
  tmp <- gsub("Distal Intergenic","Intergenic", tmp)
  
  pos <- grep("5' UTR",tmp)
  tmp[pos] <- "Intragenic"
  
  pos <- grep("3' UTR",tmp)
  tmp[pos] <- "Intragenic"
  
  pos <- grep("Exon",tmp)
  tmp[pos] <- "Intragenic"
  
  pos <- grep("Intron",tmp)
  tmp[pos] <- "Intragenic"
  
  pos <- grep("Downstream",tmp)
  tmp[pos] <- "Intergenic"
  
  pos <- grep("Promoter",tmp)
  tmp[pos] <- "Promoter"
  
  return (tmp)
}

inputTEcountfiles <- function(sampleName, simpleName, countDataPath){
  
  suppressMessages(library(dplyr))
  
  count <- data.frame()
  for (i in 1:length(sampleName)){
    inFile <- paste0(countDataPath, sampleName[i], ".cntTable")
    
    count.tmp <- read.table(inFile, sep = "\t", header = T, stringsAsFactors = FALSE)
    colnames(count.tmp) <- c("id", "count")
    #count.tmp <- count.tmp[,c("Gene.ID", "Gene.Name", "FPKM")]
    #for duplicated IDs, just sum up.
    #rpkm.tmp <- aggregate(FPKM~Gene.ID+Gene.Name, data = rpkm.tmp, FUN=sum)
    
    if(nrow(count) == 0){
      count <- count.tmp
      
    } else {
      count <- left_join(count, count.tmp, by = "id")
    }
  }
  colnames(count) <- c("id", as.character(simpleName))
  return(count)  
}

inputStringTieRPKMfiles <- function(sampleName, simpleName, RPKMDataPath){
  
  suppressMessages(library(dplyr))
  
  rpkm <- data.frame()
  for (i in 1:length(sampleName)){
    inFile <- paste0(RPKMDataPath, sampleName[i], ".gene_abund.txt")
    
    rpkm.tmp <- read.table(inFile, sep = "\t", header = T, stringsAsFactors = FALSE)
    
    rpkm.tmp <- rpkm.tmp[,c("Gene.ID", "Gene.Name", "FPKM")]
    #for duplicated IDs, just sum up.
    rpkm.tmp <- aggregate(FPKM~Gene.ID+Gene.Name, data = rpkm.tmp, FUN=sum)
    
    if(nrow(rpkm) == 0){
      rpkm <- rpkm.tmp[,c("Gene.ID", "Gene.Name", "FPKM")] 
      
    } else {
      rpkm <- left_join(rpkm, rpkm.tmp[,c("Gene.ID", "FPKM")], by = c("Gene.ID"))
    }
  }
  colnames(rpkm) <- c("id", "name",
                      as.character(simpleName))
  return(rpkm)
}

classify_distance <- function(tmp, ignore_direction = F){
  bin <- rep(0, length(tmp))
  
  if(ignore_direction){
    for (i in 1:length(bin)){
      if(abs(tmp[i]) >=0 & abs(tmp[i]) < 10000){ bin[i] <- "0-10kb" }
      else if (abs(tmp[i]) >= 10000 & abs(tmp[i]) < 30000){ bin[i] <- "10-30kb" }
      #else if (tmp[i] >= 20000 & tmp[i] < 30000){ bin[i] <- "20-30kb" }
      else if (abs(tmp[i]) >= 30000 & abs(tmp[i]) < 50000){ bin[i] <- "30-50kb" }
      #else if (tmp[i] >= 40000 & tmp[i] < 50000){ bin[i] <- "40-50kb" }
      else { bin[i] <- ">50kb" } 
    }
  } else{
    for (i in 1:length(bin)){
      if(tmp[i] >=0 & tmp[i] < 10000){ bin[i] <- "0-10kb" }
      else if (tmp[i] >= 10000 & tmp[i] < 30000){ bin[i] <- "10-30kb" }
      #else if (tmp[i] >= 20000 & tmp[i] < 30000){ bin[i] <- "20-30kb" }
      else if (tmp[i] >= 30000 & tmp[i] < 50000){ bin[i] <- "30-50kb" }
      #else if (tmp[i] >= 40000 & tmp[i] < 50000){ bin[i] <- "40-50kb" }
      else if (tmp[i] >= 50000) { bin[i] <- ">50kb" }
      else if (tmp[i] >= -10000 & tmp[i] < 0) { bin[i] <- "-0-10kb" }
      else if (tmp[i] >= -30000 & tmp[i] < -10000) { bin[i] <- "-10-30kb" }
      #else if (tmp[i] >= -30000 & tmp[i] < -20000) { bin[i] <- "-20-30kb" }
      else if (tmp[i] >= -50000 & tmp[i] < -30000) { bin[i] <- "-30-50kb" }
      #else if (tmp[i] >= -50000 & tmp[i] < -40000) { bin[i] <- "-40-50kb"}
      else { bin[i] <- "< -50kb"}
    }
  }
  return(bin)
}

countsToDEseq2FDR <- function(counts, CGroup = 0, TGroup = 0, 
                              min_read = 10){
  
  suppressMessages(library(DESeq2))
  
  counts <- counts[apply(counts, 1, function(x){sum(x)}) > min_read, ]
  groups <- factor(c(rep("CGroup",CGroup),rep("TGroup",TGroup)))
  sampleInfo <- data.frame(groups, row.names = colnames(counts))
  
  dds <- DESeqDataSetFromMatrix(countData = counts, colData = sampleInfo, design = ~ groups)
  dds$groups = relevel(dds$groups,ref="CGroup")
  dds <- DESeq(dds)
  res <- results(dds,independentFiltering=F)
  
  #normalized counts obtained based on DESeq2 "plotCounts" source code
  normCounts <- counts(dds, normalized = TRUE)
  results <- as.data.frame(cbind(normCounts, res))
  results$id <- row.names(results)  
  return (results)
}

normalizeCounts <- function(counts, min_read = 10){
  #input: output of function: inputTEcountfiles
  
  counts <- counts[apply(counts[, c(2:ncol(counts))],
                         1,
                         function(x){sum(x)}) > min_read,]
  
  #randomly divided samples into two groups ofr normalization
  group1_num <- round((ncol(counts)-1))/2
  group2_num <- ncol(counts) - 1 - group1_num
  groups <- factor(c(rep("CGroup", group1_num),
                     rep("TGroup", group2_num)))
  
  sampleInfo <- data.frame(groups, 
                           row.names = colnames(counts)[2:ncol(counts)])
  
  row.names(counts) <- counts$id
  dds <- DESeqDataSetFromMatrix(countData = counts[2:ncol(counts)], 
                                colData = sampleInfo, design = ~ groups)
  dds <- DESeq(dds)
  
  #normalized counts obtained based on DESeq2 "plotCounts" source code
  normCounts <- counts(dds, normalized = TRUE)
  return(normCounts)
}

classifyDEG <- function(df, ctr.rpkm, trt.rpkm, FDR.col, log2FC.col,  
                        log2FC = 1, RPKM = 1, FDR = 0.05){
  
  df$ctr.mean <- rowMeans(df[ctr.rpkm])
  df$trt.mean <- rowMeans(df[trt.rpkm])
  group <- rep(0, nrow(df)) 
  
  for (i in 1:nrow(df)){
    if(df$ctr.mean[i] >= RPKM | df$trt.mean[i] >= RPKM){
      if(df[log2FC.col][i,] > log2FC & df[FDR.col][i,] < FDR){
        group[i] <- "up-regulated"
      }
      else if (df[log2FC.col][i,] < -log2FC & df[FDR.col][i,] < FDR){
        group[i] <- "down-regulated"
      }
      else{
        group[i] <- "similar_level"
      }
    } 
    else{
      group[i] <- "low_expression_level"
    }
  }
  return(group)
}

ggScatterplot <- function(df, x, y, group, gene,
                          my.color=c("blue", "grey50", "grey50", "red3"),
                          label.up, label.down, xlab, ylab, title,
                          genes4Label = NULL,
                          FC.line = 2){
  
  suppressMessages(library(ggplot2))
  suppressMessages(library(cowplot))
  suppressMessages(library(ggrepel))
  suppressMessages(library(gridExtra))
  suppressMessages(library(grid))
  
  g <- ggplot(df, aes_string(x=x, y=y, color=group,label=gene)) + 
    geom_point() + scale_color_manual(values=my.color) + 
    xlab(xlab) + ylab(ylab) + ggtitle(title) + theme_cowplot(13) + 
    theme(legend.position = "none") + xlim(0, 25) + ylim(0, 25) + 
    geom_abline(intercept = log2(FC.line), slope = 1, linetype = 2) + 
    geom_abline(intercept = -log2(FC.line), slope = 1, linetype = 2) + 
    geom_abline(intercept = 0, slope = 1, linetype = 1) +
    annotate("text", x = -Inf, y = Inf, label = label.up, 
             col = "red3", size = 3.8, hjust = -0.2, vjust = 1.5) + 
    annotate("text", x= Inf, y =-Inf, label = label.down, 
             col = "blue", size = 3.8, hjust = 1.2, vjust = -1) + 
    geom_text_repel(
      data = subset(df, df[, gene] %in% genes4Label),
      size = 3.8, segment.size = 0.3, segment.color = "black",
      direction = "x", nudge_y = 4.5, nudge_x =-3.5,
      point.padding = 0.25, box.padding = 0.25) +
    geom_point(
      data = subset(df, df[, gene] %in% genes4Label), 
      col = "black", size = rel(1.5))
  return(g)  
}

plotGoBars <- function(go_res,top=10, col="red", title="top 10 enriched GO"){
  tmp <- as.data.frame(go_res)
  ord <- order(tmp$p.adjust, decreasing = F)
  tmp <- tmp[ord,]
  tmp <- head(tmp, top)
  tmp$Description <- factor(tmp$Description, levels = tmp$Description)
  g <- ggplot(tmp, aes(x=Description, y=-log10(p.adjust) )) + 
    geom_bar(stat = "identity", fill = col) + coord_flip() + 
    geom_text(aes(label = -log10(p.adjust)), vjust = 1.5, colour = "white") +
    theme_cowplot(13) + ggtitle(title) +  ylab("") +
    scale_x_discrete(limits = rev(levels(tmp$Description))) 
  theme(plot.title = element_text(color="black",face="bold",hjust = 0.5)) 
  g
}

plotAnnoStats <- function(peaks_anno, ttl, simplify=F, category_order=NULL) {
  require(dplyr)
  anno_df <- data.frame()
  
  for(i in 1:length(peaks_anno)){
    tmp <- peaks_anno[[i]]@annoStat
    tmp$Cell <- names(peaks_anno)[i]
    tmp$Feature <- as.character(tmp$Feature)
    
    if(simplify){
      pos <- grep("Exon",tmp$Feature)
      tmp$Feature[pos] <- "Exon"
      
      pos <- grep("Intron",tmp$Feature)
      tmp$Feature[pos] <- "Intron"
      
      pos <- grep("Downstream",tmp$Feature)
      tmp$Feature[pos] <- "Distal Intergenic"
      
      pos <- grep("Promoter",tmp$Feature)
      tmp$Feature[pos] <- "Promoter"
      
      tmp$Feature <- gsub("5' UTR","Exon",tmp$Feature)
      tmp$Feature <- gsub("3' UTR","Exon",tmp$Feature)
      tmp$Feature <- gsub("Distal Intergenic","Intergenic",tmp$Feature)
      tmp$Feature <- factor(tmp$Feature, levels = rev(c("Promoter","Exon","Intron","Intergenic")))
    }
    
    anno_df <- rbind(anno_df, tmp)
  }
  
  anno_df = anno_df %>% group_by(Feature,Cell) %>% summarize(Frequency = sum(Frequency))
  
  anno_df$Feature <- factor(anno_df$Feature)
  
  if(is.null(category_order)){
    anno_df$Cell <- factor(anno_df$Cell)
  }else{
    anno_df$Cell <- factor(anno_df$Cell, levels = category_order)
  }
  
  
  g <- ggplot(anno_df, aes(x=Cell, y=Frequency, fill=Feature)) +
    coord_flip() +
    geom_bar(stat="identity") + ggthemes::theme_base() +
    #scale_fill_brewer(palette = "Set1") +
    scale_fill_npg(breaks=c("Promoter","Exon","Intron","Intergenic")) +
    ggtitle(ttl)+
    xlab("Cell") + ylab("Percentage (%)") + theme(plot.title = element_text(hjust = 0.5,face = "plain",size = 12),
                                                  axis.text = element_text(color="black")) #+ boldText
  g
}

ggMAplot <- function(df, x, y, group, gene,
                     my.color=c("blue", "grey50", "grey50", "red3"),
                     label.up, label.down, xlab, ylab, title,
                     genes4Label = NULL,
                     FC.line = 2){
  
  suppressMessages(library(ggplot2))
  suppressMessages(library(cowplot))
  suppressMessages(library(ggrepel))
  suppressMessages(library(gridExtra))
  suppressMessages(library(grid))
  
  g <- ggplot(df, aes_string(x=x, y=y, color=group,label=gene)) + 
    geom_point() + scale_color_manual(values=my.color) + 
    xlab(xlab) + ylab(ylab) + ggtitle(title) + theme_cowplot(13) + 
    theme(legend.position = "none") + xlim(0, 22) + ylim(-10, 10) + 
    geom_abline(intercept = log2(FC.line), slope = 0, linetype = 2) + 
    geom_abline(intercept = -log2(FC.line), slope = 0, linetype = 2) + 
    geom_abline(intercept = 0, slope = 0, linetype = 1) +
    annotate("text", x = -Inf, y = Inf, label = label.up, 
             col = "red3", size = 3.8, hjust = -0.2, vjust = 1.5) + 
    annotate("text", x= Inf, y =-Inf, label = label.down, 
             col = "blue", size = 3.8, hjust = 1.2, vjust = -1) + 
    geom_text_repel(
      data = subset(df, df[, gene] %in% genes4Label),
      size = 3.8, segment.size = 0.3, segment.color = "black",
      direction = "x", nudge_y = 4.5, nudge_x =-3.5,
      point.padding = 0.25, box.padding = 0.25) +
    geom_point(
      data = subset(df, df[, gene] %in% genes4Label), 
      col = "black", size = rel(1.5))
  return(g)  
}


#classify gene repeat relative orientation
classify_repeat_gene_pair <- function(df){
  df$group <- rep("others", nrow(df))
  for(i in 1:nrow(df)){
    if(df$gene_strand[i] != df$peak_strand[i]){ #repeat and gene different orientation
      if(df$distance[i] >= 0){ #repeat is down-stream of a gene or intragenic 
        df$group[i] <- "anti_down/intragenic"
      } else {
        df$group[i] <- "anti_up"
      }
    } else { #repeat and gene the same orientation
      if(df$distance[i] >=0){ #repeat is upstream of a gene or intragenic
        df$group[i] <- "sense_up/intragenic"
      } else {
        df$group[i] <- "sense_down"
      }
    }
  }
  return(df$group)
}

calcPeaksSignal <- function(windows, bw, method = "mean"){
  
  # check if windows have width > 1
  if( any(width(windows)==1) ){
    stop("provide 'windows' with widths greater than 1")
  }
  
  
  
  bwscores <- import.bw(bw, which= windows)
  covs = coverage(bwscores, weight=bwscores$score)
  covs = covs[seqlevels(windows)]
  windows <- GenomicRanges::binnedAverage(windows, covs, "meanScore")
  
  
  # set a uniq id for the GRanges
  #windows.len=length(windows)
  #windows = genomation:::constrainRanges(covs, windows)
  
  # fetches the windows and the scores
  # chrs = sort(intersect(names(covs), as.character(unique(seqnames(windows)))))
  # windows2 = split(windows,seqnames(windows))
  # myViews=Views(covs[chrs],as(windows2,"RangesList")[chrs]) # get subsets of RleList
  #
  # #  get a list of matrices from Views object
  # #  operation below lists a matrix for each chromosome
  # if(method == "mean"){
  #   mat = lapply(myViews,function(x) mean(x) )
  # }else{
  #   mat = lapply(myViews,function(x) sum(x) )
  # }
  #
  # windows$meanScore = 0
  #
  # for(chrom in names(mat)){
  #   tmp = subset(windows, seqnames == chrom)
  #   windows$meanScore[tmp$X_rank] = mat[[chrom]]
  # }
  return(windows)
}